---
output:
  pdf_document: default
  html_document: default
---
# Introduction

## Introduction to gene expression regulation 
Gene expression is determined by a number of genetic elements. Perhaps the best studied is the role of the promoter sequence, which plays a central role in coordinating the rate of new transcript production (Ottoz et al. 2018). The promoter encompasses two major regions, the core promoter and upstream regulatory elements. The core promoter region contains the essential elements required to initiate transcription, such as the transcription initiation start site (TSS). Upstream regulatory elements (upstream activating sequences - UAS and upstream repressing sequences - URS) can modulate promoter activity through the recruitment of single transcription factors or synergistic combinations to facilitate recruitment of transcriptional machinery and ultimately enhance transcription rates. In contrast to the promoter, the terminator coordinates transcriptional termination and polyadenylation of the mRNA at the 3’-terminal end. Within the terminator are three functional elements, 1) the efficiency element, 2) positioning element and 3) the actual polyadenylation signal (Guo et al. 1996). The efficiency element determines the efficiency of the 3’-end formation and the positioning element serves to direct cleavage to a position ~20 bases downstream of the element. 
 
## Introduction to 3'UTRs
Encoded within the terminator sequence, the 3’ untranslated region (UTR) has been shown to play an important role in coordinating transcript stability and localisation (Mayr 2016). These functions are largely mediated by the association of RNA binding proteins (RBP) to the corresponding cis-regulatory motif on the 3’UTR and subsequent recruitment effector proteins. Therefore the changes to the mRNA fate is determined by the available binding sites on the 3’UTR for RBPs at a given moment. Several RBPs have also been found to have dual roles, functioning as metabolic enzymes and moonlighting as regulators of mRNA (Sherrer et al. 2010). This suggests a direct connection via a feedback loop between the yeast metabolism and post-transcriptional regulation of related genes (Hogan et al. 2008, Tsvetanova et al. 2010, Beckmann et al. 2016).These RNA-protein interactions are dynamic, and rapid restructuring can occur in response to metabolic or environmental stimuli. The accessibility of RBPs to these cis-regulatory elements are also influenced by structural modifications or localisation to different compartments of the cell. Formation of secondary or tertiary structures have been suggested to occlude RBPs from binding to cis-regulatory elements in the 3’UTR. The formation of a  stable stem loop within the 3’UTR has also shown to have a stabilising effect (Giesberg et al. 2014). 

RBPs have been found to bind many mRNA targets however their binding specificity is increased through their association with protein interaction partners. The cooperativity between multiple RBP binding to adjacent cis-regulatory motifs in an mRNA have been shown to improve affinity and specificity of She2p and She3p RBP blinding to ASH1 mRNA compared to their individual interactions (Shen et al.  2009) (Niedner et al. 2014). This joint binding of both RBPs along with Myo4p are necessary for the localisation of the mRNA-protein complex to the bud tip. Adjacent cis-regulatory motifs on a 3’UTR or motifs brought into close proximity due to secondary structures can also have a competitive interaction, with preferential binding of one RBP through improved affinity mediated under specific conditions (Dassi 2017). 

Another factor that influences mRNA abundance post-transcriptionally is the choice of promoter used to drive the expression of a coding sequence. In particular the trans-acting factor, Rap1 and the presence of the Rap1p-binding sites on the UAS have been proposed to alter the messenger ribonucleoprotein (mRNP) complex composition prior to its export to the cytoplasm, and subsequently lead to an increase in mRNA transcript turnover (Bregman et al. 2011).

The majority of work on the impact of different yeast 3’UTRs on transcript abundance has primarily focussed on the identification of 3’UTRs/terminators that are associated with an increased protein production for their application for metabolic engineering of yeast, in conjunction with strong promoters. One example is the tTPS1 which resulted in a 20% higher protein output due to improved mRNA stability (Yamanishi et al. 2011). Similarly, Ito et al. identified tDIT1 was associated with a 2.5x improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). However, important questions emerge around the independence of promoter and 3’UTR/terminator choice in relation to transcript abundance. If the promoter sequence and 3’UTR/terminator sequence do not independently impact transcript abundance then there is the potential for unexpected interactions and unpredictable transcript abundance. 

Another question worth exploring is the composition and grammar of sequence elements of 3’UTRs that contribute to an improved stability of an mRNA. With improved sequencing methods and analysis of large data sets, much work has been carried out over the past decade to mine for cis-regulatory elements in yeast 3’UTRs associated with different mRNA fate (localisation, stability or decay). 


## Computational biology of 3’UTRs and motifs (Computational approaches) 
Searching for novel cis-regulatory elements, such as 3’UTR motifs, associated with post-transcriptional mechanisms consists of filtering for mRNA transcripts suspected to be enriched with motifs and comparing these sequences to determine a common motif.  mRNA transcripts have previously been filtered by using functional transcriptomic data sets to select for those with high abundance or half lives (Shalgi et al., 2005, Cheng et al., 2017). Alternatively, enriched binding of RNA to proteins can be experimentally confirmed, using immunoprecipitation and UV cross linking, providing either whole transcripts to compare (Riordan, Herschlag and Brown, 2010) or specific intra-transcript binding regions, (Zhao et al., 2010, Granneman et al, 2009). Given transcripts/regions of interest, the deduction of actual consensus sequences is itself non-trivial.  Motifs can be of variable length and consist of highly redundant nucleotides separating conserved regions. Current motif finding packages, such as MEME-suite (Bailey et al., 2015), overcome these issues by first finding smaller seed sequences that regularly occur in the enriched transcripts/regions and then attempting to pairs these up into longer motifs (Hogan et al., 2008). Finally, recent developments in efficient algorithms for fitting linear models also allow for the detection of  novel cis-regulatory elements. A model predicting the results of a functional genomics data set, i.e. genome-wide transcript abundance, using all possible contributing elements can be filtered down by penalised maximum likelihood to only those elements with significant contributions (Vijayabaskar et al., 2019).
 
## 3’UTR screens (Experimental approaches)
[Section still to be added]



The portability of these cis-regulatory motifs have been tested by introducing a putative destabilisation motif ‘ATATTC’ into different 3’UTRs, showing a decrease in mRNA stability for both 3’UTRs tested. However, the magnitude of changes in mRNA destabilisation with the presence of the destabilising motif varied between 3’UTRs, suggesting the interplay of other factors in determining mRNA fate (Cheng et al. 2017). Therefore, another important question needs to be addressed - Are motifs in relation to mRNA abundance independent of 3’UTR context?

## Overview of the results

Here, we utilise modular cloning to construct chimeras to investigate the contribution of 3’UTRs on gene expression. Following a similar approach carried out by Cheng et al. (2017), we computationally predict the mRNA half-life in yeast with linear combinations of sequence features. We modelled the log2 half-life as a linear combination of codon usage and motif counts following Cheng et al. (2017). Using the model, we have isolated several stabilising and decay motifs and probed their role on mRNA fate using a modular cloning assembly method. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs. 
