# Introduction

Our understanding of gene expression relies on the abstraction of a gene into discrete cis-regulatory elements (CRE), or ‘parts’, based on their most prominent function.
The promoter coordinates the initiation of transcription.
Transcription then progresses across exons and introns to the terminator, where cleavage and polyadenylation of the nascent RNA transcript is coordinated.
Extending the abstraction to the messenger RNA transcript, the coding sequence is flanked by  untranslated regions (UTRs) at its 5' and  3' ends.
The 5'UTR contributes to translation regulation and the 3'UTR contributes to transcript stability and localisation.
Assigning specific characteristics to individual elements of a gene has facilitated the assumption that "parts" are modular or interchangeable with other parts.
However, it is important to distinguish two concepts: modular assembly, by which we mean how easy it is to make combinations of parts; and, composability, by which we mean how combinations of parts function as predicted from the behaviour of those parts in other contexts.

Modular assembly is now routine: modern DNA synthesis and cloning techniques allow the construction of complex gene pathways through the combinatorial assembly of chosen parts [@Curran2013; @Lee2015; @Garcia2018; @Andreou2018]. 
Several groups have constructed standardised libraries comprising promoter (including 5'UTR) and terminator (including 3'UTR) parts, which can be combined to achieve desired expression of synthesized proteins.
The characterisation of all possible promoter-terminator pairings is impractical, so standardised libraries generally characterise parts independently, for example testing many promoters with the same coding sequence and terminator [@Lee2015].
Similarly, even massively parallel approaches to characterise smaller CREs usually explore a large library of promoter elements with a single terminator [cite Segal promoter work], or conversely a library of promoters with a single terminator [@Shalem2015].
These experiments do not measure interactions of different parts, so are not informative about composability of parts.

Computational methods for discovery and quantification of short CREs, such as sequence motifs recognised by regulatory proteins, also tend to rely on composability.
Prominent methodologies for detecting CRE's include predicting functional data with sequence features, often using linear models [@Shalgi2005; @Cheng2017; @Vijayabaskar2019]; or, directly comparing the sequences of genes with similar characteristics to determine the presence of consensus sequence motifs [@Hogan2008; @Kretz2013] using motif discovery software [@Elemento2007; @Bailey2015].
Both these approaches make the implicit approximation that the contribution of short CREs is independent of context, so that the effect of combining motifs is "composed" by a linear sum (on the appropriate scale) of the individual CRE contributions.
These approximations miss multi-part motifs, interactions between motifs, or the fact that motifs may be active only in specific contexts.
The approximation that short CREs act independently helps to find elements that have clear contributions, and to simplify a vast search space that would be made exponentially larger by accounting for interactions.

However, the importance of CRE interactions has been clear since the discovery of cooperative DNA binding by transcription factors [@Giniger1988].
Short post-transcriptional regulatory CREs are functional or not depending on the RNA sequence context [Taliaferro et al., 2016].
A large-scale study of the composability of adjacent bacterial promoters and ribosomal binding sites showed that gene expression output of 20% of the constructs had a over 2-fold deviation, and 5% had over 13-fold deviation, from predictions of a linear model without interactions [@Kosuri2013].
A study of promoter-terminator swaps in budding yeast showed some promoter-terminator interactions [@Wei2017], although these are 100's of nucleotides apart. 

Here, we provide further evidence for the limitations of quantitative composability in regulatory regions, and show that the effect of short CREs depends on host gene context. 
First, we use modular cloning to construct 120 combinations of promoter (including 5'UTR), coding sequence, and terminator (including 3'UTR) modules, showing that the quantitative effect of terminators depends on both promoter and coding sequence.
We then investigate the effect of promoter and terminator context on the regulatory behaviour of short CREs by inserting shortlisted 3'UTR motifs into different constructs.
In order to select for suitable 3'UTR motifs, we adapt a method to computationally predict mRNA log2-half-life as a linear combination of codon usage and motif counts [@Cheng2017].
We then quantify their effect on mRNA and protein output when inserted in, or removed from, distinct 3'UTRs in reporter genes.
We further explore the dependence of promoter selection on the relationship between these motifs and mRNA abundance.
Our results demonstrate that the incorporation of candidate decay motifs indeed results in a decrease in RNA abundance, however their quantitative contribution depends the host 3'UTR, paired promoter and chosen ORF.
