# Introduction
Saccharomyces cerevisiae is a well-established host platform for industrial production due to its traits as a robust organism under fermentation conditions, the wealth of molecular biology tools developed for its genetic manipulation and its status as a generally regarded as safe (GRAS) organism. Over the past two decades, it has been optimised for the production of antibiotics, painkillers, antioxidants and anti-cancer drugs (Siewers et al 2009, Gidijala et al. 2009, Yan et al. 2005, Koopman et al. 2012, Rodriguez et al. 2017). This process for pathway engineering is reliant on precise capabilities in modulating native and heterologous enzyme expression. Over the past few years, the development of an increasing number of modular toolkits containing standardised genetic parts for the rapid and combinatorial assembly of genes have expedited our capability to undergo multiple iterations of the design-build-test-learn cycle for pathway optimisatiion (Lee et al. 2015, Garcia-Ruiz et al. 2018, Andreou et al. 2018). 

These toolkits often utilise native yeast promoter/5'UTR and 3'UTR/terminator regions to provide parts to enable modular assembly of genes for pathway expression. Non-coding regulatory elements, comprising of promoters, untranslated regions (5’UTR and 3’UTR) and terminators, possess a large amount of information for the regulation of mRNA and protein abundance. Perhaps the best studied is the role of the promoter, which plays a central role in coordinating the rate of new transcript production (Ottoz et al. 2018). Transcription factor (TF) binding sites upstream of the core promoter region modulate promoter activity through the recruitment of single TFs or synergistic combinations to facilitate recruitment of transcriptional machinery. Coupled with chromatin context and other bound TFs, these ultimately control transcription rates (Levo 2017, Blazeck et al. 2012). The importance of terminator choice in gene expression has not been as extensively studied, with only a handful of common terminators (tADH1, tCYC1, tPGK1) often used for heterologous expression or pathway asssemblies.

Several groups have sought to elucidate the effect of different native yeast terminators with a goal of maximal protein expression. Yamanishi et al. identified a terminator (tTPS1) which increased fluorescent protein production by 50% compared to the tCYC1 terminator (Yamanishi et al. 2011). Ito et al. identified tDIT1 was associated with a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). The same group further identified that the effect of a library of 5302 terminators spanned an 81-fold range difference in fluorescence production, with several resulting in a doubling of protein production relative to tPGK1 (Yamanishi et al. 2013). When coupled with different promoters, the relative difference in protein output between terminators vary, with a maximum difference of 11-fold when coupled with a low expression promoter (Curran et al. 2013, Wei et al. 2017).

The identification of major functional elements (efficiency element, positioning element and the polyadenylation signal) in the terminator as well as the construction of a minimal terminator (Guo and Sherman 1996) formed the basis for the synthesis and charaterisation of large synthetic terminator libraries (Curran et al. 2015).

The functional mechanisms through which the terminator region controls gene expression are confounded by its involvement in transcriptional, post-transcriptional and translational regulation. recoordinates transcriptional termination and polyadenylation of the mRNA at the 3’-terminal end (Guo et al. 1996).  

Flanking the coding sequence on the mRNA are untranslated regions (UTR) which have been shown to play important roles in coordinating mRNA stability, translation and localisation. These functions are largely mediated by the association of RNA binding proteins (RBPs) to the corresponding cis-regulatory motif on the UTR and subsequent recruitment effector proteins. These RNA-protein interactions are dynamic, and rapid restructuring can occur in response to metabolic or environmental stimuli. The accessibility of RBPs to these cis-regulatory elements are also influenced by structural modifications or localisation to different compartments within the cell Formation of secondary or tertiary structures have been suggested to prevent or permit RBPs from binding to cis-regulatory elements in the 3’UTR (**reference**). RBPs have also been found to bind many mRNA targets however their binding specificity can be modulated through their association with protein interaction partners. For example She2p and She3p RBPs show enhanced blinding to ASH1 mRNA compared to their individual interactions, leading to localisation of ASH1 mRNA to the budding daughter cell (Shen et al.  2009, Niedner et al. 2014). As such changes to the mRNA fate is determined by the available binding sites on the UTR for RBPs at a given moment. 

The identification and dissection of these RNA-protein interactions networks have relied on in vitro experimental methods, that centre around the immobilisation of either mRNA (Hartmuth et al. 2004) or proteins (Kretz et al. 2013, Siprashvili et al. 2016) before exposing them to cell extract. Whilst these approaches maximise control, they remove much of the cellular context, with synthesised mRNA being devoid of post transcriptional chemical modifications. In contrast, in vivo strategies centre around cross-linking RNA and RBPs in living cells using UV or formaldehyde to form covalent bonds between RNA and proteins that are in direct contact with each other. Formaldehyde based cross-linking has the advantage of much higher efficiency of cross-linking between RNA and RNA binding proteins, however a key caveat is that protein-protein cross-linking also occurs, meaning that it can be more challenging to discern between RBPs and proteins that interact with RBPs. In contrast UV cross-linking approaches are much more specific, however not only is cross-linking less efficient, but further biases have been observed for cross-linking with uridines (Sugimoto et al. 2012) and varying efficiencies of cross-linking are seen across different amino acids (Meisenheimer et al. 1997).

Enriched binding of RNA to proteins experimentally validated using methods previously described provide either whole transcripts (Riordan, Herschlag and Brown, 2010) or specific intra-transcript binding regions (Zhao et al., 2010, Granneman et al, 2009) for comparison to determine a common motif. Alternatively, searching for novel cis-regulatory elements, such as 3’UTR motifs, consists of filtering for mRNA transcripts suspected to be enriched with motifs and comparing these sequences to determine a common motif.  mRNA transcripts have previously been filtered by using functional transcriptomic data sets to select for transcripts with high abundance or half lives (Shalgi et al., 2005, Cheng et al., 2017). Motifs can be of variable length and consist of highly redundant nucleotides separating conserved regions. Current motif finding packages, such as MEME-suite (Bailey et al., 2015), overcome these issues by first finding smaller seed sequences that regularly occur in the enriched transcripts/regions and then attempting to pairs these up into longer motifs (Hogan et al., 2008). Finally, recent developments in efficient algorithms for fitting linear models also allow for the detection of novel cis-regulatory elements. A model predicting the results of a functional genomics data set, i.e. genome-wide transcript abundance, using all possible contributing elements can be filtered down by penalised maximum likelihood to only those elements with significant contributions (Vijayabaskar et al., 2019).

The paragraph is fine. I would divide it up a bit differently. The difference between searching for CREs in post-tx and tx'l regulation is about what data sets and which candidate regions, not about the methods I think. So separate those ideas. I like the discussion of functional enrichment vs binding enrichment, and the distinction between whole-trasncript and intra-transcript measurement. The glms discussion needs Julien Gagneur's papers referenced, and also a discussion of k-mers directly. The MEME-like approach is to find motif profiles (really position-weight-matrices) from searching sequences, which automatically allows redundancy. The Cheng/Gagneur or DREME-like approach is to first search for k-mers whose exact match is predictive and then afterwards improve that by allowing redundancy or overlap.

Here, we utilise modular cloning to construct reporters, consisting of different combinations of promoter-5'UTR and 3'UTR-terminator swaps to better understand the contribution of 3’UTRs on gene expression. We adapt a method used by Cheng et al. to computationally predict mRNA half-life in yeast, with linear combinations of sequence features. This was achieved by modelling the log2 half-life as a linear combination of codon usage and motif counts (Cheng et al. 2017). Using our model, we identify several putative stabilisation and decay motifs and subsequently probe their role on mRNA fate when inserted into different 3’UTRs or removed from 3’UTRs containing these motifs. We further explore the dependence of promoter selection on the relationship between these motifs and mRNA abundance. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs.
