# Introduction

Our understanding of gene expression relies on the abstraction of a gene into discrete modules, or ‘parts’, based on their characterised function. At the DNA level, the promoter sequence coordinates the initiation of transcription and the rate at which this occurs. Transcription then progresses through the downstream promoter region and coding sequence (CDS), before termination and polyadenylation of the nascent transcript is coordinated by the terminator sequence. Building from this model for genetic determinants of protein expression, a number of standardised and modularised parts libraries have been designed to allow combinatorial assembly of parts and subsequent construction of complex pathways (Lee et al. 2015, Garcia-Ruiz et al. 2018, Andreou et al. 2018). Subsequent efforts have focussed on the the characterisation of these parts libraries assuming independent contribution to gene expression from promoter (Lee et al. 2015, Guo et al. 2015, Lubliner et al. 2013) and terminator parts (Yamanishi et al. 2013). In addition, the characterised libraries have also facilitated the discovery of novel sequence determinants of gene expression by comparing the sequences of parts with similar characteristics. In both the discovery and manipulation of genetic determinants of protein expression, the pragmatic assumption that each part functions independently has aided research by reducing the experimental degrees of freedom. However, the validity of this assumption and the consequences when parts do interact remains understudied. 

Previous papers characterising the effect of different native yeast terminators on protein expression have begun to consider differences in expression when different promoter pairings are used. It is known that with a given promoter terminator selection alone can double fluorescent protein production (Yamanishi et al. 2013). However, when coupled with different promoters, the relative difference in protein output between terminators vary, with a maximum difference of 11-fold when coupled with a low expression promoter (Curran et al. 2013, Wei et al. 2017). Interestingly, it has also been possible to associate expression changes directly to sequence elements within a terminator by identifing a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). The evidence for interactions between parts is incomplete but the possible implications for the discovery of novel regulatory sequences elements within these parts remains of interest.

The identification of sequence elements within genes that control gene expression, otherwise known as cis-regulatory elements, have well established experimental and computational methods, but almost always under the assumption of independence between parts. Lists of suspected cis-regulatory elements have been collated either by grouping similar genes using abundance or half life transcriptomic data sets and then searching for common motifs (Shalgi et al., 2005, Cheng et al., 2017), or by experimentally verifying protein binding sites with cross-linking assays (Zhao et al., 2010, Granneman et al, 2009) or with sequencing mRNA bound to immobilised proteins and then building consensus sequences (Kretz et al. 2013, Siprashvili et al. 2016). Whether specific intra-transcript binding regions or whole transcripts of interest are available, the task of motif discovery is non-trivial as they can be of variable length and consist of highly redundant nucleotides separating conserved regions. Packages like the MEME-suite (Bailey et al., 2015), excel at determining short consensus sequences by finding the likelihood of each RNA base appearing at each position using the set of suspected binding sites, which automatically allows redundancy. Alternatively, the transcripts of interest can be broken into shorter sequences called k-mers, typically less then 10 nucleotides in length, which can be tested for predicting a particular behavior, such as high abundance. If specific binding sites are not available, k-mer searches  together with highly efficient penalised linear regression methods, such as GLMNET (Friedmann et al., 2010), allow for the simultaneous search of large numbers of exact motifs with automatic removal of ineffective ones (Vijayabaskar et al., 2019). Exact match k-mer searches can then be further improved by introducing overlapping or redundant motifs and finally with experimental confirmation.

Here, we utilise modular cloning to construct reporters, consisting of different combinations of promoter-5'UTR and 3'UTR-terminator swaps to better understand the contribution of 3’UTRs on gene expression. We adapt a method used by Cheng et al. to computationally predict mRNA half-life in yeast, with linear combinations of sequence features. This was achieved by modelling the log2 half-life as a linear combination of codon usage and motif counts (Cheng et al. 2017). Using our model, we identify several putative stabilisation and decay motifs and subsequently probe their role on mRNA fate when inserted into different 3’UTRs or removed from 3’UTRs containing these motifs. We further explore the dependence of promoter selection on the relationship between these motifs and mRNA abundance. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs.
