# Main
 
## Gene expression regulation
The core promoter region contains the essential elements required to initiate transcription, such as the transcription initiation start site (TSS). The consensus TSS sequence is A(Arich)5NPyA(A/T)NN(Arich)6 with A representing the first transcribed base (Zhang and Dietrich, 2005). The presence of multiple TSS in the core promoter can result in the transcription of multiple unique isoforms of the same gene (Pelechano, Wei and Steinmetz, 2013). Upstream regulatory elements can modulate promoter activity through the recruitment of single transcription factors or synergistic combinations to facilitate recruitment of transcriptional machinery and ultimately enhance transcription rates. 

The terminator, located downstream of the stop codon, contribute to transcriptional termination and post-transcriptional regulation. The former involves the cleavage of the 3’mRNA and poly(A) addition and the latter determines the stability, translational efficiency and localisation of the mRNA, mediated by RNA-binding proteins. 

Terminators play several roles in gene expression, namely to terminate transcription and polyadenylate mRNA at the 3’-terminal end. Within the terminator are three functional elements, 1) the efficiency element, 2) positioning element and 3) the actual polyadenylation signal.  The efficiency element determines the efficiency of the 3’-end formation. The first evidence of the importance of this element came from the analysis of a 38 bp deletion of the CYC1 3’UTR (cyc1-512). This resulted in the destruction of normal 3’-end formation and resulting in a long cyc1-512 mRNA with many discrete 3’-ends ranging from the wild-type polyA site to sizes greater than 2000 bp downstream. The sequence TATATA has been found to have the greatest capacity. This is found in 3’-end forming signals in GAL7 and MRP2 genes. Similar sequences such as TATGTA, found in TRP1, ARO4 and TRP4, and TTTATA, found in CYC1, GCCN4, PHO5 and ADH1 also function as efficiency elements but are only fairly strong. The mutation of this sequence, in the case of PHO5, had no effect on 3’-end formation suggesting the presence of other efficiency elements. In some cases, several copies of the motif exists in the 3’UTR and the simultaneous deletion of both reduced 3’-end processing in vitro.  

The positioning element is almost always located downstream of the efficiency element and upstream of the polyadenylation signal, with at least one notable exception (Aranda et al., 1998). The positioning element serves to direct cleavage to a position ~20 bases downstream of the element. Two known positioning elements are AAGAA and TTAAGAAC, however most yeast mRNAs appear to lack the TTAAGAAC motif (Guo and Sherman, 1996). Deletion or the introduction of mutations too this site has resulted in a decrease on mRNA and protein abundance. ___ et al showed that the deletion of the positioning element, TTAAGAAC from the CYC1 terminator sequence resulted in the utilisation of an alternative poly(A) site downstream. In several cases, such as that of Gal7, the deletion of the positioning element can result in the unsuccessful 3’-end formation (Abe 1990). Russo carried out a saturation mutagenesis of AATAAA and found that the positioning element sequence was tolerant of certain base replacements, with the optimal sequences AATAAA and AAAAAA (Russo et al., 1993). The introduction of guanosine resides at the first position, however, rendered the positioning element ineffective. Finally the poly(A) site itself is redundant displaying a preferred Py(A)n sequence (Bennetzen and Hall, 1982; Heidmann et al., 1994). 

## 3’UTRs
3’UTRs regulate the mRNA fate by orchestrating its degradation, translation and localisation. These functions are mediated by the binding of RNA binding proteins (RBP) to the cis-elements on the 3’UTR which recruit effector proteins. Different RBPs can interact with a diverse range of effector proteins. The action of these effector proteins are dependent on the RBPs and their access to the binding sites on the 3’UTR. However accessibility of the RBPs to the 3’UTR can be influenced by binding of other RBPs, structural modifications due to secondary and tertiary structure formation or localisation to different parts of the cell. 

If half-life is determined by the 3’UTR, a longer half life should correlate with a higher protein expression. In a comprehensive analysis of the mRNA half-lives of 4987 yeast genes, four were found to have longer half-lives (~105 mins) (Wang et al., 2002). These four genes are involved in trehalose metabolism (TPS1, TPS2, TPS3 and TSL1). Yamanishi et al. took their 3’UTRs and cloned them downstream of a GFP reporter, driven by the TDH3 promoter (Yamanishi, Katahira and Matsuyama, 2011). These constructs were integrated into the PDC6 locus on the W303-1a strain. They found that the expression of GFP with the tTSL1, tTPS2 and tTPS3 terminators had comparable protein expression to the control (pTDH3-GFP-tPGK1) despite the higher mRNA half life of the genes for tTSL1, tTPS2 and tTPS3. In contrast, the GFP construct under the control of tTPS1 had a 20% higher protein output compared to the control. Yamanishi et al. expanded on this strategy and cloned 5302 terminators downstream of a GFP, driven by pPGK1 to investigate the contribution of the terminator on protein production. They found terminators (tRPL41B, tRPL15A, tDIT1, tRPL3, tIDP1) with up to 2.5x the activity of the cassette under the control of tPGK1 (Yamanishi et al., 2013). Ito et al. further analysed these five terminators in combination with different promoters, reporters and carbon sources and found that tDIT1 showed the highest activity under almost all conditions tested (Ito et al., 2013). The DIT1 gene is involved in the synthesis of a spore cell-wall component and its gene regulation is strictly regulated only at specific stages of sporulation. Ito et al. found that the enhancement of protein production by the tDIT1 terminator is mediated by two RNA-binding proteins, Nab6 and Pap1 (Ito et al., 2016). 
 
Through the comparison of half-lives of different isoforms, Giesberg et al. (2014) have identified features in the 3’UTR that contribute to an increased stability of the mRNA. Specifically, they have identified a stabilising motif consisting of a polyU tract which its hypothesised to form a secondary structure with the polyA tail, and thus stabilising the mRNA. They also experimentally validated the stabilising effect of a GC secondary structure in the 3’UTR of an mRNA 


    Computational biology of 3’UTRs and motifs (Computational approaches) 
Searching for novel cis-regulatory elements, such as 3’UTR motifs, associated with post-transcriptional mechanisms consists of filtering for mRNA transcripts suspected to be enriched with motifs and comparing these sequences to determine a common motif.  mRNA transcripts have previously been filtered by using functional transcriptomic data sets to select for those with high abundance or half lives (Shalgi et al., 2005, Cheng et al., 2017). Alternatively, enriched binding of RNA to proteins can be experimentally confirmed, using immunoprecipitation and UV cross linking, providing either whole transcripts to compare (Riordan, Herschlag and Brown, 2010) or specific intra-transcript binding regions, (Zhao et al., 2010, Granneman et al, 2009). Given transcripts/regions of interest, the deduction of actual consensus sequences is itself non-trivial.  Motifs can be of variable length and consist of highly redundant nucleotides separating conserved regions. Current motif finding packages, such as MEME-suite (Bailey et al., 2015), overcome these issues by first finding smaller seed sequences that regularly occur in the enriched transcripts/regions and then attempting to pairs these up into longer motifs (Hogan et al., 2008). Finally, recent developments in efficient algorithms for fitting linear models also allow for the detection of  novel cis-regulatory elements. A model predicting the results of a functional genomics data set, i.e. genome-wide transcript abundance, using all possible contributing elements can be filtered down by penalised maximum likelihood to only those elements with significant contributions (Vijayabaskar et al., 2019).
 

    3’UTR screens (Experimental approaches) 

 

 

## Brief overview of the paper
Here, we utilise modular cloning to construct chimeras to investigate the contribution of 3’UTRs on gene expression. Following a similar approach carried out by Cheng et al. (2017), we computationally predict the mRNA half-life in yeast with linear combinations of sequence features. We modelled the log2 half-life as a linear combination of codon usage and motif counts following Cheng et al. (2017). Using the model, we have isolated several stabilising and decay motifs and probed their role on mRNA fate using a modular cloning assembly method. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs. 