# Introduction

Our understanding of gene expression relies on the abstraction of a gene into discrete cis-regulatory element (CRE) modules, or ‘parts’, based on their most prominent function.
The promoter coordinates the initiation of transcription and the rate at which this occurs on DNA.
Transcription then progresses through the 5' untranslated region (5'UTR), coding sequence (CDS), and 3' UTR, before termination and polyadenylation of the nascent transcript is coordinated by the terminator sequence.
Post-transcriptional regulation includes effects of the 5'UTR on translation initiation, the coding sequence on translation elongation and mRNA decay, and of 3'UTR on localization and decay.
Modern DNA synthesis and cloning techniques make gene construction practically modular: a number of standardised and modularised parts libraries have been designed to allow combinatorial assembly of parts and subsequent construction of complex pathways (Lee et al. 2015, Garcia-Ruiz et al. 2018, Andreou et al. 2018).
These parts libraries generally approximate promoter to include the 5'UTR, and terminator to include the 3'UTR (Lee et al. 2015, Guo et al. 2015, Lubliner et al. 2013) [@Yamanishi2013].
They also make the simplifying assumption that the parts are functionally modular, or composable: the contribution of each part to gene expression is predictable when combined with other parts.
However, nearby parts do interact: in *E. coli* the protein expression driven by combinations of promoter and ribosomal binding site is predictable within 2-fold only 2/3 of the time (Kosuri et al., 2013).

How much do cis-regulatory elements on either end of the coding sequence behave as functional modules?
More precisely, do eukaryotic modular parts libraries behave predictably according to simple linear models? 
In budding yeast, promoter identity can change the relative contribution of terminator up to 11-fold [@Curran2013] (Wei et al. 2017).
Short CREs within 3' UTRs also affect protein output via recruitment of RNA-binding proteins (Singh 2015) including through shaping transcription termination (Shalem 2013),  post-transcriptional localization and decay (Puf protein cite), and other mechanisms (Ito et al., 2016).
However, it is unclear to what extent the function of CREs in 3'UTRs is consistent across distinct promoters.

Experimental and computational methods to identify CREs within genes generally operate under an assumption of independence.
Lists of suspected cis-regulatory elements have been collated by searching for common sequence motifs in genes grouped by similar characteristics, such as transcript half lives, [@Shalgi2005; @Cheng2017] or protein binding (Hogan 2008; Kretz et al. 2013, Siprashvili et al. 2016).
Alternatively, direct evidence for protein binding sites on transcripts can be obtained by cross-linking assays (Zhao et al., 2010; cite a recent Jernej Ule review) [@Granneman2009].
Computational packages such as MEME-suite [@Bailey2015] and BAMM motif (Siebert and Söding 2016) excel at determining short consensus sequences with some redundant positions, but do not support interacting effects between motifs or regions.
Alternatively, gene regions can be broken into sequences called k-mers, typically less then 10 nucleotides in length, which are then tested for regulatory effects. 
Restricting searches to k-mers allows for the detection of large numbers of exact motifs with model selection algorithms, such as penalised linear regression [@Friedman2010]. 
Many k-mer searches consider the effect of motifs in isolation of  overall context [@Vijayabaskar2019], although some methods use latent variables to label motif instances as active or inactive (Dai et al., 2017).
CRE context matters and inactive motifs are real: in mice, most high-affinity RBP motifs in expressed RNAs are not bound *in vivo* (Taliaferro et al., 2016).

Here, we investigate the contribution of 3'UTR cis-regulation to gene expression using reporter genes combined with computational analysis. 
We use modular cloning to construct 120 combinations of promoter-5'UTR, coding sequence, and 3'UTR-terminator modules, showing that the quantitative effect of terminators depends on both promoter adn coding sequence.
We then investigate the effect of motifs in 3'UTRs by adapting a method to computationally predict mRNA log2-half-life as a linear combination of codon usage and motif counts [@Cheng2017].
Having selected putative stabilisation and decay motifs, we quantify their effect on mRNA and protein output when inserted in, or removed from, distinct 3'UTRs in reporter genes.
We further explore the dependence of promoter selection on the relationship between these motifs and mRNA abundance.
Our results demonstrate that the incorporation of candidate decay motifs indeed results in a decrease in RNA abundance, however their quantitative contribution depends both on the 3'UTR and on the promoter.
