---
output:
  pdf_document: default
  html_document: default
---
# Introduction
Saccharomyces cerevisiae is a well established host platform for industrial production due to its traits as a robust organism under fermentation conditions, the wealth of molecular biology tools developed for its genetic manipulation and its status as a GRAS organism. Over the past two decades, it has been optimised for the production of antibiotics, painkillers, antioxidants and anti-cancer drugs (Siewers et al 2009, Gidijala et al. 2009, Yan et al. 2005, Koopman et al. 2012, Rodriguez et al. 2017). This process for its pathway engineering is reliant on precise capabilities in modulating biosynthetic enzyme gene expression. Increasing number of modular toolkits containing standardised genetic parts for the rapid and combinatorial assembly of genes have expedited the design-build-test-learn cycle (REF). 

Coding and non-coding regulatory elements, comprising of promoters, untranslated regions (5’UTR and 3’UTR) and terminators, possess a large amount of information for the regulation of mRNA and protein levels. Perhaps the best studied is the role of the promoter, which plays a central role in coordinating the rate of new transcript production (Ottoz et al. 2018). TFBS upstream of the core promoter region modulate promoter activity through the recruitment of single transcription factors or synergistic combinations to facilitate recruitment of transcriptional machinery and ultimately enhance transcription rates (Levo 2017). On the 3’-end of the CDS, the terminator coordinates transcriptional termination and polyadenylation of the mRNA at the 3’-terminal end. 

Much of our current knowledge of the regulations of gene expression is based on high-throughput characterisation of promoter and terminator libraries. [Add Promoter characterisation section]. A comprehensive analysis of the yeast ‘terminatome’ (5302 terminators) by Yamanishi et al. showed that the regulation of gene expression spanning an 81-fold range. Analysis of the transcript and protein abundance in a small subset of terminator constructs showed that the regulation of gene expression of the terminator is in the transcript level, suggesting that they contain cis-elements associated with an increase in mRNA decay (Yamanishi 2013).

Analysis of the relationship between gene regulatory elements and their cooperative regulation of expression have shown varying levels of regulation mediated by the terminator when coupled with different promoters (Yamanishi et al. 2011, Yamanishi et al. 2013, Lee et al. 2015, Curran et al. 2013). Important questions emerge around the independence of promoter and 3’UTR/terminator choice in relation to transcript abundance. If the promoter sequence and 3’UTR/terminator sequence do not independently impact transcript abundance then there is the potential for unexpected interactions and unpredictable transcript abundance. 

Flanking the coding sequence on the mRNA are untranslated regions (UTR) which have been shown to play important roles in coordinating mRNA stability, translation and localisation (Mayr 2016). These functions are largely mediated by the association of RNA binding proteins (RBP) to the corresponding cis-regulatory motif on the UTR and subsequent recruitment effector proteins. These RNA-protein interactions are dynamic, and rapid restructuring can occur in response to metabolic or environmental stimuli. The accessibility of RBPs to these cis-regulatory elements are also influenced by structural modifications or localisation to different compartments of the cell. Formation of secondary or tertiary structures have been suggested to occlude RBPs from binding to cis-regulatory elements in the 3’UTR. RBPs have also been found to bind many mRNA targets however their binding specificity is increased through their association with protein interaction partners. The cooperativity between multiple RBP binding to adjacent cis-regulatory motifs in an mRNA have been shown to improve affinity and specificity of She2p and She3p RBP blinding to ASH1 mRNA compared to their individual interactions (Shen et al.  2009) (Niedner et al. 2014). Therefore the changes to the mRNA fate is determined by the available binding sites on the UTR for RBPs at a given moment. 

The identification and dissection of these RNA-protein interactions networks have relied on in vitro experimental methods that allow the enrichment of proteins associated with RNA of interest. Biotinylation of commercially synthesised or in vitro transcribed RNA enable its immobilisation on streptavidin beads, and subsequent incubation of the RNA with cell extract can allow the identification of RBPs with affinity to the bound RNA (Hartmuth et al. 2004). Other in vitro strategies centre around immobilisation of candidate RBPs. Protein microarrays can be employed to screen for relative binding strength for RNAs of interest labelled with Cy5 dye (Kretz et al. 2013, Siprashvili et al. 2016). Whilst the mutagenesis of RNA and putative RBP partners in these in vitro methods enable much easier  exploration of RNA-RBP interactions, they however, do not take into consideration the cellular context or account for chemical modifications of RNA that can occur post-transcription. 

In contrast, in vivo strategies centre around cross-linking RNA and RBPs in living cells using UV or Formaldehyde to form covalent bonds between RNA and proteins that are in direct contact with each other. Formaldehyde based cross-linking has the advantage of much higher efficiency of cross-linking between RNA and RNA binding proteins, however a key caveat is that protein-protein cross-linking also occurs, meaning that it can be more challenging to discern between RBPs and proteins that interact with RBPs. In contrast UV cross-linking approaches are much more specific, however not only is cross-linking less efficient, but further biases have been observed for cross-linking with uridines (Sugimoto et al. 2012) and varying efficiencies are seen across different AAs (Meisenheimer et al. 1997).

Enriched binding of RNA to proteins experimentally validated using methods previously described provide either whole transcripts (Riordan, Herschlag and Brown, 2010) or specific intra-transcript binding regions (Zhao et al., 2010, Granneman et al, 2009) for comparison to determine a common motif. Alternatively, searching for novel cis-regulatory elements, such as 3’UTR motifs, associated with post-transcriptional mechanisms consists of filtering for mRNA transcripts suspected to be enriched with motifs and comparing these sequences to determine a common motif.  mRNA transcripts have previously been filtered by using functional transcriptomic data sets to select for transcripts with high abundance or half lives (Shalgi et al., 2005, Cheng et al., 2017). Motifs can be of variable length and consist of highly redundant nucleotides separating conserved regions. Current motif finding packages, such as MEME-suite (Bailey et al., 2015), overcome these issues by first finding smaller seed sequences that regularly occur in the enriched transcripts/regions and then attempting to pairs these up into longer motifs (Hogan et al., 2008). Finally, recent developments in efficient algorithms for fitting linear models also allow for the detection of  novel cis-regulatory elements. A model predicting the results of a functional genomics data set, i.e. genome-wide transcript abundance, using all possible contributing elements can be filtered down by penalised maximum likelihood to only those elements with significant contributions (Vijayabaskar et al., 2019).

Here, we utilise modular cloning to construct chimeras to investigate the contribution of 3’UTRs on gene expression. Following a similar approach carried out by Cheng et al. (2017), we computationally predict the mRNA half-life in yeast with linear combinations of sequence features. We modelled the log2 half-life as a linear combination of codon usage and motif counts following Cheng et al. (2017). Using the model, we have isolated several stabilising and decay motifs and probed their role on mRNA fate using a modular cloning assembly method. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs. 



