# Introduction
Over the past two decades, the development of molecular biology toolkits for the genetic manipulation of host platforms, such as Saccharomyces cerevisiae, have established the bioproduction of antibiotics, painkillers, antioxidants and anti-cancer drugs in industry (Siewers et al 2009, Gidijala et al. 2009, Yan et al. 2005, Koopman et al. 2012, Rodriguez et al. 2017). These toolkits often utilise native yeast promoter/5'UTR and 3'UTR/terminator regions to provide standardised parts to enable modular assembly of genes for pathway expression. Industrial demands for efficient production has driven the search for regulatory regions that maximise expression, often focusing on the promoter region. Recent studies have shown that terminator selection alone can increase fluorescent protein production by 50% (Yamanishi et al. 2011) and that the regulatory effects of the promoter regions can be dependent on the terminator. 

Several groups have sought to elucidate the effect of different native yeast terminators with a goal of maximal protein expression. Yamanishi et al. identified that the effect of a library of 5302 terminators spanned an 81-fold range difference in fluorescence production, with several resulting in a doubling of protein production relative to tPGK1 (Yamanishi et al. 2013). When coupled with different promoters, the relative difference in protein output between terminators vary, with a maximum difference of 11-fold when coupled with a low expression promoter (Curran et al. 2013, Wei et al. 2017). Ito et al. have been able to associate expression changes directly to terminator elements by identifing a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). There is growing evidence for interactions between the regulatory effects of promoter and terminator regions but how this translates onto individual and combinations of cis-regulatory sequences within these regions remain unknown.

The identification of functional elements within 3'UTRs that control gene expression have been previously conducted either by grouping similar genes using abundance or half life transcriptomic data sets and then searching for common motifs (Shalgi et al., 2005, Cheng et al., 2017), or by experimentally verifying protein binding sites with cross-linking assays (Zhao et al., 2010, Granneman et al, 2009) or with sequencing mRNA bound to immobilised proteins and then building consensus sequences (Kretz et al. 2013, Siprashvili et al. 2016). Whether specific intra-transcript binding regions or whole transcripts of interest are available, the task of motif discovery is non-trivial as they can be of variable length and consist of highly redundant nucleotides separating conserved regions. Packages like the MEME-suite (Bailey et al., 2015), excel at determining short consensus sequences by finding the likelihood of each RNA base appearing at each position using the set of suspected binding sites, which automatically allows redundancy. Alternatively, the transcripts of interest can be broken into shorter sequences called k-mers, typically less then 10 nucleotides in length, which can be tested for predicting a particular behavior, such as high abundance. If specific binding sites are not available, k-mer searches  together with highly efficient penalised linear regression methods, such as GLMNET (Friedmann et al., 2010), allow for the simultaneous search of large numbers of exact motifs with automatic removal of ineffective ones (Vijayabaskar et al., 2019). Exact match k-mer searches can then be further improved by introducing overlapping or redundant motifs and finally with experimental confirmation.

Here, we utilise modular cloning to construct reporters, consisting of different combinations of promoter-5'UTR and 3'UTR-terminator swaps to better understand the contribution of 3’UTRs on gene expression. We adapt a method used by Cheng et al. to computationally predict mRNA half-life in yeast, with linear combinations of sequence features. This was achieved by modelling the log2 half-life as a linear combination of codon usage and motif counts (Cheng et al. 2017). Using our model, we identify several putative stabilisation and decay motifs and subsequently probe their role on mRNA fate when inserted into different 3’UTRs or removed from 3’UTRs containing these motifs. We further explore the dependence of promoter selection on the relationship between these motifs and mRNA abundance. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs.
