# Introduction

## Introduction to gene expression regulation 
Gene expression is determined by a number of genetic elements. Perhaps the best studied is the role of the promoter, which plays a central role in coordinating the rate of new transcript production (Ottoz et al. 2018). The promoter encompasses two major regions, the core promoter and upstream regulatory sequences (URS). Within the core promoter region are essential elements required to initiate transcription, such as the TATA box and the transcription initiation start site (TSS), as well as sequence determinants at defined locations within the core promoter that result a wide spectrum of activities (Lubliner et al. 2013, Lubliner 2015). Another layer of regulation is through transcription factor binding sites (TFBS) present upstream of the core promoter. TFBS modulate promoter activity through the recruitment of single transcription factors or synergistic combinations to facilitate recruitment of transcriptional machinery and ultimately enhance transcription rates. However, these transcription factors are susceptible to chromatin context, which determines accessibility to their cognate and adjacent TFBS (Levo 2017). These interactions between different TFBS, chromatin conformation and chromatin modulation capacity of transcription factors can result in changes in the expression program. The joint regulatory mechanism mediated by these two regions have been explored by generating random combinations of URS and core promoters and measuring their effect on gene expression. In line with previous findings, Dhillon et al. (2019) demonstrated that different core promoters increases URS mediated combinatorial control of gene expression. 

In contrast to the promoter, the terminator coordinates transcriptional termination and polyadenylation of the mRNA at the 3’-terminal end. Within the terminator are three functional elements, 1) the efficiency element, 2) positioning element and 3) the actual polyadenylation signal (Guo et al. 1996). The efficiency element determines the efficiency of the 3’-end formation and the positioning element serves to direct cleavage to a position ~20 bases downstream of the element. 

The effect of terminator choice has been studied in the context of a single promoter, showing mild effects on gene expression. In the characterisation of over 3000 terminators, regulating the expression of a fluorescent protein (GFP) driven by a the pTDH3 promoter, Yamanishi et al (2013) demonstrated that the choice of terminator can result in a 2.3-fold increase relative to the activity of the control (tPGK1). ‘Expression-enhancing’ terminators were identified to increases gene expression when in conjunction with strong promoters. One example is the tTPS1 which resulted in a 20% higher protein output due to improved mRNA stability (Yamanishi et al. 2011). Similarly, Ito et al. identified tDIT1 was associated with a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). Lee et al (2015) observed a maximum of 3.6-fold change when comparing the effect 6 terminators (tENO2, tSSA1, tADH1, tPGK1, tENO2, tTDH1) on gene expression for a given promoter (three tested - pTDH3, pTEF1, pRPL18B). However, when coupled with low-expressing promoters, Curran et al (2013) demonstrated a maximum difference of 11-fold difference compared to the control (tCYC1). 

## Introduction to 3'UTRs
Encoded within the terminator sequence, the 3’ untranslated region (UTR) has been shown to play an important role in coordinating transcript stability and localisation (Mayr 2016). These functions are largely mediated by the association of RNA binding proteins (RBP) to the corresponding cis-regulatory motif on the 3’UTR and subsequent recruitment effector proteins. Therefore the changes to the mRNA fate is determined by the available binding sites on the 3’UTR for RBPs at a given moment. 

A number of in vitro strategies have been developed, with various approaches employed for the enrichment of proteins associated with RNA of interest. Biotinylation of synthesised RNA (commercially synthesised or In-vitro transcribed) enables immobilisation of the RNA on streptavidin beads. Subsequent incubation with cell extracts followed by wash steps and finally boiling allows for enrichment of proteins associated with the RNA of interest (Hartmuth et al. 2004). Comparable aptamer-based approaches rely on the tagging of RNA with an S1 aptamer. In this case the RNA can again be immobilised to streptavidin beads, as the S1 aptamer has strong affinity for streptavidin (Leppek et al. 2014). This binding can then be reversed through the addition of biotin.

An alternative approach relies on tagging the RNA of interest with a Csy4 hairpin. This RNA can then be bound by a catalytically inactivate, biotinylated Csy4, which in turn can bind avidin beads. A key difference in this strategy was that the authors showed that through the addition of imidazole they could enable Csy4 catalysis of RNA cleavage, which enabled release of the tagged RNAs along with associated bound protein (Lee et al. 2013). Finally divergent in-vitro strategies centre around immobilisation of candidate RBPs. Protein microarrays can be employed to screen for relative binding strength for RNAs of interest labelled with Cy5 dye (Kretz et al. 2013, Siprashvili et al. 2016).

Whilst in vitro strategies do enable much easier exploration of mutagenesis for RNA and putative protein partners, these strategies can strip out several layers of information that can be applied within the cell. In particular all of these strategies do not currently account for chemical modifications of RNA that can occur post-transcription.

A number of in-vivo approaches have also been developed, with many of these strategies centre around cross-linking RNA and RBPs using UV or Formaldehyde. Formaldehyde based cross-linking has the advantage of much higher efficiency of cross-linking between RNA and RNA binding proteins, however a key caveat is that protein-protein cross-linking also occurs, meaning that it can be more challenging to discern between RBPs and proteins that interact with RBPs. In contrast UV cross-linking approaches are much more specific, however not only is cross-linking less efficient, but further biases have been observed for cross-linking with uridines (Sugimoto et al. 2012) and varying efficiencies are seen across different AAs (Meisenheimer et al. 1997).

Alternatively a newer strategy RaPID, has shown that through the incorporation of BoxB aptamers, a -N aptamer binding protein fused to the BirA biotin ligase is recruited to the mRNA. BirA subsequently biotinylates proximal proteins, which can then be immobilised on streptavidin beads (Ramanathan et al. 2018). Whilst this strategy is in some ways much less invasive than cross-linking strategies and relies upon much fewer cells, the modifications to the through incorporation of aptamer sequences may impact the RNA structure. Further the limited range of the biotin ligase (<62 nt) means that this strategy is currently only suitable for characterising shorter stretches of RNA.

Searching for novel cis-regulatory elements, such as 3’UTR motifs, associated with post-transcriptional mechanisms consists of filtering for mRNA transcripts suspected to be enriched with motifs and comparing these sequences to determine a common motif.  mRNA transcripts have previously been filtered by using functional transcriptomic data sets to select for transcripts with high abundance or half lives (Shalgi et al., 2005, Cheng et al., 2017). Alternatively, enriched binding of RNA to proteins can be experimentally confirmed, using immunoprecipitation and UV cross linking, providing either whole transcripts to compare (Riordan, Herschlag and Brown, 2010) or specific intra-transcript binding regions, (Zhao et al., 2010, Granneman et al, 2009). Given transcripts/regions of interest, the deduction of actual consensus sequences is itself non-trivial.  Motifs can be of variable length and consist of highly redundant nucleotides separating conserved regions. Current motif finding packages, such as MEME-suite (Bailey et al., 2015), overcome these issues by first finding smaller seed sequences that regularly occur in the enriched transcripts/regions and then attempting to pairs these up into longer motifs (Hogan et al., 2008). Finally, recent developments in efficient algorithms for fitting linear models also allow for the detection of  novel cis-regulatory elements. A model predicting the results of a functional genomics data set, i.e. genome-wide transcript abundance, using all possible contributing elements can be filtered down by penalised maximum likelihood to only those elements with significant contributions (Vijayabaskar et al., 2019).

The portability of these cis-regulatory motifs have been tested by introducing a putative destabilisation motif ‘ATATTC’ into different 3’UTRs, showing a decrease in mRNA stability for both 3’UTRs tested. However, the magnitude of changes in mRNA destabilisation with the presence of the destabilising motif varied between 3’UTRs, suggesting the interplay of other factors in determining mRNA fate (Cheng et al. 2017). Therefore, another important question needs to be addressed - Are motifs in relation to mRNA abundance independent of 3’UTR context?

Another question worth exploring is the composition and grammar of sequence elements of 3’UTRs that contribute to an improved stability of an mRNA. With improved sequencing methods and analysis of large data sets, much work has been carried out over the past decade to mine for cis-regulatory elements in yeast 3’UTRs associated with different mRNA fate (localisation, stability or decay).

Several RBPs have also been found to have dual roles, functioning as metabolic enzymes and moonlighting as regulators of mRNA (Sherrer et al. 2010). This suggests a direct connection via a feedback loop between the yeast metabolism and post-transcriptional regulation of related genes (Hogan et al. 2008, Tsvetanova et al. 2010, Beckmann et al. 2016).These RNA-protein interactions are dynamic, and rapid restructuring can occur in response to metabolic or environmental stimuli. The accessibility of RBPs to these cis-regulatory elements are also influenced by structural modifications or localisation to different compartments of the cell. Formation of secondary or tertiary structures have been suggested to occlude RBPs from binding to cis-regulatory elements in the 3’UTR. The formation of a  stable stem loop within the 3’UTR has also shown to have a stabilising effect (Giesberg et al. 2014). 

RBPs have been found to bind many mRNA targets however their binding specificity is increased through their association with protein interaction partners. The cooperativity between multiple RBP binding to adjacent cis-regulatory motifs in an mRNA have been shown to improve affinity and specificity of She2p and She3p RBP blinding to ASH1 mRNA compared to their individual interactions (Shen et al.  2009) (Niedner et al. 2014). This joint binding of both RBPs along with Myo4p are necessary for the localisation of the mRNA-protein complex to the bud tip. Adjacent cis-regulatory motifs on a 3’UTR or motifs brought into close proximity due to secondary structures can also have a competitive interaction, with preferential binding of one RBP through improved affinity mediated under specific conditions (Dassi 2017). 

Another factor that influences mRNA abundance post-transcriptionally is the choice of promoter used to drive the expression of a coding sequence. In particular the trans-acting factor, Rap1 and the presence of the Rap1p-binding sites on the UAS have been proposed to alter the messenger ribonucleoprotein (mRNP) complex composition prior to its export to the cytoplasm, and subsequently lead to an increase in mRNA transcript turnover (Bregman et al. 2011).

## Overview of the results

Here, we utilise modular cloning to construct chimeras to investigate the contribution of 3’UTRs on gene expression. Following a similar approach carried out by Cheng et al. (2017), we computationally predict the mRNA half-life in yeast with linear combinations of sequence features. We modelled the log2 half-life as a linear combination of codon usage and motif counts following Cheng et al. (2017). Using the model, we have isolated several stabilising and decay motifs and probed their role on mRNA fate using a modular cloning assembly method. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a decrease in RNA abundance. Furthermore, decreases in mRNA abundance are proportional to the number of incorporated decay motifs. 
