# Introduction

Our understanding of gene expression relies on the abstraction of a gene into discrete cis-regulatory elements (CRE), or parts, based on their most prominent function. The promoter coordinates the initiation of transcription. Transcription then progresses until it reaches the terminator, where cleavage and (in eukaryotes) polyadenylation of the nascent RNA transcript is coordinated. Extending the abstraction to the mature mRNA transcript, the coding sequence is flanked by untranslated regions (UTRs) at its 5’ and 3’ ends. The 5’UTR primarily contributes to translation regulation [@Dai2015] and the 3’UTR largely contributes to transcript stability and localisation [@Naseri2020]. Important exceptions do exist with cis-regulatory elements contributing to translation and stability across all parts of an mRNA transcript [@Hogan2008]. However, this abstraction of specific characteristics to individual parts of a gene has facilitated the development of genetic toolkits that are based on the modular assembly of interchangeable parts. Several groups have constructed modular libraries comprising promoter (including 5’UTR) and terminator (including 3’UTR) parts [@Curran2013; @Guo2015; @Lee2015; @Garcia2018; @Andreou2018], which have not only facilitated the characterisation of individual CREs [@Curran2013; @Guo2015; @Lee2015; @Andreou2018;@Sharon2012] but also enabled optimisation of downstream compound production in yeast cell factories. 

Modular assembly however, blurs concepts of modularity: modular assembly, how easy it is to construct combinations of parts; and composability, how function and regulation can be decomposed into contributions from individual parts. When using these parts either for tailoring gene expression or the characterisation of the regulatory parts within a library, understanding the relationship and dependence of the selected parts is essential as the assembly of parts whose functions are not composable will result in constructs with unpredictable functions or regulatory behaviours.

Functional characterisation of modular libraries of promoters and terminators tend to focus on one kind of part at a time, for example testing many promoters with the same coding sequence and terminator [@Guo2015] or conversely a library of terminators with a single promoter [@Shalem2015]. Similarly, large libraries of short CREs or short sequence motifs within promoters have been characterised in combination with a single terminator [@Sharon2012]. Further, when exploring the design of novel synthetic terminators, involving the insertion of sequence elements into the terminator, these have been characterised within the context of a single promoter [@Felippes2022]. Computational methods for the discovery and quantification of sequence motifs recognised by regulatory proteins, also tend to rely on models of independent effects, that is, composability. Prominent methods for mapping CRE sequence-function relationships include predicting functional data with short sequence features, often using linear models [@Shalgi2005; @Cheng2017;@Vijayabaskar2019]; or, directly comparing the sequences of genes with similar characteristics to determine the presence of short consensus sequence motifs [@Hogan2008; @Kretz2013] using motif discovery software [@Elemento2007, @Bailey2015]. Both of these approaches make the implicit approximation that the contribution of a short CRE is independent of context, so that the effect of combining motifs is composed of a linear sum (on the appropriate scale) of the individual CRE contributions. The approximation that short CREs act independently helps to find elements that have clear contributions, and to simplify a vast search space that would be made exponentially larger by accounting for CRE interactions. However, framing the search for CREs around independent contributions from short motifs overlooks multi-part motifs, interactions between motifs, and that motifs may be active only in specific contexts.
\enlargethispage{-65.1pt}

The interaction of these modular CREs in combination with other CREs, and their impact on gene expression remain under explored. In yeast, Lee et al screened a subset of their promoters and terminators, and saw a 3.6-fold difference in protein production, however they did not test for dependence between the regulatory elements [@Lee2015]. Further, a recent study in budding yeast found strong pairwise interactions between the enhancer and core promoter regions [@Dhillon2020].

The importance of interactions among CREs has been clear since the discovery of cooperative DNA binding by transcription factors [@Ptashne1980].
For example, whether post-transcriptional regulatory CREs are functional or not depends on the RNA sequence context [@Taliaferro2016].
One large-scale study of synthetic combinations of bacterial promoters and ribosomal binding sites showed that expression was largely composed of independent contributions; 64% of constructs had protein expression levels with less than a 2-fold deviation from the predictions of a linear model without interactions [@Kosuri2013]. 
However, 5% of the constructs had, on average, a 13-fold deviation.
A recent study in budding yeast also found strong pairwise interactions between different promoter regions [@Dhillon2020].
Experiments with promoter-terminator swaps in budding yeast revealed 2-fold changes due to promoter-terminator interactions [@Ito2013, @Lee2015], even though these CREs are separated by hundreds of nucleotides of coding sequence. 

Here, we are looking to identify dependence between the CDS, cis-regulatory elements (promoters and terminators) and 3’UTR motifs, on gene expression and further explore how this changes across different growth phases. Modular cloning is used to create 120 combinations of promoter (including 5’UTR), coding sequence, and terminator (including 3’UTR) modules and the RNA and protein levels are measured. In these constructs, we show that terminator effects on RNA and protein levels depend on both promoter and coding sequence. We further explore the effect of promoter and terminator context on the regulatory behaviour of short CREs in 3’UTRs. In order to select suitable 3’UTR motifs, we adapt a method to computationally predict mRNA log2-half-life as a linear combination of codon usage and motif counts [@Cheng2017]. We then quantify the effect on mRNA and protein levels of selected 3’UTR motifs when inserted in, or removed from, diverse terminators. Our results demonstrate that the incorporation of candidate decay motifs indeed results in a decrease in mRNA abundance. However, the quantitative contribution from motifs depends on both the local 3’UTR context and the selected promoter. These results demonstrate the limits to the composability of cis-regulatory elements, and sequence motifs within them, in modular toolkits. It further highlights the consideration that needs to be taken into account when using the libraries to tailor gene expression or characterising elements within a library.