# Results

```{r}
library(here)
library(tidyr)
knitr::opts_chunk$set(fig.pos = "p", dpi = 300)
```

## Terminator effects on gene expression depend on cis-regulatory context

To investigate the context dependence of cis-regulatory regions, we created a library of 120 constructs (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)A), containing all combinations of 6 promoters (including 5'UTR), 2 coding sequences, and 10 terminators (including 3'UTR).
We selected promoters and terminators from native yeast genes spanning a variety of different expression patterns and functions (Table \ref{tab:terminator-summary-table}).
To choose specific sequence lengths for the terminators of our constructs, we referred to published measurements of median 3'UTR length [@Pelechano2013] because sequences that are necessary and sufficient for efficient transcriptional termination are found upstream of the termination site [@Guo1996c]. 
For terminators with measured length under 200nt, we used the standardised parts length of 200nt from the YeastFab library.
The 2 coding sequences (CDS) expressed mCherry or mTurquoise2 fluorescent proteins, which are bright fluorophores with only 30% amino acid identity [@Shaner2004; @Goedhart2012].


```{r pro-ter-platereader-mCherry-mTurq-raw, fig.cap="\\textbf{Both terminator and promoter contribute to gene expression.}  (\\textbf{A}) Design of chimeric reporter plasmids with all combinations of 6 promoters, 2 fluorescent proteins, and 10 terminators. (\\textbf{B}) Protein abundance estimated by mCherry and mTurquoise2 (mTurq) fluorescence measured at the time of each sample's maximum growth rate for 10 terminators paired with 4 high expressing promoters. Each diamond represents a biological replicate, averaged over 3 technical replicates.  The vertical line is the mean of all 6 biological replicates. (\\textbf{C}) Same as panel B, but for 2 low expressing promoters. Negative fluorescence values arising from instrument noise dominating measurements of constructs with negligible fluorescence are automatically set to 0. (\\textbf{D}) mCherry fluorescence correlates with RT-qPCR mRNA abundance for 3 promoters paired with all 10 terminators. Note that both axes use a $log_2$ scale.", fig.align='center', out.width="82%"}
# NOTE: Currently this figure is included at 85% size so we can see the caption
# Later we will probably want to display the caption on a separate page
knitr::include_graphics(here("results_chapter/figures/pro_ter_swap_protein_and_rna_exp_figure.png"))
```


(ref:tab-1-caption) \textbf{Summary of the terminator library.} The common gene name from which the terminator is extracted from is included alongside its systematic name. The median 3'UTR reported by [@Pelechano2013] is a median over the lengths of each distinct isoform they detect. The usage of each motif is signified by S, for promoter-terminator swaps, or M, for motif insertion or deletion. A short summary of the protein function is included.

```{r terminator-summary-table}
readr::read_csv(here("results_chapter/data/terminator_summary_table.csv")) %>%
  knitr::kable(format = "latex", align = "c", caption = "(ref:tab-1-caption)") %>%
  kableExtra::column_spec(2:3, width = "1.8cm") %>%
  kableExtra::column_spec(4, width = "2cm") %>%
  kableExtra::column_spec(5, width = "1.4cm") %>%
  kableExtra::column_spec(6, width = "2.3cm") %>%
  kableExtra::column_spec(1, width = "1.4cm") %>%
  kableExtra::row_spec(0, bold = TRUE)
```

Measuring fluorescence with a plate reader showed that, as expected, promoter choice dominated overall protein output.
We observed 100-fold changes in fluorescence between the 4 highest expressing promoters (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)B) and the 2 lowest expressing promoters (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)C) across both coding sequences.
Expression from the stress-induced pHSP26 was notably more variable than from pPGK1 and pRPS's, across 6 biological replicates, when combined with both coding sequences and a variety of terminators. 
We also confirmed that most differences in protein outputs are accounted for by changes in mRNA abundance by checking a subset of mCherry constructs using RT-qPCR ($R = 0.888$, Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)D).

Terminators also affect protein output with 5-fold changes in fluorescence seen within the same promoter-CDS sets, relative to the tPGK1 terminator of each group (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)A).
We focus on constructs with high-expression promoters due to the poor signal to noise ratio at low expression levels.
The interaction of coding sequence and terminator is seen most clearly for tPAB1.
tPAB1 is consistently the most highly expressed terminator in mTurq constructs, but is more variable in mCherry constructs.
Meanwhile, tPGK1 highlights the interactions of promoter and terminator.
tPGK1 is one of the most highly expressing terminators when paired with pPGK1 and pHSP26, but is up to 40% lower in expression when paired with pRPS3 or pRPS13.
Overall, our results show that the contributions of terminators (including 3'UTRs) to gene expression depend on other parts within the gene.

We further investigated the effects of terminators on mRNA levels by comparing constructs with extended vs truncated terminators. 
It is known that disrupting the transcription termination signal lowers expression [@Guo1996c; @Shalem2015].
However, standardised parts libraries often assume a fixed terminator length for all genes, which is likely to omit the termination signal for genes with longer terminators.
In the case of the YeastFab parts library [@Guo2015] the fixed terminator length is 200bp.
We compared a gene with a median 3'UTR length less than 200bp, RPS3 to a gene with a median 3'UTR length greater than 200bp, SRO9.
The median length of the native RPS3 3'UTR is 86nt [@Pelechano2013]; truncating the terminator to 86bp or 59bp reduces protein output by almost 2-fold (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)B).
The median length of the native SRO9 3'UTR is 543nt [@Pelechano2013]; extending the terminator length to 543bp increases protein output by almost 2-fold (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)B).
This validates our assay's ability to detect known regulatory signals affecting transcription termination, while highlighting the importance of using well-informed annotations to construct parts libraries for synthetic biology.
Note that we used the longer 543bp SRO9 terminator, and a similarly extended 421bp PMA1 terminator, for the main set of constructs (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)).

```{r pro-ter-platereader-mCherry-mTurq-norm, fig.cap="\\textbf{Terminator contributions to gene expression are promoter, coding sequence and length dependent.} (\\textbf{A}) Relative protein abundance from each terminator, normalized to a reference terminator tPGK1 for each matched promoter. The plot shows statistical summaries (mean and standard error) of 6 replicates for high-expression promoter data shown in Figure 1A. (\\textbf{B}) RT-qPCR mRNA results targeting the mCherry ORF for pRPS3-mCherry-tRPS3 and pSRO9-mCherry-tSRO9 constructs with differing terminator lengths.", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/pro_ter_platereader_norm_mTurq_and_mCh.png"))
```

## Candidate cis-regulatory elements contribute to transcript decay rates

Next, we investigated how the regulatory effects of CREs contained within terminator regions depend on their context.
First, 69 suitable CREs to test for context dependence were found through a literature search. 
All were suspected sequence motifs for mRNA binding proteins, several directly associated with proteins involved in mRNA degradation [@Hogan2008; @Shalgi2005; @Cheng2017].
Any motifs that were found in fewer than 6 gene 3'UTRs, as annotated by @Pelechano2013, were removed.

We quantified the regulatory effects of the remaining 38 candidate motifs by applying a linear model predicting half-life to 2 recent transcriptome-wide analyses of mRNA decay that used metabolic labeling [@Sun2013; @Chan2018].
These datasets are loosely correlated in their half-life measurements across 4188 genes reported in both datasets, $R = 0.63$ (Figure \@ref(fig:hlife-decay-model)A). However, @Chan2018 estimated substantially smaller half-lives.
@Chan2018 also had greater coverage of genes in the yeast genome, 5529 vs 4304,  and used multiple time points to determine half-lives. 
Following @Cheng2017, we constructed a linear model to predict a transcript's half-life using the counts of motifs in its 3'UTR, the length of the 3'UTR, and the relative codon usage in each transcript's coding sequence (see Material and Methods). 
The linear model performed similarly on both datasets by explaining 44% and 41% of the variability in half-lives for the @Chan2018 and @Sun2013 datasets respectively (Figure \@ref(fig:hlife-decay-model)C).
This predictive power is comparable to the squared correlation between the datasets ($R^2 = 0.40$).
Motifs that did not significantly contribute to the model were automatically filtered out using a greedy algorithm maximising the Akaike information criterion (AIC) during both training stages.
Approximately 1.7% of the variance is explained by 7 significant motifs, with 42.0% explained by codon usage (Supplementary Table \@ref(tab:hlife-variance-table)), consistent with previous analyses [@Presnyak2015;@Cheng2017].
The top 7 most significant motifs from the @Chan2018 data showed similar regulatory behaviour when tested on their own in the @Sun2013 data, except for TGTAAATA which was stabilising in one dataset and destabilising in another, as we later discuss (Figure \@ref(fig:hlife-decay-model)B).

We selected 4 motifs for exploring context dependence: TGTAHMNTA, GTATACCTA, HWNCATTWY, and ATATTC (Table \@ref(tab:motif-summary-table)).
TGTAHMNTA and GTATACCTA were chosen as they had the largest coefficients amongst significant decay and stability motifs, respectively.
HWNCATTWY was chosen due to its statistically significant effect in both datasets and, as it co-occurs with TGTAHMNTA in 68 native 3'UTRs, because it could be used for testing motif interactions.
The final selected motif was ATATTC, as it is a statistically significant decay motif in both datasets, and it has been previously shown to lower mRNA abundance when inserted in reporter constructs [@Cheng2017].
Functionally, TGTAHMNTA is the binding motif for Puf4p, and HWNCATTWY is associated with Khd1p/Hek2p-bound transcripts [@Hogan2008].
However, it is not known how ATATTC and GTATACCTA affect mRNA decay.


(ref:tab-2-caption) \textbf{Summary of shortlisted motif characteristics.} The first 3 columns hold the consensus sequence for each motif and the exact versions deleted from or inserted into  the host terminators. Then, we report the paper from which the motif was selected from; @Hogan2008, @Shalgi2005 or @Cheng2017. Next, each motif's coefficient given by the linear model predicting either the @Chan2018 or the @Sun2013 half-life datasets are included. Finally, the table includes notes on motif functions.


```{r motif-summary-table}
readr::read_csv(here("results_chapter/data/motif_summary_table.csv")) %>%
  knitr::kable(format = "latex", align = "c", caption = "(ref:tab-2-caption)") %>%
  kableExtra::column_spec(c(1,2,3,7), width = "2.3cm") %>%
  kableExtra::column_spec(c(5,6), width = "1.3cm") %>%
  kableExtra::row_spec(0, bold = TRUE)
```


(ref:hlife-decay-model-caption) \textbf{A linear model of transcript half-life quantifies the effect of candidate terminator motifs on half-life.} (\textbf{A}) Correlation between the 2 transcript half-lives ($\lambda$), in minutes, reported in the @Chan2018 and @Sun2013 datasets. (\textbf{B}) Predicted contributions to log2 half-life for chosen motifs in the @Chan2018 and @Sun2013 datasets. (\textbf{C}) Predicted vs actual transcript half-lives  calculated by a linear model of codon and motif usage trained on the @Chan2018 and @Sun2013 datasets.

```{r hlife-decay-model, fig.cap="(ref:hlife-decay-model-caption)", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/hlife_model_multi_fig.png"))
```

## Quantification of differential expression due to motif insertion or mutagenesis in multiple 3'UTRs

To quantify the effects and composability of selected motifs in different contexts, we designed a further set of reporter constructs (Figure \@ref(fig:tRPS3-tTSA1-design-and-qpcr)A).
Insertion reporter constructs contain combinations of motifs inserted in each of 2 host terminators, expressing mCherry from each of 3 different promoters.
We first chose the ribosomal protein terminator tRPS3, as it was the only terminator in our initial library that did not contain any of the selected motifs.
We selected thioredoxin peroxidase tTSA1 as the second host terminator because it also lacks selected motifs and has similar length to tRPS3.
In each host terminator we chose 3 motif insertion sites, selecting for: minimum impact on transcript secondary structure, avoiding known transcription termination elements, and matching the positions of motifs in native genes.
Having 3 insertion sites enabled us to quantify combinations of motifs, including duplicates of weaker motifs to increase the likelihood of detecting a clear effect on gene expression. 
We chose TGTACAATA and TTTCATTTC sequences as explicit versions of the TGTAHMNTA and HWNCATTWY consensus motifs respectively, and checked that these explicit versions have similar predicted effects on half-life transcriptome-wide (Supplementary Tables \@ref(tab:HWNCAUUWY-motif-coef), \@ref(tab:TGTAHMNTA-motif-coef)).
Altogether, 7 variant terminators were designed for these 2 host terminators: the wildtype terminator, a control to test the insertion sites with randomly generated sequences, 4 testing the effects of inserting each motif individually and a final variant to test interactions between the TGTAHMNTA and HWNCATTWY motifs.
We created a construct library by pairing each terminator with three different promoters; its native promoter pairing (pRPS3 or pTSA1), the high-expression promoter pPGK1, and the low-expression promoter pSRO9.

Motifs predicted to affect half-life affect mRNA abundance when inserted into tRPS3 in reporter constructs (Figure \@ref(fig:tRPS3-tTSA1-design-and-qpcr)B).
We measured mRNA abundance by RT-qPCR across 6 biological replicates, each quantified in 3 technical replicates and normalised by the $\Delta Cq$ method against values from 3 reference mRNAs (see methods).
Insertion of 2 copies of ATATTC (mod_NAA) generally lowers the mRNA abundance, as much as 4-fold when paired with the pRPS3 promoter.
Insertion of either TGTAHMNTA (mod_NTN), or 2 copies of HWNCATTWY (mod_HNH), tends to decrease mRNA abundance, and their combined insertion (mod_HTH) tends to decrease mRNA abundance even further.
The putative stability motif GTATACCTA (mod_NGG) does not consistently or strongly affect mRNA abundance.
However, comparison of the WT and control (mod_NNN) terminators does show that the creation of the insertion sites alone has an effect on mRNA levels.

Inserting the same motifs into our second host terminator gives qualitatively similar results (Figure \@ref(fig:tRPS3-tTSA1-design-and-qpcr)C).
Decay motifs generally lead to decay, although ATATTC (mod_NAA) has a weaker effect in tTSA1 than in tRPS3, and TGTAHMNTA (mod_NTN) has a stronger effect in tTSA1 than tRPS3.
The putative stability motif GTATACCTA (mod_NGG) again has little effect.

We next quantified the effects of removing decay motifs from a native yeast terminator. 
We selected the cell wall protein PIR1 as our host terminator as it is only 258 bp (@Pelechano2013) and a de-novo-synthesizable terminator that contains the ATATTC, TGTAHMNTA, and HWNCATTWY motifs.
We designed 8 terminators in which the motif occurrences in tPIR1 were replaced by scrambled sequences (Figure \@ref(fig:tPIR1-design-and-qpcr)A).
We found that the removal of almost any decay motif from tPIR1 results in an increase in mRNA levels (Figure \@ref(fig:tPIR1-design-and-qpcr)B).

We confirmed that motif-dependent changes in mRNA abundance are reflected in protein abundance by measuring the fluorescence from a subset of reporter constructs with native promoter-terminator pairings (Supplementary Figure \@ref(fig:protein-vs-RNA-plot-motifs)). 
The high correlation ($R = 0.96, 0.68, 0.96$ for tRPS3, tTSA1, tPIR1 constructs respectively) demonstrates that these combinations of decay motifs that change mRNA abundance also change the protein output, as expected.

Comparison of mRNA abundance across all constructs (Figure \@ref(fig:tRPS3-tTSA1-design-and-qpcr)B, \@ref(fig:tRPS3-tTSA1-design-and-qpcr)C, \@ref(fig:tPIR1-design-and-qpcr)B) shows motif contributions change in magnitude but not direction depending on the context of the rest of the construct. 
The insertion of almost any decay motif into tTSA1 or tRPS3 results in a decrease in mRNA abundance, and removal of these from from tPIR1 results in an increase in mRNA abundance.
However, the quantitative effects vary depending both on the immediate motif context in the host terminator and on the more distant context given by the promoter.

```{r tRPS3-tTSA1-design-and-qpcr, fig.cap="\\textbf{Motifs inserted into RPS3 and TSA1 host terminators change transcript abundance in RT-qPCR measurements.} (\\textbf{A}) Design of motif insertion sites in native RPS3 and TSA1 terminators, highlighting random insertion used as a negative control. (\\textbf{B}) Fold changes in transcript abundance for tRPS3 constructs paired with 3 promoters; pRPS3, pPGK1 and pSRO9. (\\textbf{C}) Fold changes in transcript abundance for tTSA1 constructs paired with three promoters; pTSA1, pPGK1 and pSRO9. Each diamond represents a biological replicate, averaged over 3 technical replicates. The vertical line represents the mean of all 6 biological replicates. Fold changes are relative to the abundance of the mod\\_NNN construct, i.e. $2^{\\Delta\\Delta Cq}$ (see methods).", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/insertion_constructs_design_and_qpcr.png"))
```


```{r tPIR1-design-and-qpcr, fig.cap="\\textbf{Motifs removed from PIR1 host terminators change transcript abundance in RT-qPCR measurements.} (\\textbf{A}) Design of PIR1 constructs with combinations of motifs replaced by random nucleotide sequences. (\\textbf{B}) Fold changes in transcript abundance for tPIR1 constructs paired with 3 promoters; pPIR1, pPGK1 and pSRO9. Each diamond represents a biological replicate, averaged over 3 technical replicates. The vertical line represents the mean of all 6 biological replicates. Fold changes are relative to the abundance of the WT construct, i.e. $2^{\\Delta\\Delta Cq}$ (see methods).", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tPIR1_design_and_qpcr.png"))
```


## Motif effects on gene expression depend both on terminator context and promoter pairing

We compared the effects of cis-regulatory motifs on mRNA abundance to predicted effects inferred from the transcriptome-wide measurements of half-life.
First, we trained a linear model using the RT-qPCR results to estimate the change in log2 mRNA abundance (i.e. $\Delta Cq$) due to the presence of a motif in each promoter and terminator combination.
Using a simple model of transcript production and decay, we can argue that changes in mRNA abundance are directly proportional to changes in mRNA half-life (see methods).
Therefore, we compared the motif's estimated change in log2 mRNA abundance to that predicted due to changes in log2 half-life, estimated from our transcriptome-wide analysis of the @Chan2018 dataset.
The effect of motifs in reporter constructs is correlated with the predictive model, but the strength of the correlation depends on context (Figure \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)A).
Constructs with inserted motifs (i.e. tRPS3 and tTSA1 constructs) had a lower correlation with predicted effects than constructs with deleted motifs (i.e. tPIR1 constructs).
Interestingly, motif effects on mRNA abundance appear to be greater than that predicted from their effect on half-life when their host terminator is paired with its native promoter.

We next directly compared the estimated coefficients for each motif's effect on mRNA abundance across promoter-terminator pairing (Figure \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)B).
The effect of a motif depends on terminator context.
For example, ATATTCA reduces mRNA abundance substantially more when inserted in tRPS3 than in tTSA1. 
Meanwhile, TGTAHMNTA significantly reduces mRNA abundance when inserted in tTSA1, but not tRPS3, whichever promoter is chosen.
Promoter choice also influences the magnitude of a motif's contribution to mRNA levels.
For the ATATTC, TGTAHMNTA and HWNCAUUWY motifs the greatest reduction in mRNA abundance occurred when native promoter-terminator pairings are measured.
This is true for all 3 decay motifs across all 3 host terminators, except for HWNCAUUWY in pRPS3-tRPS3 constructs.

Regulatory interactions between different motifs also change depending on host terminator and promoter context.
We included an interaction term that quantifies how the effect of including both HWNCAUUWY and TGTAHMNTA together differs from the sum of the effects of including these motifs individually.
The combination of TGTAHMNTA and HWNCAUUWY in tRPS3 has no additional effect beyond a simple sum of their individual effects when paired with pPGK1.
However, when tRPS3 is paired with pRPS3 or pSRO9, the combination has a greater effect than expected from the sum of each motif's individual effect.
The combination of TGTAHMNTA and HWNCAUUWY in tTSA1 has no additional effect, except when paired with pPGK1, where it has a lesser effect than expected.
Finally, the combination of TGTAHMNTA and HWNCAUUWY in tPIR1 always has an additional effect across the three promoters. 
The combination has a greater effect than expected when paired with pPIR1 but a lesser effect when paired with pPGK1 or pTSA1.


```{r hlife-predict-vs-abundance-and-motif-context-dependence, fig.cap="\\textbf{Promoter and terminator context alter the regulatory effect of motifs.} (\\textbf{A}) Predicted change in transcript abundance inferred from transcriptome-wide motif contributions to half-life, compared to RT-qPCR measurements of reporter transcript abundance. The y-axis shows statistical summaries (mean and standard error) of 6 replicates for data shown in figures 4 and 5. The fold change is relative to the mod\\_NNN construct in each promoter-terminator pairing for the tRPS3 and tTSA1 sets, and relative to the WT constructs for tPIR1 sets. Native promoter panels show the promoter paired with the terminator from the same set, e.g. pRPS3 with tRPS3. (\\textbf{B}) Motif contributions to fold changes in mRNA abundance for reporter constructs with different promoter and terminator contexts. This is calculated by a linear model with a coefficient for the effect of each motif in each set, applied to $\\Delta Cq$ against 3 reference genes. The last column shows the interaction term between TGTAHMNTA and HWNCATTWY.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("results_chapter/figures/qPCR_model_coef_and_pred_vs_exp_abund.png"))
```

## Inserting motifs into terminators shifts poly(A) site usage downstream

We next mapped poly(A) site usage in a subset of reporter constructs, for 2 reasons.
First, changes in poly(A) site usage might mean that motifs placed in our reporters were unintentionally absent from the mature mRNA.
Second, we wanted to know if motif effects on mRNA abundance might be due to changes in the poly(A) site.
We chose three constructs with large effect sizes in the qPCR results: mod_NAA, mod_HTH and mod_NTN, together with WT and mod_NNN controls, within three promoter-terminator contexts: pRPS3-tRPS3, pPGK1-tRPS3 and pTSA1-tTSA1.
For these constructs we performed poly(A)-proximal RNA-seq using the QuantSeq Rev protocol with paired-end reads ([@Moll2014] and see methods).
Read 1 allows precise inference of poly(A) site position while read 2 generally overlaps the CDS, so distinguishing terminators in native loci from reporter constructs.
We mapped the reads to genome sequences extended by the relevant reporter plasmid sequence.

We detected 1000s of reads on each reporter construct, which is enough to quantify expression confidently as well as to assign poly(A) sites.
We checked that counts of all other RNAs are highly correlated between samples, giving us confidence that changes in construct detection are meaningful (Figure \@ref(fig:rnaseq-QC-correlation)).
Transcript abundance, relative to mod_NNN, for most constructs correlate strongly with qPCR results (Figure \@ref(fig:quantseq-polyA-site-usage)A).
However, some constructs were detected as more abundant by RNA-seq for reasons that are unclear.

We display the poly(A) sites as the cumulative fraction of poly(A)-site reads mapped at each location downstream of construct stop codons, out of all reads mapped to the terminator (Figure \@ref(fig:quantseq-polyA-site-usage)B).
We checked that in other genes the poly(A) site locations were highly reproducible across samples, giving us confidence that changes in construct poly(A) sites are meaningful (Figure \@ref(fig:rnaseq-QC-QuantSeq-and-5PSeq-genomic-polyA)).
Poly(A) sites are in the same relative positions in native loci and the construct with wild-type terminator (Figure \@ref(fig:rnaseq-QC-genomic-vs-plasmid-polyA)).
Then, we compared 3'end positions of reads between modified and wild-type reporter constructs to determine the poly(A) site usage.
In both wild-type and mod-NNN tTSA1 constructs, the same poly(A) sites are used.
Surprisingly, tRPS3 mod_NNN constructs appear to be using a novel upstream poly(A) site, that appears in about $50%$ percent of reads and is located upstream of the 3rd motif insertion site.
For tTSA1 constructs the same poly(A) sites are being used in mod_NNN constructs as wildtype (Figure \@ref(fig:quantseq-polyA-site-usage)B).

Next, we compare changes in poly(A) site usage between constructs with different inserted motifs.
We highlight the 2 major poly(A) sites for the tRPS3 constructs and 2 for the tTSA1 constructs (black vertical lines on the mod_NNN constructs in Figure \@ref(fig:quantseq-polyA-site-usage)B) and track the cumulative fraction of reads upstream of each major site.
However, for tRPS3 constructs there is a distinct shift to downstream poly(A) sites in constructs with verified, rather than random, motifs inserted. 
For tTSA1 constructs there is little change in poly(A) site usage across all constructs.


```{r quantseq-polyA-site-usage, fig.cap="\\textbf{Inserting motifs into RPS3 and TSA3 terminators changes 3'UTR length.} A subset of constructs were chosen to investigate changes in poly(A) site usage: WT, mod\\_NNN, mod\\_NTN, mod\\_NAA, and mod\\_HTH; within three promoter-terminator contexts: pRPS3-tRPS3, pPGK1-tRPS3, and pTSA1-tTSA1. (\\textbf{A}) Comparison of construct transcript abundance as independently measured by qPCR and RNA-Seq assays. Transcript abundance was normalised to the median abundance of  plasmid URA3, genomic PGK1, and RPS3 or TSA1 transcripts for each construct. Fold change is relative to the mod\\_NNN construct in each promoter-terminator context. The black diagonal line represents the expected values if RNAseq and qPCR results correlated perfectly. (\\textbf{B}) Cumulative count of reads mapped downstream of WT (grey) and mod\\_NNN (golden) construct stop codons as fraction of total reads mapped to the constructs terminator. (\\textbf{Left}) shows the poly(A) sites for the pRPS3-tRPS3 and pPGK1-tRPS3 promoter-terminator constructs. (\\textbf{Right}) shows the same for the pTSA1-tTSA1 constructs. WT reads have been shifted downstream to align with the mod\\_NNN sequence by accounting for motif insertion sites. These are represented by red regions in the cumulative plot. Major poly(A) sites have been highlighted by a black vertical line. (\\textbf{C}) Similar to Figure \\textbf{B} but with each motif insertion construct plotted separately. Columns designate cumulative plots from different terminator constructs and rows designate cumulative plots from different promoter contexts.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("results_chapter/figures/polya_usage_plot_QuantSeq.png"))
```

We then asked whether modifications in the 3'-UTR region can impact translation through changing ribosome dynamics and 5'-3' co-translational degradation pathway.
Our 5PSeq data shows that there are no significant changes in ribosome stalling between wild-type and modified reporter constructs (Figure \@ref(fig:rnaseq-QC-genomic-vs-plasmid-polyA)), suggesting that mechanisms by which inserted motifs regulate the expression of reporter protein do not impact ribosome dynamics.
Moreover, the poly(A) site distribution for each construct matches that obtained from QuantSeq data, suggesting that 3'-UTR isoforms are not differentially degraded using this pathway regardless of the motifs inserted.
Finally, the abundances of different reporter mRNAs from different constructs correlate well between 5PSeq and QuantSeq data **correlation plot needed**, but are not correlated genome wide (Figure \@ref(fig:rnaseq-QC-QuantSeq-vs-5PSeq)).
