# Results

```{r}
library(here)
```

## Initial characterisation of promoter-terminator swaps and determining their effects on RNA abundance
Initially, to test the effects of 3'UTRs on RNA abundance, promoter-terminator swaps consisting of a collection of 12 (**10 ?**) terminators and 3 promoters driving the expression of an mCherry CDS in a plasmid were constructed. Unlike in previous studies where expression enhancing terminators were identified for further investigation, we selected terminators of from genes of varying expression profiles (Curran et al. 2013, Ito et al. 2013, Ito et al. 2016)

- pRPS3 with 12x terminator swaps
- pPGK1 withh 12x terminator swaps
- pSRO9 withh 12x terminator swaps

## Selecting cis-regulatory elements by their contributions to linear models of transcript half life

Potential cis-regulatory elements to test for context dependence were initially identified through a literature search. We focused on cis-regulatory elements associated with post-transcriptional regulation only. This was in order to a) enable confirmation of each element's regulatory role by predicting their contributions to transcript degradation and b) minimise the possibility of selecting motifs with confounding transcriptional, post-transcriptional and translational roles. 69 suspected post-transcriptional, cis-regulatory elements were collated from three separate studies; Hogan et al (2008), Shalgi et al (2005) and Cheng et al (2017). All collated cis-regulatory elements were sequence motifs found in transcript 3' untranslated regions (3'UTR) as confirmed either in vivo, with RNA-protein binding assays (Hogan et al., 2008), or in silico, by comparing the sequences of transcripts with similar half lives (Shalgi et al., 2005; Cheng et al., 2017).

We shortlisted the collated 3'UTR motifs by determining their contributions to a linear model predicting native transcript half lives.  The linear model used the counts of the motifs in the median length 3'UTR of a gene's transcript as a predicate. A transcript's median length 3'UTR was determined either as reported by Pelechano et al. (2013) or, for genes missing in the TIF-seq dataset, inferred by taking the first 200 nucleotides downstream of the gene's stop codon. 31 of the original 69 motifs occurred 5 or fewer times in the 3'UTRs as defined above and were subsequently removed. The linear model also included terms for the relative codon usage in each transcript's ORF as this has previously been shown to be the most significant determinate of decay rate (Cheng et al., 2017; Presnyak et al., 2015), see Material and Methods. 

The linear model was trained separately on two independent, genome-wide studies of yeast mRNA decay rates, Sun et al. (2013) and Chan et al. (2017). The two data sets are loosely correlated, R = 0.63 Figure \@ref(fig:chan-vs-sun), with both using non-invasive pulse labeling assays to measure decay rates. However, only Chan et al. (2017) used multiple time points to determine transcript half lives. The motifs were then selected for their contribution to half life by using a greedy algorithm to determine the combination of motifs that maximised the Akaike information criterion (AIC) of the model. The predictive power of the models with the maximum AIC can be seen in Figure \@ref(fig:model-predictive-power); ??% and 51% of the variability in half lives was explained for the Sun et al (2013) and Chan et al (2017) data sets respectively. Motifs predicted to be significant in both decay data sets are plotted in Figure \@ref(fig:motif-coefficient-comparison) with their magnitudes calculated according to the data sets plotted against each other. Interestingly, the TGTAAATA motif was predicted to be stabilising in the Sun et al (2013) data set but destabilising in the Chan et al (2017) data set. The puf3 protein associated with motif is known to change function in carbon deficient condition which were likely present due to the bespoke CSM-lowURA media the cultures were grown in overnight, Sun et al (2013) used standard YPD media. 


```{r chan-vs-sun, fig.cap="Comparison of Sun et al (2013) and Chan et al (2017) genome wide transcipt half lives.", out.width='40%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/chan_vs_sun_plot.png"))
```

```{r model-predictive-power, fig.cap="Performance of codon and motif linear model predicting the transcript half lives as reported in the Chan et al (2017) and Sun et al (2013).", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/model_predictive_power.png"))
```


```{r motif-coefficient-comparison, fig.cap="Predicted coefficients of motifs with significant regulartory effects in both Sun et al (2013) and Chan et al (2017) datasets.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/motif_coefficient_comparison.png"))
```

Four motifs suitable for testing for context dependence were selected from the list of motifs with statically significant effects on half life according to the Chan et al (2017) data set, Table \@ref(tab:chan_motif_coef). First, the motifs with largest predicted stabilising and destabilising coefficients were chosen, GTATACCTA and TGTAHMNTA respectively. Second, the HWNCATTWY motif was chosen due to its statistically significant effect in both data sets and because it is a suitable candidate for testing for interactions between motifs as it regularly occurs with other motifs, for example in 68 of the 317 native 3'UTRs containing TGTAHMNTA. The final selected motif was ATATTC again a statistically significant motif in both data set but also as an experimental control by comparing results with previous work conducted by Cheng et al (2017). 

```{r chan_motif_coef}
chan_motif_table <- read.csv(here("./results_chapter/data/chan_motif_coefficients.csv"))
knitr::kable(chan_motif_table)
```

## Terminator selection

To comprehensively investigate the context dependence of the chosen cis-regulatory elements, three native terminators were selected for either the insertion of the motifs or the deletion of the motifs. Terminators suitable for insertion were required to have less than two of the motifs detected to be significant according to the Chan et al (2017) trained linear model. Meanwhile, terminators suitable for deletion had to have all of the motifs present except the stability motif GTATACCTA as this only occurs in limited number of native terminators. In addition, all terminators were chosen to contain relatively short, <300nt median 3'UTRs according to Pelechano et al (2013), to minimise confounding effects from the formation of complex secondary structures. Terminators were also selected to be within the top 5% most abundant genes according to the Nadal-Ribelles et al (2019) scRNA-seq dataset. The terminators selected for insertion had an addition constraint of a minimum median 3'UTR length of 50nt to enable some flexibly in the positions of insertion to reduce the likelihood of affecting other cis-regulatory elements known to contribute significantly to transcription rates, such as efficiency elements and polyadenylation sites.  The shortlisted terminators for insertion were RPS3, a ribosomal protein, and TSA1, a redox homeostasis protein, and the terminator for deletion was PIR1, a cell wall protein. The selection process is summarised in supplementary figure 1 with supplementary table 1 containing the shortlist of genes. 

## Functional validation of cis-regulatory motifs in selected 3'UTRs

Candidate cis-regulatory motifs on the 3'UTR (UGUAHMNUA, HWNCAUUWY, ATATTC, GTATACCTA) previously predicted with large contribution on RNA half life from the linear model trained on the Chan et al. (2017) half life dataset were tested for their modularity in different 3'UTR contexts and combined linearly to determiine their effects on RNA abundance. These sequence motifs were introduced to two 3'UTRs (tRPS3 and tTSA1) devoid of the candidate consensus motifs of interest and functionally characterised for their effect on RNA abundance by RT-qPCR. The TGTACAATA and TTTCATTTC sequences that satisfy the UGUAHMNUA and HWNCAUUWY consensus motifs respectively were chosen and used for subsequent experimental validation. **[Check why specifically these sequencces were chosen - out of all combinations of the consensus motifs]**

#### Effects of candidate cis-regulatory motifs in tRPS3 3'UTR-terminator in combination with different promoters

tRPS3 3'UTR-terminator were modified to incorporate three 9 bp cloning sites for motifs (M1, M2 and M3). The M1 was designed 24 bp downstream of the stop codon. Between M1 and M2 is a 15 bp spacer. M3 is located 4 bp downstream of M2 (Figure \@ref(fig:tRPS3-design-and-qpcr)). **[Supplementary figure - rationale for spacing]** A control tRPS3 3'UTR (mod0) was designed to incorporate scrambled/random bases in each cloning site, each with a GC content of 56% (Figure 1A). Two copies of the candidate decay motif ATATTC were incorporated to tRPS3-modB at sites M2 and M3. Similarly, two copies of the candidate stability motif (GTATACCTA) were introduced in tRPS3-modC 3'UTR. One copy of the putative decay motif (UGUAHMNUA) was introduced to tRPS3-modA at site M2 while two copies of another putative decay motif HWNCAUUWY motif were introduced to tRPS3-modD at sites M1 and M3. In order to test the effect of linear combinations of putative decay motifs  on mRNA abundance, the final design incorporated a combination of UGUAHMNUA at site M2 and HWNCAUUWY at sites M1 and M3. These 3'UTR/terminators were asssembled with either pRPS3 or pPGK1 promoters driving the expression of an mCherry reporter gene in a plasmid and expressed in yeast.

**[To add - modB ATATTC design mimicked Cheng et al design]**




Analysis of RNA abundance by RT-qPCR was carried out on these strains showed a decrease in RNA abundance with the incorporation of putative decay motifs (ATATTC, TGTACAATA and TTTCATTTC). The relative fold change associated with each motif varies depending on promoter choice. A ~4 fold decrease in RNA abundance with the incorporation of ATATTC when the tRPS3 3'UTR-terminator is coupled with the pRPS3 promoter. When coupled with the pPGK1 promoter, only a ~2 fold decrease in RNA abundance was observed with the incorporation of ATATTC when the tRPS3 3'UTR-terminator. With the incorporation of both candidate decay motifs - TGTACAATA and TTTCATTTC, a ~4 fold decrease in RNA abundance was observed when the tRPS3 3'UTR-terminator is coupled with the pRPS3 promoter however, only a ~2 fold decrease in RNA abundance was observed when coupled with the pPGK1 promoter. Our results demonstrate that, consistent with our model, the incorporation of candidate decay motifs result in a moderate but significant decrease in protein abundance.

```{r tRPS3-design-and-qpcr, fig.cap="Design of modified tRPS3 3'UTRs and RT-qPCR measurement of RNA abundance in each reporter construct.", out.width='85%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tRPS3_design_and_qpcr.png"))
```
#### Effects of candidate cis-regulatory motifs in tTSA1 3'UTR-terminator in combination with different promoters

tTSA1 3'UTR-terminator were modified to also incorporate three 9 bp cloning sites for motifs (M1, M2 and M3). The M1 was designed 21 bp downstream of the stop codon. M2 is designed 20 bp downstream of M1 and M3 is designed 24 bp downstream of M2. **[Supplementary figure - rationale for spacing.]** The tTSA1 mod0 control scrambled sequences inserted into the M1, M2 and M3 sites contained scrambled/random bases with GC contents of 33%, 22% and 33% respectively (Figure \@ref(fig:tRPS3-design-and-qpcr)). Similar to tRPS3 constructs, two copies of the candidate decay motif ATATTC were incorporated to tTSA1-modB at sites M2 and M3. Similarly, Two copies of the candidate stability motif (GTATACCTA) were introduced in tTSA1-modC 3'UTR. One copy of the putative decay motif (UGUAHMNUA) was introduced to tTSA1-modA at site M2 while two copies of another putative decay motif HWNCAUUWY motif were introduced to tRPS3-modD at sites M1 and M3. In order to test the effect of linear combinations of putative decay motifs  on mRNA abundance, the final design incorporated a combination of UGUAHMNUA at site M2 and HWNCAUUWY at sites M1 and M3. These 3'UTR/terminators were asssembled with either pTSA1 or pPGK1 promoters driving the expression of an mCherry reporter gene in a plasmid and expressed in yeast.

- RT-qPCR data and results analysis for tTSA1 swaps with pTSA1 and pPGK1  

```{r tTSA1-design-and-qpcr, fig.cap="Design of modified tTSA1 3'UTRs and RT-qPCR measurement of RNA abundance in each reporter construct.", out.width='85%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tTSA1_design_and_qpcr.png"))
```

#### Modelling the relationship of cis-regulatory motifs on RNA abundance

We fitted a linear model to each dataset (based on promoter - pRPS3 or pPGK1) to determine the contribution of each motif on RNA abundance. The linear model explains ~70% of the variability in RNA abundance due to cis-regulatory motifs in the 3'UTR. With both promoters, the putative stability (GTATACCTA) sequence show no significant effect on RNA abundance. The ATATTC and TTTACTTTC cis-regulatory sequences contribute to a moderate but significant decrease RNA abundance (p<0.001 wth both promoters). Interestingly, the decay (TGTACAATA) motif shows variable effects when in combination with different promooters (p<0.01 with pRPS3, n.s with pPGK1). The combination of both TGTACAATA and TTTCATTTC is not additive, with no significant downregulation of gene expression when present in combination within a 3'UTR with both promooters.

```{r motif-coeff-RNA-abundance, fig.cap="Modelling the relationship of genetic elements on RNA abundance (using experimental data). Confidence intervals are within two standard errors.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/motif_coeff_RNA_abundance.png"))
```


## Comparison of motif effects from qPCR and half-life dataset models

```{r motif-context-dependence, fig.cap="Experimental transcript abundance vs predicted abundance for all constructs according to predicted changes in decay rate due to the presence of 3'UTR motifs. Confidence intervals are within a standard error.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/pred_vs_actual_abundance.png"))
```

```{r motif-context-dependence, fig.cap="Motif contributions to fold changes in RNA abundance for different promoter and terminator contexts, with each promoter-terminators pairing's mod0 construct as base value. Confidence intervals are within a standard error.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/motif_multi_context_coefficient.png"))
```

## Investigating the effects of cis-regulatory motifs in the 3'UTR using 5PSeq
We used high-throughput 5PSeq (HT-5Pseq) technique to investigate whether modifications of 3’ UTRs influence the process of 5’-3’ co-translational degradation of mRNA, which can also serve as a proxy of translation process (Zhang & Pelechano 2020), and modify ribosome stalling in the 3’-end region of coding sequence. The 5PSeq method allows to investigate the process of  5’-3’ degradation of mRNA by capturing only de-capped 5’ ends of mRNA molecules, and at the same time – the polyadenylated 3’ ends. We generated sequencing reads with insert sizes up to 600 bp, which included entire 3’UTRs and up to 400 bp of coding sequence.  

The distributions of 5’ ends did not differ significantly between the samples (figure 4), while at the same time they differed from those observed in native RPS3, suggesting that the coding sequence and not 3’UTR is the major contributor to ribosome stalling observed near 3’ end of coding sequence of RPS3 (figure **missing figure**, which I forgot to make – I will by the end of today). However, there were major differences between distributions of 3’ ends between different constructs corresponding to different polyadenylation sites, especially between constructs with non-modified (tRPS3-WT) and modified (tRPS3-mod0, tRPS3-modA and tRPS3-modB) 3’UTRs (figure 5 and 6). In all modified 3’ UTRs we observed additional poly(A) sites at 44 nt and 56 nt from STOP codon, which corresponds to 35 nt and 47 nt in WT 3’ UTR. Secondly, we observed differences in the distribution of poly(A) sites between modified 3’UTRs with different motifs inserted. Poly(A) sites located 44 nt and 56 nt from STOP codon are more abundant in tRPS3-mod0 compared to tRPS3-modA, with tRPS3-modB between them. Poly(A) site at 67 nt is pronounced only in tRPS3-mod0 and at 74 nt is especially visible in tRPS3-modA and at 80 nt in tRPS3-modB. The differences could be caused either by new polyadenylation signals introduced into the 3’ UTR or by increased stability of these isoforms conferred by addition of new motifs.  


![5PSeq figure](`r here("results_chapter/figures/results_figure6.png")`){ width=70% }

![5PSeq figure](`r here("results_chapter/figures/results_figure7.png")`){ width=70% }

![5PSeq figure](`r here("results_chapter/figures/results_figure8.png")`){ width=70% }


## Comparison of whole-genome ATATTC motif effect vs. reporter-gene ATATTC motif effect
- Comparison of our qPCR data of modB with that of Cheng et al. (2017)

## Effect of motif copy number and position in the 3'UTR on RNA abundance
- increasing number of ATATTC motifs
- increasing number of TTTCATTTC motifs (interesting because this often occurs in 2x)
