# Results

```{r}
library(here)
knitr::opts_chunk$set(fig.pos = "p")
```

## Terminator effects on RNA abundance and protein production are context dependent

```{r pro-ter-platereader, fig.cap="\\textbf{Terminator contributions to gene expresion are promoter dependent.} (\\textbf{A}) mCherry fluorescence measured at the time of each sample's maximum growth rate for 10 termintators paired with three highly expressing promoters. (\\textbf{B}) Same as (\\textbf{A}) but with two lowly expressing promoters.", out.width='90%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/pro_ter_platereader.png"))
```

Initially, to test the effects of terminator selection on mRNA and protein production, we selected a set of 10 terminators from transcripts with of varying expresssion levels.
These terminators were cloned into a pRS415-derived plasmid along with mCherry or mTurquoise2 and a promoter.
The effect of terminator choice was tested with 5 promoters (pPGK1, pRPS3, pRPS13, pCLN2 and pSRO9) to determine any differring effects protein production.
Terminator choice on protein output was evaluated by measuring the fluorescence over time from these expression casssettes in S. cerevisiae BY4741.

## Short vs long terminator experiment

```{r RPS3-SRO9-short-vs-long, fig.cap="\\textbf{Extending terminator sequence beyond transcript 3'UTR length enhances gene expression}. (\\textbf{A}) Fold changes in mRNA abundance for two RPS3 terminator sequences equaling reported 3'UTR isoform lengths, 59bp and 86bp, and one terminator sequence extended beyond the maximum reported 3'UTR length, 200bp. (\\textbf{B}) Same as (\\textbf{A}) but for one SRO9 termintor sequence shorter than reported 3'UTR isoform lengths, 200bp, and one beyond, 500bp.", out.width='60%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/RPS3_SRO9_short_vs_long.png"))
```

## Candidate cis-regulatory elements contribute to transcript decay rates


We wanted to confirm that the context dependent regulation of gene expression observed in terminator regions is replicated by the individual cis-regulatory elements (CREs) contained with these regions.
A literature search revealed a list of 69 suitable CREs that were suspected sequence motifs for RNA binding proteins (Sup Table 1) several of which appeared regularly in the 3' untranslated regions (3'UTR) of our library of terminators (Sup Table 2) [@Hogan2008] [@Shalgi2005] [@Cheng2017].
Across the yeast genome 31 of the 69 motifs occurred 5 or fewer times in the median length 3'UTRs reported by @Pelechano2013.
They were therefore removed.
A subset of the remaining 38 3'UTR motifs had been verified as binding motifs for proteins associated with various mRNA degradation pathways (Sup Table 1).

Active regulatory roles of candidate CREs were validated using a linear model predicting half life.
As some motifs had previously been shown to be involved with transcript degradation, we constructed a linear model to predict a transcript's half life using the counts of motifs in its median length 3'UTR.
The linear model also included terms for the relative codon usage in each transcript's ORF as this has previously been shown to be the most significant determinate of decay rate [@Cheng2017] [@Presnyak2015], see Material and Methods. 
The linear model was trained separately on two independent, non-invasive pulse labeling assays measuring yeast mRNA decay rates, [@Sun2013] and [@Chan2018].
With the two data sets loosely correlated, R = 0.63 Figure \@ref(fig:hlife-decay-model)B, the model performed similarly on both by explaining 44% and 41% of the variability in half lives for the @Chan2018 and @Sun2013 data sets respectively, \@ref(fig:hlife-decay-model)A.
CREs that did not significantly contribute to the model were automatically filter out using a greedy algorithm maximising the Akaike information criterion (AIC) during both training stages.

We shortlisted the collated 3'UTR motifs by determining their contributions to a linear model predicting native transcript half lives.
The linear model used the counts of the motifs in the median length 3'UTR of a gene's transcript as a predicate.

However, only [@Chan2018] used multiple time points to determine transcript half lives.
Motifs predicted to be significant in both decay data sets are plotted in Figure \@ref(fig:hlife-decay-model)C with their magnitudes calculated according to the data sets plotted against each other.
Interestingly, the TGTAAATA motif was predicted to be stabilising in the [@Sun2013] data set but destabilising in the [@Chan2018] data set.
The puf3 protein associated with motif is known to change function in carbon deficient condition which were likely present due to the bespoke CSM-lowURA media the cultures were grown in overnight, [@Sun2013] used standard YPD media. 


```{r hlife-decay-model, fig.cap="\\textbf{A linear model predicting transcript half life verifies candidate RBP motifs regulate gene expression.} (\\textbf{A}) Predicted vs actual log2 transcript half lives calculated by a linear model of codon and motif usage trained on the Chan et al (2017) and Sun et al (2013) data sets. (\\textbf{B}) Correlation between the two log2 transcript half lives reported in the Sun et al (2013) and Chan et al (2017) data sets. (\\textbf{C}) Predicted contributons to log2 half life for choosen motifs in the Sun et al (2013) and Chan et al (2017) datasets. Motifs not detected to be significant in the Sun et al (2013) data set are plotted with an x-axis value of 0.", out.width='100%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/hlife_model_multi_fig.png"))
```

Four motifs suitable for testing for context dependence were selected from the list of motifs with statically significant effects on half life according to the [@Chan2018] data set, Table \@ref(tab:chan_motif_coef).
First, the motifs with largest predicted stabilising and destabilising coefficients were chosen, GTATACCTA and TGTAHMNTA respectively.
Second, the HWNCATTWY motif was chosen due to its statistically significant effect in both data sets and because it is a suitable candidate for testing for interactions between motifs as it regularly occurs with other motifs, for example in 68 of the 317 native 3'UTRs containing TGTAHMNTA.
The final selected motif was ATATTC again a statistically significant motif in both data set but also as an experimental control by comparing results with previous work conducted by [@Cheng2017]. 

```{r chan_motif_coef}
chan_motif_table <- read.csv(here("./results_chapter/data/chan_motif_coefficients.csv"))
knitr::kable(chan_motif_table)
```

## Characterisation of candidate cis-regulatory motifs via insertion or mutagenesis in selected 3'UTRs

To comprehensively investigate the context dependence of the chosen cis-regulatory elements, three native terminators were selected for either the insertion of the motifs or the deletion of the motifs.
Terminators suitable for insertion were required to have less than two of the motifs detected to be significant according to the [@Chan2018] trained linear model.
Meanwhile, terminators suitable for deletion had to have all of the motifs present except the stability motif GTATACCTA as this only occurs in limited number of native terminators.
In addition, all terminators were chosen to contain relatively short, <300nt median 3'UTRs according to [@Pelechano2013], to minimise confounding effects from the formation of complex secondary structures.
Terminators were also selected to be within the top 5% most abundant genes according to the [@Nadal2019] scRNA-seq dataset.
The terminators selected for insertion had an addition constraint of a minimum median 3'UTR length of 50nt to enable some flexibly in the positions of insertion to reduce the likelihood of affecting other cis-regulatory elements known to contribute significantly to transcription rates, such as efficiency elements and polyadenylation sites. 
The shortlisted terminators for insertion were RPS3, a ribosomal protein, and TSA1, a redox homeostasis protein, and the terminator for deletion was PIR1, a cell wall protein.
The selection process is summarised in supplementary figure 1 with supplementary table 1 containing the shortlist of genes.
The TGTACAATA and TTTCATTTC sequences that satisfy the UGUAHMNUA and HWNCAUUWY consensus motifs respectively, were chosen and used for subsequent experimental validation.

The tRPS3 3'UTR-terminator was modified to incorporate three 9 nt insertion sites for motifs (M1, M2 and M3).
The M1 was inserted 24 nt downstream of the stop codon, M2 was inserted 15 nt downstream of M1 and the final insert M3 was inserted 4 nt downstream of M2 (Figure \@ref(fig:tRPS3-design-and-qpcr)).
These positions were selected based out key design criteria (SUP FIGURE/NOTE X).
A control tRPS3 3'UTR mod0 was designed to incorporate scrambled/random bases in each insertion site (Figure 1A).
Further modified 3’UTR-terminator designs were designed to incorporate individual motifs of interest previously identified, within the insertion sites described (Figure \@ref(fig:tRPS3-design-and-qpcr)).
In particular modC was designed to characterise a putative stability motif, whilst modA, modB, modD and modE were designed to characterise putative decay motifs.
These 3'UTR/terminators were assembled in combination with either pRPS3, pPGK1 or pSRO9 promoters driving the expression of an mCherry reporter gene in a plasmid and expressed in yeast BY4741.
The strains were subsequently screened by RT-qPCR to detect abundance of the mCherry encoding transcripts.
The analysis of the constructs showed a decrease in RNA abundance with the incorporation of the ATATTC candidate decay motif (Figure \@ref(fig:tRPS3-design-and-qpcr)).
The relative fold change associated with ATATTC varies depending on promoter choice.
A 4 fold decrease in RNA abundance with the incorporation of ATATTC in the tRPS3 3'UTR was observed when assesmbled with the pRPS3 promoter.
However, when paired with the pPGK1 and pSRO9 promoters, only a 2 fold decrease in RNA abundance was observed with the incorporation of ATATTC in the tRPS3 3'UTR-terminator.
Similarly, introduction of both candidate decay motifs - TGTACAATA and TTTCATTTC in tRPS3 3'UTR resulted in a ~4 fold decrease in RNA abundance when coupled with the pRPS3 promoter however, only a 2 fold decrease in RNA abundance was observed when coupled with the pPGK1 and pSRO9 promoters.
No significant effect on RNA abundance was observed with the incorporation of a single copy of TGTACAATA in tRPS3 modA.

```{r tRPS3-design-and-qpcr, fig.cap="\\textbf{Shortlisted motifs inserted into RPS3 host terminators change transcript abundance in RT-qPCR measurements}. (\\textbf{A}) Design of motif insertion sites into the native RPS3 terminator with a diagram of the final constructs. (\\textbf{B}) Fold changes in transcript abundance for tRPS3 constructs paired with three promoters; pRPS3, pPGK1 and pSRO9. Fold changes are relative to the abundance of the mod0 construct", out.width='85%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tRPS3_design_and_qpcr.png"))
```

These motifs were also tested in a different 3'UTR context to determine if effects on RNA abundance is modular.
To this end, motifs were introduced in the TSA1 3'UTR.
Similar to the modified RPS3 3'UTR-terminator designs, three 9 nt insertion sites (M1, M2 and M3).
The M1 was designed 21 nt downstream of the stop codon, M2 was designed 20 nt downstream of M1, and M3 was designed 24 bp downstream of M2. 
The tTSA1 mod0 construct contained scrambled/random bases iin the M1, M2 and M3 sites (Figure \@ref(fig:tTSA1-design-and-qpcr)).
Similar to the modified tRPS3 3'UTR-terminator designs, individual motifs of interest previously described were introduced into the insertion sites (Figure \@ref(fig:tTSA1-design-and-qpcr)), whereby modC was designed to characterise a candidate stability motif, whilst modA, modB, modD and modE were designed to characterise candidate decay motifs.
These 3'UTR/terminators were assembled with either pTSA1, pPGK1 or pSRO9 promoters driving the expression of an mCherry reporter gene in a plasmid and expressed in yeast.

```{r tTSA1-design-and-qpcr, fig.cap="\\textbf{Shortlisted motifs inserted into TSA1 host terminators change transcript abundance in RT-qPCR measurements}. (\\textbf{A}) Design of motif insertion sites into the native TSA1 terminator with a diagram of the final constructs. (\\textbf{B}) Fold changes in transcript abundance for tTSA1 constructs paired with three promoters; pTSA1, pPGK1 and pSRO9. Fold changes are relative to the abundance of the mod0 construct", out.width='85%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tTSA1_design_and_qpcr.png"))
```

```{r tPIR1-design-and-qpcr, fig.cap="\\textbf{Shortlisted motifs removed from PIR1 host terminators change transcript abundance in RT-qPCR measurements}. (\\textbf{A}) Diagram of the final PIR1 constructs with permutations of removed motifs. (\\textbf{B}) Fold changes in transcript abundance for tPIR1 constructs paired with three promoters; pPIR1, pPGK1 and pSRO9. Fold changes are relative to the abundance of the WT construct", out.width='85%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tPIR1_design_and_qpcr.png"))
```


## Comparison of motif effects from qPCR and half-life dataset models

Using a simple model of transcript production and decay the predicted effects of motifs on half lives can be used to infer predicted changes in overall RNA abundance.
Figure \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)A shows the inferred predicted fold changes in abundance to compared to the experimentally verified changes from the qPCR experiments.
Constructs with TSA1 terminators have highing $R^2$ values than those in RPS3 terminators suggesting that motif TSA1.
It is also interesting that the SRO9 promoter pairing has the greatest correlation to the decay model suggesting that the less expressing promoter is more influenced by decay pathways. 

## Motif effects on gene expression are dependant on termintor context and promoter pairing

Host terminator dictates overall regulatory behavior of motif.
UGUAHMNUA displays strong host dependent behavior with no significant contribution to RNA levels in RPS3 terminator constructs, less than 1 fold changes in log2 abundance in PIR1 constructs and over 1 fold changes in TSA1.
By contract, ATATTC has the greatest contribution in RPS3 terminator constructs, upto 1 fold change, and less than 0.5 fold changes in PIR1 and TSA1 constructs.
Although GTATACCTA has relatively insignificant contributions to changes in mRNA abundance it is interesting that the small increase in mRNA levels for pTSA-tTSA constructs is matched by an increase in protein levels.
HWNCAUUWY appears to have a similar effect across terminators.
However,  when it appear alongside UGUAHMNUA their interaction shows antagonistic, synergistic and agnostic behavior depending on the host terminator.
Finally, the effect on abundance of removing motifs natively occurring in tPIR1 has a greater correlation, 0.72 < R < 0.92, with the predict effect according to motif contributions to half life than the effect of inserting motifs not native to tRPS3 and tTSA1, 0.47 < R < 0.74.

Promoter choice influences the magnitude of a motifs contribution to gene expression.
For the ATATTC, TGTACAATA and TTTCATTTC motifs the greatest reduction in RNA abundance occurred when native promoter-terminator pairings are measured.
This is true for all three motifs across all three host terminators, except for TTTCATTTC in pRPS3-tRPS3 constructs.
In addition, native promoter-terminator paired constructs exhibit greater than expected contributions to gene expression when compared to predicted effects inferred from motif contributions to half-life. 



A linear model to predict the $\Delta Cq$ values from the qPCR using only motif presence was trained on each promoter-terminator construct library separately.
The inferred coefficients for each of the motifs, together with an additional interactive term for the linear combination of TTTCATTTC and TGTACAATA, in all of the promoter-terminator pairs are shown in Figure  \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)B.
The significance of terminator context on motif effect can be seen in both ATATTCA and TGTACAATA constructs.
ATATTCA is only effective in RPS3 terminators, whichever promoter is chosen, meanwhile TGTACAATA is only effective in the TSA1 terminator, irrespective of promoter.
The significance of promoter choice is shown with the stability motif GTATACCTA which only appear to imbue an increase in RNA abundance for the weakest promoter SRO9.
In addition, the implications of inserting two different motifs into the terminator also changes with promoter selection.
For the PGK1 promoter, the combination of TGTACAATA and TTTCATTTC together either has no added effect, or in fact a reduce efficiency, when compare to the addition of the reductions in abundance when each motif is present separately.
However with native or SRO9 promoter pairing the combined effect is actually greater than the sum of their individual effects.

```{r hlife-predict-vs-abundance-and-motif-context-dependence, fig.cap="\\textbf{Promoter and terminator context alter the regulatory behaviour of shortlisted motifs} (\\textbf{A}) Predicted transcript abundance inferred from motif contributions to half life vs RT-qPCR measured transcript abundance. (\\textbf{B}) Motif contributions to fold changes in RNA abundance for different promoter and terminator contexts. The fold change is relative to the mod0 construct in each promoter-terminator pairing for the tRPS3 and tTSA1 sets. For the tPIR1 sets the fold change is relative to the WT constructs. Confidence intervals are within a standard error.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/qPCR_model_coef_and_pred_vs_exp_abund.png"))
```

## Motif effects on transcript expression is repeated in protein expression data

The effects of motif presence on protein expression was investigate with a platereader assay detecting the fluorescence of the mCherry reporter gene on the pRPS3-tRPS3 and pTSA1-tTSA1 constructs.
The right column in Figure \@ref(fig:protein-verse-RNA-plot) confirms that motif effects can be detected in protein expression expression alone.
Comparing the corresponding RNA abundance levels, left column of Figure \@ref(fig:protein-verse-RNA-plot), the same effects of the motifs on transcript abundance are seen in protein expression, together with the same context dependence. 

```{r protein-verse-RNA-plot, fig.cap="\\textbf{Motif contributions to protein abundance correlate with RNA abundance}. (\\textbf{Left}) Fold changes in mRNA abundance for all terminator constructs in the pRPS3-tRPS3 and pTSA1-tTSA1 pairings. (\\textbf{Right}) mCherry fluorescence at the maximum growth rate for all terminator constructs in the pRPS3-tRPS3 and pTSA1-tTSA1 pairings.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/combined_TSA1_RPSA_protein_RNA_plot.png"))
```

## Investigating the effects of cis-regulatory motifs in the 3'UTR using 5PSeq
We used high-throughput 5PSeq (HT-5Pseq) technique to investigate whether modifications of 3’ UTRs influence the process of 5’-3’ co-translational degradation of mRNA, which can also serve as a proxy of translation process (Zhang & Pelechano 2020), and modify ribosome stalling in the 3’-end region of coding sequence.
The 5PSeq method allows to investigate the process of  5’-3’ degradation of mRNA by capturing only de-capped 5’ ends of mRNA molecules, and at the same time – the polyadenylated 3’ ends.
We generated sequencing reads with insert sizes up to 600 bp, which included entire 3’UTRs and up to 400 bp of coding sequence.  

The distributions of 5’ ends did not differ significantly between the samples.
There were major differences between distributions of 3’ ends between different constructs corresponding to different polyadenylation sites (figure \@ref(fig:alt-polyA-5PSeq)).
In all modified 3’ UTRs we observed additional poly(A) sites at 44 nt and 56 nt from STOP codon, which corresponds to 35 nt and 47 nt in WT 3’ UTR.
Secondly, we observed differences in the distribution of poly(A) sites between modified 3’UTRs with different motifs inserted.
Poly(A) sites located 44 nt and 56 nt from STOP codon are more abundant in tRPS3-mod0 compared to tRPS3-modA, with tRPS3-modB between them.
Poly(A) site at 67 nt is pronounced only in tRPS3-mod0 and at 74 nt is especially visible in tRPS3-modA and at 80 nt in tRPS3-modB.
The differences could be caused either by new polyadenylation signals introduced into the 3’ UTR or by increased stability of these isoforms conferred by addition of new motifs.  

```{r alt-polyA-5PSeq, fig.cap="\\textbf{Inserting motifs into native RPS3 termintors may cause alternative polyadenylation.} Relative polyadenylation site usage of transcripts signaled for decay for pRPS3-tRPS3 constructs, acquired using using 5PSeq.", out.width='80%', fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/alternative_polyadenylation_RPS3_5PSeq.png"))
```



## Comparison of whole-genome ATATTC motif effect vs. reporter-gene ATATTC motif effect
- Comparison of our qPCR data of modB with that of Cheng et al. (2017)
