# Results

```{r}
library(here)
knitr::opts_chunk$set(fig.pos = "p", dpi = 300)
```

## Terminator effects on gene expression depend on cis-regulatory context

To investigate the context dependence of regulatory regions, we created a library of 120 constructs containing all combinations of 6 promoters (including 5'UTR), 2 coding sequences, and 10 terminators (including 3'UTR).
We selected promoters and terminators from native yeast genes spanning a variety of different expression patterns: the highly-expressed glycolytic enzyme PGK1, two ribosomal proteins RPS3 and RPS13, RNA-binding proteins PAB1 and SRO9, heat shock protein HSP26, cell cyclin CLN2, cell wall proteins SUN4 and TOS6, and plasma membrane ATPase PMA1.
We selected coding sequences (CDS) expressing mCherry or mTurquoise2 fluorescent proteins, which are bright fluorophores with only 30% amino acid identity [@Shaner2004, @Goedhart2012].
We assembled all constructs in a centromeric plasmid using the YeastFab system [@Garcia2018], and expressed them in BY4741 yeast. 

Measuring fluorescence with a plate reader showed that, as expected, promoter choice dominated overall protein output.
We observed 100-fold changes in fluorescence between the four highest expressing promoters, Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)A, and the two lowest expressing promoters, Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)B, across both coding sequences.
Expression from the stress-induced pHSP26 was notably more variable than from pPGK1 and pRPS's, across 6 biological replicates, when combined with both coding sequences and a variety of terminators.

To investigate the correlation of mRNA abundance and protein output, we checked a subset of mCherry constructs using RT-qPCR. 
This confirms that most differences in protein outputs are already present in mRNA abundance, Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)C.

Terminators also affect protein output, seen by 5-fold changes in fluorescence within the same promoter-CDS sets, normalised to the tPGK1 terminator, in Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)A.
pSRO9 and pCLN2 were removed due to the poor signal to noise ratio at low fluorescence levels.
Terminators that have consistently low expression include tSUN4 and tTOS6, taken from cell wall proteins whose transcripts are bound and regulated by shared RNA-binding proteins [@Hogan2008].
The effect of terminators depends on coding sequence: tPAB1 is consistently the most highly expressed terminator in mTurq constructs, but not in mCherry constructs.
The effect of terminators also depends on the promoter: tPGK1 is one of the most highly expressing terminators when paired with pPGK1 and pHSP26, but is up to 40% lower in expression when paired with pRPS3 or pRPS13.
Terminators have similar effects when paired with either of the two ribosomal protein promoters pRPS3 or pRPS13, which is consistent with the co-regulation of ribosomal proteins in cells.
Overall, our results show that the contributions of terminators (including 3'UTRs) to gene expression depend on the sequences in other parts of the gene, and are consistent across CREs co-regulated genes.

To further validate our assay, we tested the effect of truncated terminators, because disruption of termination signals lowers expression [@Guo1996,@Shalem2015].
We chose two 200bp terminators from the YeastFab parts library [@Guo2015], tRPS3 and tSRO9.
The median length of the native RPS3 3'UTR is 86nt [@Pelechano2013]; truncating the terminator to 86bp or 59bp reduces protein output by almost 2-fold, Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)B.
The median length of the native SRO9 3'UTR is 543nt [@Pelechano2013]; extending the terminator length to 543bp increases protein output by almost 2-fold, Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-norm)B.
Note that we used the longer 543bp SRO9 terminator, and a similarly extended 421bp tPMA1, for the constructs in Fig \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw); we used 200bp terminator parts for the other modules whose measured  3'UTR length was below 200nt.
This validates our assay's ability to detect known regulatory signals affecting transcription termination, while highlighting the importance of using well-informed annotations to construct parts libraries for synthetic biology.


```{r pro-ter-platereader-mCherry-mTurq-raw, fig.cap="\\textbf{Both terminator and promoter contribute to mCherry and mTurq gene expresion.}  (\\textbf{A}) mCherry and mTurq fluorescence measured at the time of each sample's maximum growth rate for 10 termintators paired with 4 high expressing promoters. Negative fluorescence values arising from instrument noise dominating measurements of constructs with neglegible fluorescence are automatically set to 0. (\\textbf{B}) Same as above but for 2 low expressing promoters. (\\textbf{C}) mCherry fluorescence vs RT-qPCR mRNA abundance for 3 promoters paired with all 10 terminators.", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/pro_ter_swap_protein_and_rna_exp_figure.png"))
```

```{r pro-ter-platereader-mCherry-mTurq-norm, fig.cap="\\textbf{Terminator contributions to gene expresion are promoter and length dependent.} (\\textbf{A}) mTurq and mCherry fluorescence measured at the time of each sample's maximum growth rate for 10 termintators paired with four high expressing promoters. Negative values arising from instrument noise dominating measurements of constructs with neglegible fluorescence are automatically set to 0. (\\textbf{B}) RT-qPCR mRNA abundance results for pRPS3-mCherry-tRPS3 and pSRO9-mCherry-tSRO9 constructs with differing terminator lengths.", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/pro_ter_platereader_norm_mTurq_and_mCh.png"))
```

## Candidate cis-regulatory elements contribute to transcript decay rates

We next investigated how the regulatory effects of short CREs contained within terminator regions depend on their context.
First, 69 suitable CREs to test for context dependence were found through a literature search. 
All were suspected sequence motifs for RNA binding proteins, several directly associated with proteins involved in mRNA degradation [@Hogan2008; @Shalgi2005; @Cheng2017].
Any motifs that were found in fewer than 6 gene 3'UTRs, as annotated by @Pelechano2013, were removed.

We quantified the regulatory effects of the remaining 38 candidate CREs by applying a linear model predicting half life to two recent transcriptome-wide analyses of mRNA decay using metabolic labeling [@Sun2013; @Chan2018].
These two datasets are loosely correlated in their half-life measurements across 4188 genes reported in both datasets, $R = 0.63$ (Fig \@ref(fig:hlife-decay-model)A). However, @Chan2018, which has greater coverage of genes in the yeast genome, 5529 vs 4304,  and used multiple time points to determine half lives, estimating substantially smaller half lives.
Following @Cheng2017, we constructed a linear model to predict a transcript's half life using the counts of motifs in its median length 3'UTR, the 3'UTR length, and the relative codon usage in each transcript's ORF (see Material and Methods). 
This model performed similarly on both datasets by explaining 44% and 41% of the variability in half lives for the @Chan2018 and @Sun2013 data sets respectively, Fig \@ref(fig:hlife-decay-model)C.
This predictive power is comparable to the squared correlation between the datasets ($R^2 = 0.40$).
CREs that did not significantly contribute to the model were automatically filtered out using a greedy algorithm maximising the Akaike information criterion (AIC) during both training stages.
Approximately 1.6% of the variance is explained by these 7 motifs, with 42.0% explained by codon usage, consistent with previous analyses [@Presnyak2015;@Cheng2017].
The top 7 most significant motifs from the @Chan2018 data showed similar regulatory behaviour in the @Sun2013 data, except for TGTAAATA which swap from being stabilising to destabilising, Fig \@ref(fig:hlife-decay-model)B.


We selected four motifs for exploring context dependence: TGTAHMNTA, GTATACCTA, HWNCATTWY, and ATATTC.
TGTAHMNTA and GTATACCTA were chosen as they had the largest coefficients for significant decay and stability motifs, respectively.
HWNCATTWY was chosen due to its statistically significant effect in both data sets and as a candidate for testing motif interactions, as it co-occurs with TGTAHMNTA in 68 native 3'UTRs.
TGTAHMNTA is the binding motif for Puf4p, and HWNCATTWY is associated with Khd1p/Hek2p-bound transcripts [@Hogan2008].
The final selected motif was ATATTC, that is a statistically significant decay motif in both data sets, and that lowers mRNA abundance when inserted in reporter genes [@Cheng2017].
It is not known how ATATTC occurrences lead to mRNA decay.


```{r hlife-decay-model, fig.cap="\\textbf{A linear model predicting transcript half life verifies candidate RBP motifs regulate gene expression.} (\\textbf{A}) Predicted vs actual log2 transcript half lives calculated by a linear model of codon and motif usage trained on the Chan et al (2017) and Sun et al (2013) data sets. (\\textbf{B}) Correlation between the two log2 transcript half lives reported in the Sun et al (2013) and Chan et al (2017) data sets. (\\textbf{C}) Predicted contributons to log2 half life for chosen motifs in the Sun et al (2013) and Chan et al (2017) datasets.", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/hlife_model_multi_fig.png"))
```


```{r chan_motif_coef,include = FALSE}
chan_motif_table <- read.csv(here("./results_chapter/data/chan_motif_coefficients.csv"))
knitr::kable(chan_motif_table)
```

## Quantification of cis-regulatory effects via CRE insertion or mutagenesis in multiple 3'UTRs

We next designed a further set of reporter constructs to quantify the effect of selected short CREs in different contexts (Fig \@ref(fig:tRPS3-tTSA1-design-and-qpcr)A).
Each reporter construct includes a different combination of inserted motifs in one of 2 host terminators, and one of 3 different promoters, driving expression of mCherry.
We first chose the ribosomal protein terminator tRPS3, as it was the only terminator in our initial library that did not contain any of the selected short CREs.
We selected thioredoxin peroxidase tTSA1 as the second host terminator, because it also lacks selected short CREs and has similar length to tRPS3.
In each host terminator we chose 3 motif insertion sites, selecting for: minimum impact on transcript secondary structure, avoiding known termination elements, and matching the positions of motifs in native genes.
Having 3 insertion sites enabled us to quantify combinations of motifs, including duplicates of weaker motifs to increase the likelihood of detecting a clear effect on gene expression. 
We chose TGTACAATA and TTTCATTTC sequences as explicit versions of the TGTAHMNTA and HWNCATTWY consensus motifs respectively, and checked that these explicit versions have similar predicted effect on half-life transcriptome-wide.
Altogether 7 constructs were designed for these two host terminators; the wildtype terminator, a control to test the insertion sites with randomly generated sequences, four constructs testing the effects of each motif individually and a final construct to test interactions between the TGTAHMNTA and HWNCATTWY motifs.
For each terminator, we created constructs paired with each of the native promoter (pRPS3 or pTSA1), the high-expression pPGK1, and the low-expression pSRO9.

Motifs predicted to affect half-life indeed affect mRNA abundance when inserted into tRPS3 in reporter constructs(Fig \@ref(fig:tRPS3-tTSA1-design-and-qpcr)B).
We measured RNA abundance by RT-qPCR across 6 biological replicates, each quantufied in 3 technical replicates and normalised by the $\Delta Cq$ method against values from 3 reference RNAs (see methods - CHECK THIS).
Insertion of two copies of ATATTC into generally lowers the mRNA abundance, as much as 4-fold when paired with the pRPS3 promoter. 
Insertion of either TGTAHMNTA, or 2 copies of HWNCATTWY, tends to decrease RNA abundance, and the combination tends to further decrease RNA abundance.
However, the putative stability motif GTATACCTA does not consistently or strongly affect RNA abundance.
Protein fluorescence results show a loose correlation between mRNA expression and protein expression (Sup Fig ?).
However, comparison of the WT and control (mod_NNN) terminators does show that the creation of the insertion sites alone has an effect on gene expression.

Inserting the same motifs into our second host terminator gives qualitatively similar results \@ref(fig:tRPS3-tTSA1-design-and-qpcr)A).
Decay motifs generally lead to decay, although ATATTC has a weaker effect in tTSA1 than in tRPS3, and TGTAHMNTA has a stronger effect in tTSA1 than tRPS3.
The putative stability motif GTATACCTA again has little effect.

We next quantified the effects of removing decay motifs from from a native yeast terminator. 
We selected the cell wall protein PIR1 as our host terminator, due to the presence of ATATTC, TGTAHMNTA, and HWNCATTWY motifs, within a short and de-novo-synthesizable terminator.
We designed 8 terminators in which the motif occurrences in tPIR1 were replaced by scrambled sequences, (Fig \@ref(fig:tPIR1-design-and-qpcr)A).
We find that the removal of almost any decay motif from tPIR1 results in an increase in mRNA levels (\@ref(fig:tPIR1-design-and-qpcr)B)).


Comparison of mRNA abundance across all constructs, (Figs \@ref(fig:tRPS3-tTSA1-design-and-qpcr)B, \@ref(fig:tRPS3-tTSA1-design-and-qpcr)C and \@ref(fig:tPIR1-design-and-qpcr)B), shows motif contributions change in magnitude but not direction depending on the context of the rest of the construct. 
The insertion of almost any decay motif into tTSA1 or tRPS3 results in a decrease in mRNA abundance, and removal of these from from tPIR1 results in an increase in mRNA abundance.
However, the quantitative effects vary depending both on the immediate motif context in the host terminator and on the more distant context given by the promoter.

```{r tRPS3-tTSA1-design-and-qpcr, fig.cap="\\textbf{Shortlisted motifs inserted into RPS3 and TSA1 host terminators change transcript abundance in RT-qPCR measurements}. (\\textbf{A}) Library design and diagram of motif insertion sites in native RPS3 and TSA1 terminators. (\\textbf{B}) Fold changes in transcript abundance for tRPS3 constructs paired with three promoters; pRPS3, pPGK1 and pSRO9. (\\textbf{C}) Fold changes in transcript abundance for tTSA1 constructs paired with three promoters; pTSA1, pPGK1 and pSRO9. Fold changes are relative to the abundance of the mod0 construct", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/insertion_constructs_design_and_qpcr.png"))
```


```{r tPIR1-design-and-qpcr, fig.cap="\\textbf{Shortlisted motifs removed from PIR1 host terminators change transcript abundance in RT-qPCR measurements}. (\\textbf{A}) Diagram of the final PIR1 constructs with permutations of removed motifs. (\\textbf{B}) Fold changes in transcript abundance for tPIR1 constructs paired with three promoters; pPIR1, pPGK1 and pSRO9. Fold changes are relative to the abundance of the WT construct", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/tPIR1_design_and_qpcr.png"))
```


## Motif effects on gene expression depend both on terminator context and promoter pairing

We next compared our predictive model of the effects of short CREs -- inferred from transcriptome-wide measurements of half-life -- and their effects when inserted or removed from reporter constructs.
A simple model of transcript production and decay argues that changes in RNA half-life are directly proportional to changes in RNA abundance (see methods).
So, for each promoter and host terminator combination, we trained a linear model to predict the log2 RNA abundance (i.e. $\Delta Cq$) values from the RT-qPCR using only motif presence, and compared this with the log2 half-life effect of the motif from our transcriptome-wide analysis of @Chan2018.
Figure \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)A shows that the effect of short CREs in reporter constructs is correlated with the predictive model, but that the strength of the correlation depends on context.
Constructs inserting motifs into terminators had a lower correlation with predicted effects than the construct where we remove motifs from a native terminator.
Interestingly, motifs appear to have consistently greater than predicted effects on mRNA abundance when their host terminator is paired with its native promoter. 


We next compared the inferred coefficients for individual motifs in each promoter-host terminator pair (Fig \@ref(fig:hlife-predict-vs-abundance-and-motif-context-dependence)B).
The effect of a motif depends on terminator context, for example, ATATTCA reduces RNA abundance substantially more when inserted in tRPS3 than in tTSA1. 
Meanwhile, TGTACAATA significantly reduces RNA abundance inserted in tTSA1, but not tRPS3, whichever promoter is chosen.
Promoter choice influences the magnitude of a motifs contribution to gene expression.
For the ATATTC, TGTACAATA and TTTCATTTC motifs the greatest reduction in RNA abundance occurred when native promoter-terminator pairings are measured.
This is true for all three motifs across all three host terminators, except for TTTCATTTC in pRPS3-tRPS3 constructs.

Regulatory interactions between two different motifs inserted into a terminator also change depending on both host terminator and promoter context.
We included an interaction term that quantifies how the effect of including both TTTCATTTC and TGTACAATA differs from the sum of the effects of including these motifs individually.
The combination of TGTACAATA and TTTCATTTC in tRPS3 has no additional effect when paired with pPGK1, but is significantly less stable than expected when paired with pRPS3 or pSRO9.
The combination of TGTACAATA and TTTCATTTC in tTSA1 is significantly more stable than expected, but only when paired with pPGK1.


```{r hlife-predict-vs-abundance-and-motif-context-dependence, fig.cap="\\textbf{Promoter and terminator context alter the regulatory behaviour of shortlisted motifs} (\\textbf{A}) Predicted transcript abundance inferred from motif contributions to half life vs RT-qPCR measured transcript abundance. (\\textbf{B}) Motif contributions to fold changes in RNA abundance for different promoter and terminator contexts. The fold change is relative to the mod0 construct in each promoter-terminator pairing for the tRPS3 and tTSA1 sets. For the tPIR1 sets the fold change is relative to the WT constructs. Confidence intervals are within a standard error.", fig.align='center'}
knitr::include_graphics(here("results_chapter/figures/qPCR_model_coef_and_pred_vs_exp_abund.png"))
```


