# Supplementary Data
\makeatletter
\renewcommand{\fnum@figure}{Figure S\thefigure}
\renewcommand{\fnum@table}{Table S\thetable}
\makeatother
\setcounter{figure}{0}
\setcounter{table}{0}


```{r hlife-variance-table}
knitr::kable(tibble(CRE = c("Codon Usage", "Motifs", "3'UTR Length"), Individual = c("42.0%", "3.2%", "0.6%"), Drop = c("-40.5%", "-1.6%", "0.0%"), Cumulative = c("42.0%", "43.7%", "43.7%")), caption = "\\textbf{Variance explained by each type of CRE in the half life model applied to data from Chan et al.}. Variance explained was estimated in three different ways. The variance explained by a linear containing each single type of CRE.  The loss in variance explained when one type CRE is removed from the full model. The total variance explained when each CRE is added to the linear model in sequence.", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```


```{r protein-vs-RNA-plot-motifs, fig.cap="\\textbf{Relative protein abundance correlates with relative mRNA abundance for reporter constructs.}. (\\textbf{A}) Fold changes in mRNA abundance vs fold changes in protein fluorescence for all terminator constructs in the pRPS3-tRPS3 and pTSA1-tTSA1 pairings. Transcript abundance is relative to the mod0 construct of each promoter-terminator pairing. (\\textbf{B}) Fold changes in mRNA abundance vs fold changes in protein fluorescence for all terminator constructs in the pPIR1-tPIR1 pairing. Transcript abundance is relative to WT construct of the promoter-terminator pairing. The mean and standard error calculated over 6 biological replicates is plotted for each construct.", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/RPS3_TSA1_PIR1_protein_and_RNA_plot.png"), dpi = 300) 
```

```{r TGTACAATA-motif-coef}
read.csv(here("./supplementary_data_chapter/data/HWNCAUUWY_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`) %>%
knitr::kable(format = "latex", caption="\\textbf{Selecting the HWNCAUUWY motif, TTTCATTTC}. Summary of the number of occurences and contributions to a linear model predecting half life for each possible version of the HWNCAUUWY motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```



```{r TTTCATTTC-motif-coef}
read.csv(here("./supplementary_data_chapter/data/UGUAHMNUA_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`)  %>%
knitr::kable(format = "latex", caption = "\\textbf{Selecting the UGUAHMNUA motif, TGTACAATA}. Summary of the number of occurences and contributions to a linear model predecting half life for each possible version of the UGUAHMNUA motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```


```{r secondary-structure-RPS3, fig.cap="\\textbf{Designs of modified RPS3 3'UTRs aimed at minimising disruption to the secondary structure.} The computed minimum fre energy (MFE) and the corresponding structure result using RNAfold. The modified 3'UTRs were designed, taking into consideration the MFE, and tailoring designs that would result in minimal change to the MFE.", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_rps3_small.png"), dpi = 300)
```

```{r secondary-structure-TSA1, fig.cap="\\textbf{Designs of modified TSA1 3'UTRs aimed at minimising disruption to the secondary structure.}", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_tsa1_small.png"), dpi = 300)
```

```{r secondary-structure-PIR1, fig.cap="\\textbf{Designs of modified PIR1 3'UTRs aimed at minimsing disruption to the secondary structure.}", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_pir1_small.png"), dpi = 300)
```