\onecolumn

# Supplementary Data

\makeatletter
\renewcommand{\fnum@figure}{Supplementary Figure \thefigure}
\renewcommand{\fnum@table}{Supplementary Table \thetable}
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
\makeatother
\setcounter{figure}{0}
\setcounter{table}{0}
\setcounter{page}{1}

\rhead{\scshape Composability in mRNA - Supplement - \today}


```{r}
library(here)
library(tidyr)
knitr::opts_chunk$set(fig.pos = "p", dpi = 300)
```

## Supplementary Tables

```{r hlife-variance-table}
knitr::kable(tibble(CRE = c("Codon Usage", "Motifs", "3'UTR Length"), Individual = c("42.0%", "3.2%", "0.6%"), Drop = c("-40.5%", "-1.6%", "0.0%"), Cumulative = c("42.0%", "43.7%", "43.7%")), caption = "\\textbf{Variance explained by each type of CRE in the half-life model applied to data from Chan et al (2018).} Variance explained was estimated in 3 different ways. The individual variance explained by a linear model containing each type of CRE on its own.  The drop in variance explained when one type of CRE is removed from the full model. The cumulative variance explained when each CRE is added to the linear model in sequence; starting with codon usage, then adding motifs and finally 3'UTR length.", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position",font_size = 7) %>%
  kableExtra::row_spec(0, bold = TRUE, font_size = 7)
```


\arrayrulecolor{white}

```{r mCherry-seq}
knitr::kable(tibble("mCherry construct sequences are available on the manuscript github repo." = "https://github.com/DimmestP/chimera_project_manuscript/ supplementary_data_chapter/data/ mCherry_fluorescence_reporter_dna_sequences.csv"), caption = "\\textbf{Table showing the DNA sequences for all mCherry reporter constructs.}", format = "latex", align = "c")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7) %>%
  kableExtra::column_spec(1, width="30em")

```


```{r mTurq-seq}
knitr::kable(tibble("mTurq construct sequences are available on the manuscript github repo" = "https://github.com/DimmestP/chimera_project_manuscript/ supplementary_data_chapter/data/ mTurq_fluorescence_reporter_dna_sequences.csv"), caption = "\\textbf{Table showing the DNA sequences for all mTurq reporter constructs.}", format = "latex", align = "c")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7) %>%
  kableExtra::column_spec(1, width="30em")
```

\arrayrulecolor{black}

```{r chimera-qpcr-primers-table}
qpcr_primers <- readr::read_csv(here::here("./supplementary_data_chapter/data/chimera_qpcr_primers.csv"))
knitr::kable(qpcr_primers, caption = "\\textbf{Primer sequences created for all qPCR experiments.}", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7) %>%
  kableExtra::row_spec(0, bold = TRUE, font_size = 7) %>%
  kableExtra::column_spec(1, width = "4em") %>%
  kableExtra::column_spec(3, width = "8em")
```

```{r HWNCAUUWY-motif-coef}
read.csv(here("./supplementary_data_chapter/data/HWNCAUUWY_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`) %>%
knitr::kable(format = "latex", caption="\\textbf{Selecting the HWNCAUUWY motif, TTTCATTTC}. Summary of the number of occurences and contributions to a linear model predecting half-life for each possible version of the HWNCAUUWY motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7) %>%
  kableExtra::row_spec(0, bold = TRUE, font_size = 7)
```

```{r TGTAHMNTA-motif-coef}
read.csv(here("./supplementary_data_chapter/data/UGUAHMNUA_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`)  %>%
knitr::kable(format = "latex", caption = "\\textbf{Selecting the UGUAHMNUA motif, TGTACAATA}. Summary of the number of occurences and contributions to a linear model predecting half-life for each possible version of the UGUAHMNUA motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7)  %>%
  kableExtra::row_spec(0, bold = TRUE, font_size = 7)
```

```{r minimum-free-energy-table}
minimum_free_energy <- readr::read_csv(here::here("./supplementary_data_chapter/data/terminator_mfe.csv"))
knitr::kable(minimum_free_energy, caption = "\\textbf{Table showing minimum free energies of 3'UTR constructs with inserted/deleted motifs.}", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position", font_size = 7)  %>%
  kableExtra::row_spec(0, bold = TRUE, font_size = 7) %>%
  kableExtra::column_spec(3, width = "8em")
```

\pagebreak

## Supplementary Figures

```{r protein-vs-RNA-plot-motifs, fig.cap="\\textbf{Relative protein abundance correlates with relative mRNA abundance for reporter constructs with modified 3'UTRs.} (\\textbf{A}) Fold changes in RT-qPCR vs fold changes in mCherry fluorescence for all terminator constructs in the pRPS3-mCherry-tRPS3 and pTSA1-mCherry-tTSA1 pairings. Transcript abundance is relative to the mod0 construct of each promoter-terminator pairing. (\\textbf{B}) Fold changes in RT-qPCR vs fold changes in mCherry fluorescence for all terminator constructs in the pPIR1-mCherry-tPIR1 pairing. Transcript abundance is relative to WT construct of the promoter-terminator pairing. The mean and standard error calculated over 6 biological replicates is plotted for each construct.", fig.align='center' , out.width="98%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/RPS3_TSA1_PIR1_protein_and_RNA_plot.png"), dpi = 300) 
```

```{r motif-position-histograms, fig.cap="\\textbf{Relative positions of ATATTC, TGTAHMNUA and HWNCAUUWY motifs in native 3'UTRs.} Histogram shows the counts of the motif occurrences in 3'UTRs relative to the total length of 3'UTR, where 0 would be starting exactly at the stop codon and 1 would be at the reported poly(A)-site. See (methods or ref to data repository) for details.", fig.align='center', out.width="98%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/motif_position_histograms.png"), dpi = 300) 
```


```{r polyA-site-usage-5Pseq, fig.cap="\\textbf{Decapped constructs measured by 5Pseq match mature constructs measured by QuantSeq.} pRPS3-tRPS3 constructs were chosen to investigate changes in poly(A) site usage across 5' decapped transcripts flagged for decay. (\\textbf{A}) Comparison of construct transcript abundance as independently measured by qPCR and RNA-Seq assays. Transcript abundance was normalised to the median abundance of  plasmid URA3, genomic PGK1 and RPS3 transcripts for each construct. Fold change is relative to the mod\\_NNN construct in each promoter-terminator context. The black diagonal line represents the expected values if RNAseq and qPCR results correlated perfectly. (\\textbf{B}) Cumulative counts of reads mapped downstream of WT (grey) and mod\\_NNN (golden) construct stop codons as fraction of total reads mapped to the constructs terminator. WT reads have been shifted downstream to align with the mod\\_NNN sequence by accounting for motif insertion sites. Major poly(A) sites have been highlighted by a black vertical line. Constructs also used in the QuantSeq analysis have their QuantSeq cumulative graphs plotted in dotted lines. (\\textbf{C}) Similar to Figure \\textbf{B} but with each motif insertion construct plotted separately. Columns designate cumulative plots from different terminator constructs.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/polya_usage_plot_5PSeq.png"))
```

```{r rnaseq-QC-correlation, fig.cap="\\textbf{High correlation in transcript counts between samples for both RNA-Seq assays.} (\\textbf{A}) Correlation heat map of DESeq2 normalised log2 pseudocounts across all genes for each sample pair in the QuantSeq assay. From top to bottom on the y axis (left to right on the x-axis) the terminator contexts are pRPS3-tRPS3, pPGK1-tRPS3 and pTSA1-tTSA1. (\\textbf{B}) Similar correlation heat map for all sample pairs in the 5PSeq assay. (\\textbf{C}) Comparison of mean log2 abundance of construct mRNA transcripts as measured by 5PSeq and QuantSeq. Only 5 terminator constructs were measured by both methods and only in the pRPS3-tRPS3 context. Transcript abundance for each sample is normalised to the median of the genomic PGK1, TSA1 and RPS3, and the plasmid URA3 genes to match the normalisation used in the qPCR analysis.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/sequencing_sample_correlation.png"))
```

```{r rnaseq-QC-QuantSeq-and-5PSeq-genomic-polyA, fig.cap="\\textbf{Poly(A) site usage for genomic PGK1, TSA1 and RPS3 terminators remains the same across samples for each RNA-Seq assay.} Cumulative counts of reads mapped downstream of native genomic gene stop codons as fraction of total reads mapped to the terminator. Relative usage of poly(A) sites remain similar across samples and across RNA-seq assays. Higher variability in QuantSeq PGK1 poly(A) site usage is due to low overall transcript counts.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/polya_QC_plot_QuantSeq_and_5PSeq.png"))
```

```{r rnaseq-QC-genomic-vs-plasmid-polyA, fig.cap="\\textbf{Poly(A) site usage remains the same for genomic TSA1 and RPS3 terminators as for plasmid expressed WT constructs in QuantSeq and 5PSeq.} Cumulative counts of reads mapped downstream of WT stop codons for native genomic terminators and plasmid construct terminators as a fraction of total reads mapped to the constructs terminator. Expressing constructs from a low copy number plasmid does not seem to affect poly(A) site usage.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/plasmid_vs_genome_polyA_combined_plot.png"))
```

```{r relative-polya-counts, fig.cap="\\textbf{Construct poly(A) site usage across 5PSeq and QuantSeq.} Relative counts of reads mapped downstream of construct stop codons as fraction of total reads mapped to the constructs terminator. Peaks represent the position of major poly(A) sites. Replicates are plotted on top of each other where available.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("./supplementary_data_chapter/figures/relative_polya_counts_QuantSeq_and_5PSeq.png"))
```

```{r combined-5p-end-read-plot, fig.cap="\\textbf{5PSeq data finds no detectable changes in 5'-phosphorylated intermediates between reporter constructs.} Relative counts of 5' end reads of 5'-phosphorylated intermediates as fraction of total 5' end reads mapped to the terminator. Results for plasmid constructs are plotted alongside selected genomic genes for comparison.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/combined_5p_end_read_plot.png"))
```

