# Supplementary Data
\makeatletter
\renewcommand{\fnum@figure}{Figure S\thefigure}
\makeatother
\setcounter{figure}{0}

## Variance explained by each predicate in the half life model 


```{r hlife-variance-table}
knitr::kable(tibble(CRE = c("Codon Usage", "Motifs", "3'UTR Length"), Individual = c("42.0%", "3.2%", "0.6%"), Drop = c("-40.5%", "-1.6%", "0.0%"), Cumulative = c("42.0%", "43.7%", "43.7%")), caption = "
Variance explained was explored in three different ways. The variance explained by a linear containing each CRE appearing individually.  The loss in variance explained when one CRE is removed from the full model. The total variance explained when each CRE is added to the linear model in sequence.")
```

## Motif effects on transcript expression is repeated in protein expression data

The effects of motif presence on protein expression was investigate with a platereader assay detecting the fluorescence of the mCherry reporter gene on the pRPS3-tRPS3 and pTSA1-tTSA1 constructs.
The right column in Figure \@ref(fig:protein-verse-RNA-plot) confirms that motif effects can be detected in protein expression expression alone.
Comparing the corresponding RNA abundance levels, left column of Figure \@ref(fig:protein-verse-RNA-plot), the same effects of the motifs on transcript abundance are seen in protein expression, together with the same context dependence. 

```{r protein-verse-RNA-plot, fig.cap="\\textbf{Motif contributions to protein abundance correlate with RNA abundance}. (\\textbf{Left}) Fold changes in mRNA abundance for all terminator constructs in the pRPS3-tRPS3 and pTSA1-tTSA1 pairings. (\\textbf{Right}) mCherry fluorescence at the maximum growth rate for all terminator constructs in the pRPS3-tRPS3 and pTSA1-tTSA1 pairings.", out.width='60%', fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/RPS3_TSA1_PIR1_protein_and_RNA_plot.png"))
```

## Selecting the HWNCAUUWY motif, TTTCATTTC

```{r TGTACAATA_motif_coef}
TGTACAATA_motif_table <- read.csv(here("./supplementary_data_chapter/data/HWNCAUUWY_shortlisted_motif_table.csv")) %>%
  dplyr::rename("Motif" = "term",  "Total Count" = "total_count")
knitr::kable(TGTACAATA_motif_table)
```

## Selecting the UGUAHMNUA motif, TGTACAAUA

```{r TTTCATTTC_motif_coef}
TTTCATTTC_motif_table <- read.csv(here("./supplementary_data_chapter/data/UGUAHMNUA_shortlisted_motif_table.csv")) %>%
  dplyr::rename("Motif" = "term",  "Total Count" = "total_count")
knitr::kable(TTTCATTTC_motif_table)
```


## Comparison of whole-genome ATATTC motif effect vs. reporter-gene ATATTC motif effect
- Comparison of our qPCR data of modB with that of Cheng et al. (2017)

## Secondary structure prediction of RPS3 and PIR1 3'UTR native and modified designs
```{r secondary-structure-RPS3, fig.cap="\\textbf{Designs of modified RPS3 3'UTRs aimed at minimsing disruption to the secondary structure.} The computed minimum fre energy (MFE) and the corresponding structure result using RNAfold. The modified 3'UTRs were designed, taking into consideration the MFE, and tailoring designs that would result in minimal change to the MFE.", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_rps3_small.png"), dpi = 300)
```

```{r secondary-structure-TSA1, fig.cap="\\textbf{Designs of modified RPS3 3'UTRs aimed at minimsing disruption to the secondary structure.}", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_tsa1_small.png"), dpi = 300)
```

```{r secondary-structure-PIR1, fig.cap="\\textbf{Designs of modified RPS3 3'UTRs aimed at minimsing disruption to the secondary structure.}", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/secondary_structures_pir1_small.png"), dpi = 300)
```