# Supplementary Data
\makeatletter
\renewcommand{\fnum@figure}{Supplementary Figure \thefigure}
\renewcommand{\fnum@table}{Supplementary Table \thetable}
\makeatother
\setcounter{figure}{0}
\setcounter{table}{0}


```{r}
library(here)
library(tidyr)
knitr::opts_chunk$set(fig.pos = "p", dpi = 300)
```


```{r hlife-variance-table}
knitr::kable(tibble(CRE = c("Codon Usage", "Motifs", "3'UTR Length"), Individual = c("42.0%", "3.2%", "0.6%"), Drop = c("-40.5%", "-1.6%", "0.0%"), Cumulative = c("42.0%", "43.7%", "43.7%")), caption = "\\textbf{Variance explained by each type of CRE in the half-life model applied to data from Chan et al (2018).} Variance explained was estimated in 3 different ways. The individual variance explained by a linear model containing each type of CRE on its own.  The drop in variance explained when one type of CRE is removed from the full model. The cumulative variance explained when each CRE is added to the linear model in sequence; starting with codon usage, then adding motifs and finally 3'UTR length.", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```


```{r protein-vs-RNA-plot-motifs, fig.cap="\\textbf{Relative protein abundance correlates with relative mRNA abundance for reporter constructs with modified 3'UTRs.} (\\textbf{A}) Fold changes in RT-qPCR vs fold changes in mCherry fluorescence for all terminator constructs in the pRPS3-mCherry-tRPS3 and pTSA1-mCherry-tTSA1 pairings. Transcript abundance is relative to the mod0 construct of each promoter-terminator pairing. (\\textbf{B}) Fold changes in RT-qPCR vs fold changes in mCherry fluorescence for all terminator constructs in the pPIR1-mCherry-tPIR1 pairing. Transcript abundance is relative to WT construct of the promoter-terminator pairing. The mean and standard error calculated over 6 biological replicates is plotted for each construct.", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/RPS3_TSA1_PIR1_protein_and_RNA_plot.png"), dpi = 300) 
```

```{r motif-position-histograms, fig.cap="\\textbf{Relative positions of ATATTC, TGTAHMNUA and HWNCAUUWY motifs in native 3'UTRs.}", fig.align='center'}
knitr::include_graphics(here("supplementary_data_chapter/figures/motif_position_histograms.png"), dpi = 300) 
```
\arrayrulecolor{white}
```{r mCherry-seq}
knitr::kable(tibble("mCherry construct sequences are available on the manuscript github repo." = "https://github.com/DimmestP/chimera_project_manuscript/supplementary_data_chapter/data/ mCherry_fluorescence_reporter_dna_sequences.csv"), caption = "\\textbf{Table showing the DNA sequences for all mCherry reporter constructs.}", format = "latex", align = "c")  %>%
  kableExtra::kable_styling(latex_options = "hold_position") %>%
  kableExtra::column_spec(1, width="40em")

```


```{r mTurq-seq}
knitr::kable(tibble("mTurq construct sequences are available on the manuscript github repo" = "https://github.com/DimmestP/chimera_project_manuscript/supplementary_data_chapter/data/ mTurq_fluorescence_reporter_dna_sequences.csv"), caption = "\\textbf{Table showing the DNA sequences for all mTurq reporter constructs.}", format = "latex", align = "c")  %>%
  kableExtra::kable_styling(latex_options = "hold_position") %>%
  kableExtra::column_spec(1, width="40em")
```
\arrayrulecolor{black}
```{r chimera-qpcr-primers-table}
qpcr_primers <- readr::read_csv(here::here("./supplementary_data_chapter/data/chimera_qpcr_primers.csv"))
knitr::kable(qpcr_primers, caption = "\\textbf{Primer sequences created for all qPCR experiments.}", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```

```{r HWNCAUUWY-motif-coef}
read.csv(here("./supplementary_data_chapter/data/HWNCAUUWY_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`) %>%
knitr::kable(format = "latex", caption="\\textbf{Selecting the HWNCAUUWY motif, TTTCATTTC}. Summary of the number of occurences and contributions to a linear model predecting half-life for each possible version of the HWNCAUUWY motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```



```{r TGTAHMNTA-motif-coef}
read.csv(here("./supplementary_data_chapter/data/UGUAHMNUA_shortlisted_motif_table.csv")) %>%
  dplyr::transmute(Motif = term,  `Total Count` = total_count, Coef = estimate, `p.value`)  %>%
knitr::kable(format = "latex", caption = "\\textbf{Selecting the UGUAHMNUA motif, TGTACAATA}. Summary of the number of occurences and contributions to a linear model predecting half-life for each possible version of the UGUAHMNUA motif.")  %>%
  kableExtra::kable_styling(latex_options = "hold_position") 
```

```{r minimum-free-energy-table}
minimum_free_energy <- readr::read_csv(here::here("./supplementary_data_chapter/data/terminator_mfe.csv"))
knitr::kable(minimum_free_energy, caption = "\\textbf{Table showing minimum free energies of 3'UTR constructs with inserted/deleted motifs.}", format = "latex")  %>%
  kableExtra::kable_styling(latex_options = "hold_position")
```

```{r polyA-site-usage-5Pseq, fig.cap="\\textbf{5' decapped RPS3 constructs match QuantSeq results.} pRPS3-tRPS3 constructs were chosen to investigate changes in polyA site usage across 5' decapped transcripts flagged for decay. (\\textbf{A}) Comparison of construct transcript abundance as independently measured by qPCR and RNA-Seq assays. Transcript abundance was normalised to the median abundance of  plasmid URA3 and genomic PGK1, RPS3 and/or TSA1 transcripts for each construct. Fold change is relative to the mod\\_NNN construct in each promoter-terminator context. (\\textbf{B}) Cumulative count of reads mapped downstream of WT and mod\\_NNN construct stop codons as fraction of total reads mapped to the constructs terminator. WT reads have been shifted downstream to align with the mod\\_NNN sequence by accounting for motif insertion sites. Suspected polyA sites have been highlighted by a black vertical line. Constructs also used in the QuantSeq analysis have their QuantSeq cumulative graphs plotted in dotted lines. (\\textbf{C}) Similar to Figure \\textbf{B} but with each motif insertion construct plotted separately. Columns designate cumulative plots from different terminator constructs.", fig.align='center',out.width="85%"}
knitr::include_graphics(here("supplementary_data_chapter/figures/polya_usage_plot_5PSeq.png"))
```
