# Discussion


We have confirmed that promoter-terminator pairings change the effectiveness of each region's contribution to gene expression, as previously suggested. This means the regions are not entirely modular.

We explore whether the changes in regulatory behavior according to overall context is explained by changes in the regulatory effects of individual cis-regulatory elements within these regions. We have shown that 3'UTR sequence motifs, confirmed to have some regulatory role in changing decay rates, have effects on gene expression that a dependent on overall terminator context and promoter pairing. 

This has impacts for motif searching methods that rely on modularity to simplify the search space. Could we find regular characteristics of promoters that determine which motifs will be most effective? 

Questions remain, are the motifs behaviors in RPS3, TSA1 and PIR1 representative of their behavior across the genome? Can a linear model account for the depedences using more comprehensive interaction terms?


Several groups have sought to elucidate the effect of different native yeast terminators with a goal of maximal protein expression. Yamanishi et al. identified a terminator (tTPS1) which increased fluorescent protein production by 50% compared to the tCYC1 terminator (Yamanishi et al. 2011). Ito et al. identified tDIT1 was associated with a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3â€™UTR (Ito et al. 2016). The same group further identified that the effect of a library of 5302 terminators spanned an 81-fold range difference in fluorescence production, with several resulting in a doubling of protein production relative to tPGK1 (Yamanishi et al. 2013). When coupled with different promoters, the relative difference in protein output between terminators vary, with a maximum difference of 11-fold when coupled with a low expression promoter (Curran et al. 2013, Wei et al. 2017).

