---
output:
  pdf_document: default
  html_document: default
---

# Discussion

Our measurements of protein output from promoter-terminator combinations highlight the limitations of assuming that CREs are quantitatively composable.
Unsurprisingly, fluorescent protein expression levels are dominated by promoter choice, up to 100 fold, and then terminator choice, up to 5 fold. 
However, our observations of 1.5 fold change in the relative effect of terminators depending on coding sequence and promoter choice highlight the quantitative limitations of the assumption of composability.
Previous work by @Kosuri2013 suggests that screening a larger library of CREs would reveal a subset with stronger quantitative interactions.

Terminators taken from co-regulated genes have similar behaviour in our reporter constructs.
Terminators that have consistently low expression include tSUN4 and tTOS6, taken from cell wall proteins whose transcripts are bound and regulated by shared RNA-binding proteins [@Hogan2008].
Terminators have similar effects when paired with either of the two ribosomal protein promoters pRPS3 or pRPS13, which is consistent with the co-regulation of ribosomal proteins in cells.
Further investigations are required into the shared sequence signals and proteins that explain these regulatory patterns and affect the composability of these elements.

We find that the effect of short CREs in 3'UTRs depends on the context of their host construct.
The contribution of our selected cis-regulatory motifs to mRNA stability, as inferred from transcriptome-wide analysis, correlates with their effect on the abundance of reporter constructs.
However, their contributions to mRNA abundance were inconsistent and dependent on context: TGTAHMNTA has no measurable effect when inserted into tRPS3 but the expected effect in tTSA1, ATATTC has the expected effect in tRPS3 but little effect in tTSA1, and HWNCATTWY can either decrease or increase mRNA levels when removed from tPIR1, depending on the promoter.
Also, the two tPIR1 constructs with different mutated HWNCATTWY motifs had different expression levels suggesting HWNCATTWY has a position dependent effect. 
Although, the exact sequences of the two motifs did also differ by 4 nucleotides.
In addition, our original motif search predicted a TGTAAATA motif to be stabilising in the [@Sun2013] dataset but destabilising in the [@Chan2018] dataset.
TGTAAATA is a known binding motif for Puf3 and carbon deficient conditions trigger Puf3 to degrade its targets [@Miller2014].
Carbon deficient conditions were likely present due to the bespoke CSM-lowURA media used by [@Chan2018], which is similar to the SC-URA media used in our assays, contrasting with carbon-rich YPD media used by [@Sun2013].
Thus, carbon-dependent regulation by Puf3 could explain the different behaviour of TGTAAATA in the different datasets.
These disparities in predicted and experimentally verified regulatory behaviour raises questions about current methods for discovering novel cis-regulatory elements and, more generally, for understanding and manipulating regulatory pathways.

RNA-Seq results confirmed our conclusions on the effect of motif insertions on gene expression. 
pRPS3-tRPS3 and pTSA1-tTSA1 constructs show similar relative abundances in the qPCR results as the RNA-Seq results. 
However, pPGK1-tRPS3 results are skewed due to the unexpected low abundance of mod_NNN constructs in the RNA-Seq results, which all other constructs are normalised to.
As the QuantSeq protocol we used for our RNA-Seq assay anchors reads to the polyA tail of transcripts we were able to determine the effect of inserting motifs on polyA site usage. 
For tTSA1 constructs, inserting the motifs shifted the polyA sites 27 nucleotides downstream of their absolute positions in WT, or the exact size of the three insertion site combined.
The sequences and relative usages of the polyA sites are not affected by inserting motifs.
For tRPS3 constructs, polyA sites were again appropriately shifted downstream by the length of the insertions sites.
However, a novel polyA site is introduced between the second and third motifs.
It is known that efficiency elements in the terminators of yeast genes play an important role in efficient RNA transcription and that cleavage and polyadenylation sites occur a specific distance downstream of the efficiency elements [@Graber1999].
Although efforts were made to avoid altering efficiency elements in native terminators, the creation of novel polyA site is likely due to the inserted motifs extending the distance between the efficiency elements and the native polyA sites. 
Motifs 2 and 3 were inserted into a highly conserved region of Rps3 terminator.
The corresponding 3'UTR has been shown to contain a potential binding site for an unknown RBP.
Despite this the QuantSeq results do confirm that at least one copy of the motifs of interest are present in every major isoform for all constructs.

## Mechanisms for motif terminator context dependence

This study documents CRE variability with local context, but CRE behavior is also known to vary temporally and spatially in cells, further supporting that CRE functions are not generally decomposable.
Our results support long-standing mechanistic observations that CRE contributions depend on the presence of other CREs in the host gene.
However, CREs are often regulated by cognate trans-regulatory elements (TREs) that are spatiotemporally regulated themselves.
These effects are not separable, because spatiotemporal regulation of gene expression conferred by one CRE can change the interaction of a second CRE with its relevant TREs.
For example, a motif may act as the regulatory binding site of a TRE only after the transcript is localised to a time and place where that TRE is present, as we discuss below.

Temporal regulation by successive CREs is common in developmental processes.
For example, in *C. elegans* lin-4 is a non-coding RNA gene crucial for regulating cell fates during the early stages of larval development [@Wightman1993].
Lin-4 is a small RNA that acts as a TRE by binding to a repeated motif in the 3'UTR of its target mRNA lin-14, and inhibiting the translation of lin-14 [@Lee1993].
Since lin-4 is only expressed at the end of the first larval development stage, the 3'UTR motifs in lin-14 only begin to inhibit its translation late into stage 1, initiating the start of stage 2 [@Olsen1999].
The regulation of lin-4 by lin-14 binding to the 3'UTR thus depends on the temporal expression pattern encoded by lin-4's promoter.
Similarly, to establish meiotic chromosome segregation in budding yeast, mRNA encoding cyclin CLB3 is transcribed in stage I of meiosis, but is translationally repressed until stage II of meiosis.
CLB3 is translationally repressed via its 5'UTR by the RNA-binding protein Rim4.
During the transition to meiosis II, Rim4 is phosphorylated which inhibits binding to CLB3 and enables CLB3 to be translated [@Berchowitz2013].
Here, 5'UTR-specified translational control of CLB3 by Rim4 depends on the timing of promoter-specified transcriptional control.

Spatial regulation by successive CREs is also important, as one CRE can facilitate transcript localisation whilst other CREs and/or TREs confer localisation-dependent regulation.
In yeast, the Ash1 protein represses mating-type switching only in daughter cells [@Sil1996], due to localisation of the ASH1 transcript at the bud tip and subsequent localised translation [@Niednery2014].
It is thought that co-transcriptional recruitment of She2 to the ASH1 transcript in the nucleus enables the later recruitment of cytoplasmic factors Khd1/Hek2 and Puf6, factors known for translational repression.
Furthermore, successful transport of ASH1 to the bud tip by She2-She3-Myo4 complexes depends on translational repression by Khd1 and Puf6.
Later, phosphorylation of Khd1 and Puf6 by bud-membrane-localised kinases leads to localised translational activation of the ASH1 mRNP [@Paquin2007; @Deng2008].

Another example where the effect of a CRE on a transcript depends on co-localisation with a regulatory kinase comes from the fungal RNA-binding protein Ssd1.
Yeast cell wall proteins such as Sun4 and Tos6 are translationally repressed by Ssd1 [@Jansen2009], which binds the 5'UTRs of their transcripts [@Hogan2008; @Bayne2020].
It is thought that these transcripts are translationally activated at bud sites after the phosphorylation of Ssd1 by a localised kinase, Cbk1 [@Jansen2009].
There is no evidence that Ssd1 directly acts to transport RNA, so this localised activation presumably depends on localisation that is conferred by other CREs, that recruit other RNA-binding proteins to those transcripts [@Hogan2008], that then in turn recruit transport machinery.
We note that post-transcriptional regulation of SUN4 and TOS6 might explain the repressive effect of their terminators when paired with high-expression promoters (Figure \@ref(fig:pro-ter-platereader-mCherry-mTurq-raw)).

Secondary structures can be integral to motif function, and secondary structure depends on sequences beyond a motif.
Again for ASH1, key cis-regulatory elements depend on secondary structure [@Olivier2005].
Secondary structure is also essential for Smg recognition elements found in transcript targets of Vts1p [@Aviv2006]. 
In mammalian cells, secondary structures have an overall inhibitory effect on motif functionality [@Taliaferro2016].
In yeast, the presence of stable stem loops within the 3â€™UTR reduces the binding of known destabilising RBPs [@Geisberg2020].

Motif behaviour also depends on CREs across the entire host transcript.
Regulation of the ASH1 transcript depends on the concerted action of multiple CREs [@Chartrand1999; @Gonzalez1999].
In other cases, multiple motifs, or multiple copies of the same motif, must occur together to enable recruitment of the effector protein.
Recruitment of Puf3 to promote mRNA transcript decay requires two Puf3 motifs in the 3'UTR [@Jackson2004].
Indeed, clusters of short degenerate motifs have been proposed as a general regulatory feature of RNA-binding proteins [@Cereda2014]


## Implications for novel cis-regulatory element searches

Motif detection algorithms need to progress beyond the assumption that the presence of a motif alone ensures its functional effect.
The effect of a motif can be influenced by factors such as growth conditions and time of measurement.
Uncertainty around the active state of a motif could be introduced using latent variables, as used in transcription factor (TF) binding site prediction algorithms to represent the proportion of active TF sites [@Dai2017; @Sanguinetti2006].
Alternatively, non-linear relationships between factors affecting motif activity can be modeled by partition regression techniques, such as random forests.
Again looking at TF algorithms, partitioning algorithms could be used to predict functional datasets using short sub-sequences of transcripts [@BehjatiArdakani2019]. 

Reductionist search algorithms that attempt to discover the shortest possible sequence contributing to expression may limit the discovery of novel CREs. 
High resolution maps of protein-RNA interactions are revealing RBPs with gapped, multi-partite motifs [@Bayne2020] or motifs that must be repeated in the same transcript to be effective [@Gu2004;@Jackson2004].
Common motif searching approaches, such as MEME Suite [@Bailey2015], have extended beyond fixed length, gapless motifs but may still assume motifs occur independently and only once per transcript [@Frith2008].
Although reducing the search constraints to allow longer, gapped motifs could increase the false positive rate, the availability of precise interactome datasets suggests a need for more flexible search algorithms that effectively discover complex motifs.


## Implications for engineering biology

Our results support previous studies showing unpredictability in the expression of combinations of otherwise well-characterised regulatory elements.
In the creation of biological circuits, for example, time-consuming directed evolution assays have been implemented to overcome mis-matches in component expression levels [@Yokobayashi2002].
The availability of extensively characterised parts libraries now enables large scale searches for CREs with suitable expression.
Together with high-throughput assays, pools of combinations of suitable CREs can be quickly screened for the correct expression levels [@Kosuri2013]. 

Our conceptual model of a gene influences the tools we make, and can limit our ability to predict and manipulate gene expression [@Szymanski2019].
Conceptual models of CREs as independently acting composable parts exemplify these limits.
The characterisation of an element's function in a single context is not enough to make generalisable conclusions.
Both small- and large-scale studies of CREs should ideally be carried out with multiple combinations of other regulatory elements, in cells in multiple growth conditions, to build a comprehensive understanding of transcriptional and post-transcriptional regulation.


# Acknowledgments

We thank Naomi Nakayama, Erika Szymanski, Filippo Menolascina, and members of the Wallace lab for discussions on the manuscript.
We thank Vicent Pelechano and Yujie Zhang for their generous help teaching us the 5PSeq protocol, obtaining 5PSeq data, and discussing the results.



# References

::: {#refs}
:::

\newpage
