# Discussion

We have confirmed that promoter-terminator pairings change the effectiveness of each region's contribution to gene expression.
This supports previous work by Curran et al. 2013 and  Wei et al. 2017 showing that promoter and terminator regions are not entirely modular when regulating gene expression.
We explored whether the changes in regulatory behavior according to overall context is explained by changes in the regulatory effects of individual cis-regulatory elements within these regions.
We have gone on to show that 3'UTR sequence motifs, confirmed to to have a statistically significant contributions to transcript decay rates, have effects on gene expression that are dependent on overall terminator context and promoter pairing. 

## Mechanisms for motif terminator context dependence

Detecting motifs using functional genomics data sets requires that effector proteins are able to a) bind to target transcripts and b) function effectively once bound. 
The behavior of both protein and target mRNA, such as localisation to specific cellular compartments, regulation by cell cycle or reaction to stress, may effect protein function and protein binding concurrently. 
Outside of global behavior, motifs will also be dependent on the secondary structure and presence (or absence) of other regulatory elements within the host transcript. 
Secondary structures can be integral to a motifs function, such as the She2p binding motif spread across stem loops within Ash1 transcripts or the Smg recognition elements found in transcript targets of Vts1p (Olivier et al., 2005; Aviv et al., 2006), or they can have an overall inhibitory effect on binding, with stable stem loop within the 3â€™UTR having a stabilising effect due to a reduction in binding of known destabilising RBPs (Giesberg et al. 2014).
Transcripts often contain multiple motifs for efficient binding as some effector proteins bind indirectly through multiple recruiter proteins, for example Ski2 or Mtr4 recruitment of the exosome for 5'-3' mRNA decay in the cytoplasm or nucleus respectively (Thom et al., 2015; Weick and Lima, 2021), or bind directly with multipartite motifs separated by unconserved gaps of varying length, such as SSD1's general binding motif CNYUCNYU and upstream CCAACU supplementary motif (Bayne et al., 2020) or PUF3's two bipartite motif (Jackson et al., 2004). 
The presence of multiple motifs can also have a negative effect on binding affinity by introducing competition between proteins trying to bind to the same transcript, itself a crucial mechanism of regulation (Lapointe et al., 2017).


## Implications for novel cis-regulatory element searches

The qPCR results for RPS3, TSA1 and PIR1 show motifs vary their contribution to gene expression, but further high-throughput results are required to confirm the extent to which this behavior is reflected across the genome.
Nevertheless, these results cast doubt on whether future computational and experimental searches can continue to assume modularity if more complex regulatory elements are to be found.

Determining co-dependent motifs is further complicated by the fact that distances of varying number of nucleotides can separate motifs (Wilinski et al., 2015).
Incorporating transcript secondary structures into motif searching algorithms has been conducted with limited success, such as the MEMERIS extension to MEME, but only simple conformaties are considered and the effects of protein binding are ignored (Hiller et al., 2006; Achar and Seatrom, 2015). 

## Implications for experimental biology

Our results show that sequence motifs within the 3'UTR have a significant effect on gene expression.
However it is important to note that there is evidence of dependence between the motifs and the sequence context suggesting that post-transcriptional cross-talk between RNA binding proteins can result in the fine-tuning of mRNA stability and translation. 

Further, these different UTR compositions can enable more rapid, dynamic post-transcriptional modulations of expression to be achieved with improved understanding of how different combinations of motifs and secondary structures lead to modulations in expression under different cell conditions, facilitating the generation of more complex phenotypes and enabling more predictable control of flux through metabolic pathway circuits.