---
output:
  pdf_document: default
  html_document: default
---
# Discussion

Our quantitative measurements of protein output from promoter-terminator combinations highlight the limitations of assuming that CREs are quantitatively composable.
Unsurprisingly, fluorescent protein expression levels are dominated by promoter choice, up to 100 fold, and then terminator choice, up to 5 fold. 
However, with up to 1.5 fold change in the relative effect of terminators depending on coding sequence and promoter choice, the quantitative limitations of the assumption of composability are clear. 

Furthermore, we find that the effect of short CREs in 3'UTRs depends on the context of their host construct.
The contribution of our selected motifs to mRNA stability, as inferred from transcriptome-wide analysis, correlated with their effect on the abundance of reporter constructs.
However, their contributions to mRNA abundance were inconsistent and dependent on context: TGTAHMNTA has no measurable effect when inserted into tRPS3 but the expected effect in tTSA1, ATATTC has the expected effect in tRPS3 but little effect in tTSA1, and HWNCATTWY can either decrease or increase RNA levels when removed from tPIR1, depending on the promoter.
In addition, our original motif search predicted the behaviour of a TGTAAATA motif to be stabilising in the [@Sun2013] data set but destabilising in the [@Chan2018] data set.
TGTAAATA changed its contribution in the two data sets because, as a known binding motif for Puf3, carbon deficient conditions trigger Puf3 to degrade its targets [@Miller2014].
Carbon deficient conditions were likely present due to the bespoke CSM-lowURA media used by [@Chan2018], which is similar to the SC-URA media used in our assays.
These disparities in predicted and experimentally verified regulatory behaviour raises questions about current methods for discovering novel cis-regulatory elements and, more generally, for understanding and manipulating regulatory pathways.

## Mechanisms for motif terminator context dependence

The regulatory behavior of a motif depends on the presence of other CREs and, often,  trans-regulatory elements (TREs). 
Active CREs occurring alongside a motif differ not only between genes but also between transcripts from the same gene.
Transcriptional control, such as alternative polyadenylation, and changes in transcript secondary structure, for example due to binding of TREs, can all change the either the activity of the motif directly or accompanying CREs.
Meanwhile, the availability of TREs often depends on characteristics outside of the host transcript.
The localisation of transcripts to different regions of the cell or the coupling of gene expression to the cell cycle can cause changes in TREs concentrations.

The spatiotemporal regulation of gene expression can change the availability of TREs and alter the behaviour of a motif.
Motifs that act as binding sites, as focused on in our study, are dependent on the presence of their trans-acting counterparts.
The availability of a required TRE can be spatially regulated through the localisation of a host transcript, or the binding protein, to a specific organelle.
ASH1 is a DNA-binding protein that facilitates mate-type switching between mother and daughter cells in S.cer [@Sil1996]. 
It is localised to the bud tip by spatial regulation.
A SHE2 binding motif inside the ASH1 mRNA transcript's 3'UTR inhibits translation in the cytoplasm through the recruitment of available PUF6 proteins [@Long2000]. 
As the ASH1 transcript localises to the bud tip, PUF6 proteins are phosphorylated which interferes with SHE2 binding [@Deng2008].
Without PUF6, the SHE2 binding motif is no longer associated with translation inhibition and ASH1 proteins are formed.
Alternatively, TRE availability can be temporally regulated as host transcript and TRE need to be expressed simultaneously in order to function.
In C.elegans, lin-4 is a non-coding RNA gene crucial for regulating cell fates during the early stages of larval development.[@Wightman1993]
It encodes a small RNA transcript that binds to a repeated 3'UTR motif and inhibits the translation of its target, lin-14  [@Lee1993].
Lin-14 is required for the first stage of larval development but it inhibits the progression to the second stage.
Therefore, since lin-4 is only expressed at the end of the first development stage, the 3'UTR motifs in lin-14 only begin to inhibit its translation late into stage 1, initiating the start of stage 2. [@Olsen1999]

Motif behaviour can also be affected dependent on other CRE's across the entire host transcript.
Secondary structures can be integral to a motifs function, such as the SHE2 binding motif spread across stem loops within ASH1 transcripts or the Smg recognition elements found in transcript targets of Vts1p [@Olivier2005; @Aviv2006]. 
In other cases, secondary structures can have an overall inhibitory effect on binding, with the presence of stable stem loops within the 3â€™UTR reducing the binding of known destabilising RBPs [@Geisberg2020].
For others still, multiple motifs, or multiple copies of the same motif, must occur together to enable recruitment of the effector protein.  
Recruitment of the Puf3, for example, in the signaling of mRNA transcript decay requires two motifs to be appear in the 3'UTR [@Jackson2004].


## Implications for novel cis-regulatory element searches

Motif detection algorithms need to revisit the assumption that the presence of a motif alone ensures its functional effect.
The effect of a motif can be significantly influenced by experimental factors, such as; growth conditions and time of measurement.
Uncertainty around the active state of a motif could be introduced using latent variables, as used in transcription factor (TF) binding site prediction algorithms to represent the proportion of active TF sites [@Dai2017; @Sanguinetti2006].
Alternatively, non-linear relationships between factors affecting motif activity can be modeled by partition regression techniques, such as random forests.
Again looking at TF algorithms, partitioning algorithms could be used to predict functional data sets using short sub-sequences of transcripts [@BehjatiArdakani2019]. 

Reductionist search algorithms that attempt to discover the shortest possible sequence contributing to expression are limiting the discovery of novel CREs. 
High resolution maps of protein-RNA interactions are revealing RBPs with gapped, multi-partite motifs [@Bayne2020] or motifs that must be repeated in the same transcript to be effective [@Gu2004].
Common motif searching algorithms, such as MEME Suite [@Bailey2015], have extended beyond fixed length, gapless motifs but still assume motifs occur independently and only once per transcript [@Frith2008].
Although reducing the search constraints to allow longer, gaped motifs does increase the false discovery rate, the availability of precise interactome data sets shifts the priority towards more flexible search algorithms.

## Implications for experimental biology

Our results support previous studies showing unpredictability in the expression of combinations of otherwise well-characterised regulatory elements.
In the creation of biological circuits, for example, time-consuming directed evolution assays have been implemented to overcome mis-matches in component expression levels [@Yokobayashi2002].
The availability of extensively characterised parts libraries now enables large scale searches for CREs with suitable expression.
Together with high-throughput assays, pools of combinations of suitable CREs can be quickly screened for the correct expression levels [@Kosuri2013]. 

Future work on CREs will need to respect the fact that characterisation of elements' function in a single context is not enough to make generalisable conclusions.
Both small- and large-scale studies of CREs should ideally be carried out with multiple combinations of other regulatory elements, in cells in multiple growth conditions, to build a comprehensive understanding of transcriptional and post-transcriptional regulation.
