---
output:
  pdf_document: default
  html_document: default
---
# Discussion

Our quantitative measurements of promoter-terminator swaps and cis-regulatory element combinations show that the practical abstraction of genes into modular parts which is implemented across biology does not correlate with the functional modularity of regulatory parts.
Building upon previous work by @Curran2013, we have presented quantitative differences in relative RNA and protein levels for terminators paired with different promoters and ORFs.
Furthermore, we have extended the exploration of the context dependence of gene expression regulation by determining changes in contributions of individual cis-regulatory elements.
Our chosen cis-regulatory elements, shortlisted by predicting half life, displayed contributions to overall gene expression when inserted/deleted from three host terminators.
However, contributions to gene expression depends highly on its host terminator and its paired promoter; TGTAHMNTA has no measurable effect when inserted into tRPS3, ATATTC has little to no effect in tTSA1, meanwhile HWNCATTWY can have decreased or increased RNA levels when removed from tPIR1 depending on promoter.
The disparity between practical modularity and functional modularity raises issues with current methods for discovering novel cis-regulatory elements and understanding and manipulating regulatory process across biology.

\@ref(fig:hlife-decay-model)C with their magnitudes calculated according to the data sets plotted against each other.
Interestingly, the TGTAAATA motif was predicted to be stabilising in the [@Sun2013] data set but destabilising in the [@Chan2018] data set.
The puf3 protein associated with motif is known to change function in carbon deficient condition which were likely present due to the bespoke CSM-lowURA media the cultures were grown in overnight, [@Sun2013] used standard YPD media. 


## Mechanisms for motif terminator context dependence

The regulatory behavior of a motif can be explained by its internal and external context. 
A motif's external context refers to characteristics outside of the host transcript.
It includes transcriptional control, such a differential expression or alternative polyadenylation, and the presence of trans-regulatory elements (TRE), such as RBPs.
A motif's internal context refers to the characteristics of the host transcript.
It includes the secondary structure of the host transcript and the presence of other motifs concurrently.

The external context of a motif can be controlled by spatiotemporal regulation.
Motifs that act as binding sites, as focused on in our study, are dependent on the presence of TREs in their external context.
The availability of a required TRE can be spatially regulated through the localisation of a host transcript or a TRE to a specific organelle.
ASH1 is a DNA-binding protein localised to the bud tip that facilitates mate-type switching between mother and daughter cells in S.cer. 
A SHE2 binding motif inside the ASH1 mRNA transcript's 3'UTR inhibits translation in the cytoplasm as SHE2 recruits available PUF6 proteins. 
As the ASH1 transcript localises to the bud tip, PUF6 proteins are phosphorylated which interferes with SHE2 binding.
Without PUF6 the SHE2 binding motif is no longer associated with translation inhibition and ASH1 proteins are formed.
Alternatively, TRE availability can be temporally regulated as transcript and TRE need to be expressed simultaneously in order to function.

The internal context of a motif can be critical in enabling regulation.
Motifs will are dependent on the secondary structure and presence (or absence) of other regulatory elements within the host transcript.
Secondary structures can be integral to a motifs function, such as the SHE2 binding motif spread across stem loops within ASH1 transcripts or the Smg recognition elements found in transcript targets of Vts1p (Olivier et al., 2005; Aviv et al., 2006). 
In other cases secondary structures can have an overall inhibitory effect on binding, with the presence of stable stem loops within the 3â€™UTR reducing the binding of known destabilising RBPs (Giesberg et al. 2014).
Other motifs can only effectively bind their TREs in the presence of bound cooperative proteins.
In human cells, microRNAs in association with Argonaute (AGO) proteins have been shown to be bind to target mRNA cooperatively with PUM1.  
For certain AGO mRNA targets its binding affinity is negligible unless the target's secondary structure is switched by the binding of PUM1 to its own motif. (Sternburg et al, 2018)
As AGO is involved in RNA silencing and its targets are key for cell cycle control, it has been shown that certain cancer cells remove PUM binding sites through alternative polyadenlytion in order to avoid cell cycle regulation. (Miles et al, 2012)

## Implications for novel cis-regulatory element searches

Future motif detecting algorithms need to incorporate transcript context: secondary structure, localisation and cell cycle expression, for more accurate and comprehensive search results.
Secondary structure features can be included in motif searches a priori by setting nucleotides within single stranded regions to be more likely to be part of motifs, by incorporating predicted binding energy.
Non-linear relationships between intrinsic and extrinsic transcript characteristics can be modeled by partition regression techniques, such as random forests.
Linear regression algorithms assume increased motif occurrence must increase protein binding, which causes a subsequent increase in the protein's functional effect.
However, some motifs may need to be present above a threshold before sufficiently affecting binding or can behave differently in the presence of other co-regulatory motifs.
In addition, linear models will also struggle with the behavior of some extrinsic effects, such as the availability of a binding protein in a particular organelle or at a cell cycle stage.
These are either dominant predictors of motif efficacy or entirely irrelevant with such extreme behavior being poorly modeled by linear regression. 
Combining partitioning algorithms to separate transcripts with dichotomised features followed by training linear regression models on the separate pool could best predict functional data sets, although overfitting will be serious concern. 

Transcripts often contain multiple motifs for efficient binding as some effector proteins bind indirectly through multiple recruiter proteins, for example Ski2 or Mtr4 recruitment of the exosome for 5'-3' mRNA decay in the cytoplasm or nucleus respectively (Thom et al., 2015; Weick and Lima, 2021), or bind directly with multi-partite motifs separated by unconserved gaps of varying length, such as SSD1's general binding motif CNYUCNYU and upstream CCAACU supplementary motif (Bayne et al., 2020) or PUF3's two bipartite motifs (Jackson et al., 2004). 


The growing catalogue of RBPs with multiple binding motifs has lead to a demand for methods that discover multi-partite motifs, which may be separated in a transcript's linear sequence but adjacent in the final 3D structure.
Algorithms that determine position weight matrices for a linear segment of fixed length are unreliable when searching for motifs of variable widths and struggle with multi-partate motifs unless parts are directly adjacent in the linear sequence.
k-mer based searches ignore motif position, enabling for the detection of separated, multi-partate motifs, but can only detect exact matches, therefore motifs with highly redundant nucleotides are missed.
Deep learning algorithms can accurately model complex motif interactions, but interpreting the model features into biologically relevant motifs remains a serious constraint.

## Implications for experimental biology

Our results show that sequence motifs within the 3'UTR have a significant effect on gene expression.
However it is important to note that there is evidence of dependence between the motifs and the sequence context suggesting that post-transcriptional cross-talk between RNA binding proteins can result in the fine-tuning of mRNA stability and translation. 

Further, these different UTR compositions can enable more rapid, dynamic post-transcriptional modulations of expression to be achieved with improved understanding of how different combinations of motifs and secondary structures lead to modulations in expression under different cell conditions, facilitating the generation of more complex phenotypes and enabling more predictable control of flux through metabolic pathway circuits.