---
output:
  pdf_document: default
  html_document: default
---
# Discussion

Our quantitative measurements of protein output from promoter-terminator combinations highlight the limitations of assuming that CREs are quantitatively composable.
Unsurprisingly, fluorescent protein expression levels are dominated by promoter choice, up to 100 fold, and then terminator choice, up to 5 fold. 
However, with up to 1.5 fold change in the relative effect of terminators depending on coding sequence and promoter choice, the quantitative limitations of the assumption of composability are clear. 

Furthermore, we find that the effect of regulatory motifs in 3'UTRs depends on the context of their host construct.
The contribution of our selected motifs to mRNA stability, as inferred from transcriptome-wide analysis, correlated with their effect on the abundance of reporter constructs.
However, their contributions to mRNA abundance were inconsistent and dependent on context: TGTAHMNTA has no measurable effect when inserted into tRPS3 but the expected effect in tTSA1, ATATTC has the expected effect in tRPS3 but little effect in tTSA1, and HWNCATTWY can either decrease or increase RNA levels when removed from tPIR1, depending on the promoter.
These disparities in predicted and experimentally verified regulatory behaviour raises questions about current methods for discovering novel cis-regulatory elements and, more generally, for understanding and manipulating regulatory pathways.

In addition, our transcriptome-wide motif analysis predicted the behaviour of a TGTAAATA motif to be stabilising in the [@Sun2013] data set but destabilising in the [@Chan2018] data set.
This difference could be explained by carbon-dependent regulation of Puf3: TGTAAATA is the binding motif for Puf3, and carbon deficient conditions trigger Puf3 to degrade its targets [@Miller2014].
Carbon deficient conditions were likely present in the CSM-lowURA media used by [@Chan2018], which is similar to the SC-URA media used in our assays, but not in the YPD-rich media used by [@Sun2013].
This raises further questions about the need to explore CRE contributions in diverse growth conditions.

## Mechanisms for CRE context dependence

The regulatory behavior of a motif depends on the presence of other CREs in the host gene and, usually, on trans-regulatory elements (TREs) that are regulated in turn.
These effects are not separable, because spatiotemporal regulation of gene expression conferred by one CRE can change the interaction of a second CRE with its relevant TREs.
For example, a motif may act as the regulatory binding site of an RNA-binding protein only after the transcript is localised to a time and place where that protein is present.

Temporal regulation by successive CREs is common in developmental processes.
For example, *C. elegans* lin-4 is a non-coding RNA gene crucial for regulating cell fates during the early stages of larval development.[@Wightman1993]
Lin-4 is a small RNA transcript that acts as a TRE by binding to to a repeated motif in the 3'UTR of its target mRNA lin-14, and inhibiting the translation of lin-14 [@Lee1993].
Since lin-4 is only expressed at the end of the first larval development stage, the 3'UTR motifs in lin-14 only begin to inhibit its translation late into stage 1, initiating the start of stage 2 [@Olsen1999].
The regulation of lin-4 by lin-14 binding to the 3'UTR thus depends on the temporal expression pattern encoded by lin-4's promoter.

Spatial regulation by successive CREs is also important, as one CRE can confer localisation followed by another CRE conferring localisation-dependent regulation.
In yeast, the Ash1 protein represses mating-type switching only in daughter cells [@Sil1996], due to localisation of the ASH1 transcript at the bud tip and subsequent localised translation [@Cosma2004].
It is thought that recruitment of cytoplasmic factors Khd1/Hek2 and Puf6 to ASH1 depends on prior co-transcriptional recruitment of She2, both recognising cognate CREs [@Long2000].
Conversely, successful transport by She2-She3-Myo4 complexes depends on translational repression by Khd1 and Puf6 [@Long2000].
Later, translational activation due to phosphorylation of Khd1 and Puf6 by bud-membrane-localised kinases depends on the successful localisation of the ASH1 mRNP [@Deng2008].
Similarly, yeast cell wall proteins such as Sun4 and Tos6 are translationally regulated by the repressive RNA-binding protein Ssd1 [@Jansen2009]. 
Ssd1 binds the 5'UTRs of SUN4 and TOS6 transcripts [@Hogan2008,@Bayne2020], and it is thought that these transcripts are translationally activated at bud sites by the phosphorylation of Ssd1 by a localised kinase, Cbk1 [@Jansen2009].
So, the effect of the Ssd1 CRE on a transcript depends on the co-localisation of the Ssd1-bound transcript with the Cbk1 kinase.

Secondary structures can be integral to motif function, and secondary structure depends on sequences beyond a motif.
Again for ASH1, key cis-regulatory elements depend on secondary structure
Secondary structure is required for Smg recognition elements found in transcript targets of Vts1p [@Olivier2005; @Aviv2006]. 
In mammalian cells, secondary structures have an overall inhibitory effect on motif functionality [@Taliaferro2016].
In yeast, the presence of stable stem loops within the 3â€™UTR reduces the binding of known destabilising RBPs [@Geisberg2020].

Motif behaviour also depends on CREs across the entire host transcript.
Regulation of ASH1 transcript depends on the concerted action of multiple CREs [@Long1999, @Gonzalez1999].
In other cases, multiple motifs, or multiple copies of the same motif, must occur together to enable recruitment of the effector protein.
Recruitment of Puf3 to promote mRNA transcript decay requires two Puf3 motifs in the 3'UTR [@Jackson2004].
Clusters of short degenerate motifs, referred to as multivalent RNA motifs, have been proposed as a general regulatory feature of RNA-binding proteins [@Cereda2014]

Active CREs occurring alongside a motif differ not only between genes but also between transcript isoforms.
Alternative splicing or polyadenylation are encoded by CREs, and then affect which CREs are present on the mature transcript.
HERE WE NEED A COUPLE OF EXAMPLES (Geisberg?) OR TO REMOVE THE PARAGRAPH.


## Implications for novel cis-regulatory element searches

Motif detection algorithms need to revisit the assumption that the presence of a motif alone ensures its functional effect.
The effect of a motif can be significantly influenced by experimental factors, such as; growth conditions and time of measurement.
Uncertainty around the active state of a motif could be introduced using latent variables, as used in transcription factor (TF) binding site prediction algorithms to represent the proportion of active TF sites [@Dai2017; @Sanguinetti2006].
Alternatively, non-linear relationships between factors affecting motif activity can be modeled by partition regression techniques, such as random forests.
Again looking at TF algorithms, partitioning algorithms could be used to predict functional data sets using short sub-sequences of transcripts [@BehjatiArdakani2019]. 

Reductionist search algorithms that attempt to discover the shortest possible sequence contributing to expression may limit the discovery of novel CREs. 
High resolution maps of protein-RNA interactions are revealing RBPs with gapped, multi-partite motifs [@Bayne2020] or motifs that must be repeated in the same transcript to be effective [@Gu2004].
Common motif searching approaches, such as MEME Suite [@Bailey2015], have extended beyond fixed length, gapless motifs but may still assume motifs occur independently and only once per transcript [@Frith2008].
Although reducing the search constraints to allow longer, gapped motifs does increase the false discovery rate, the availability of precise interactome data sets shifts the priority towards more flexible search algorithms.

## Implications for engineering biology

Our results support previous studies showing unpredictability in the expression of combinations of otherwise well-characterised regulatory elements.
In the creation of biological circuits, for example, time-consuming directed evolution assays have been implemented to overcome mis-matches in component expression levels [@Yokobayashi2002].
The availability of extensively characterised parts libraries now enables large scale searches for CREs with suitable expression.
Together with high-throughput assays, pools of combinations of suitable CREs can be quickly screened for the correct expression levels [@Kosuri2013]. 

Future work on CREs will need to respect the fact that characterisation of elements' function in a single context is not enough to make generalisable conclusions.
Both small- and large-scale studies of CREs should ideally be carried out with multiple combinations of other regulatory elements, in cells in multiple growth conditions, to build a comprehensive understanding of transcriptional and post-transcriptional regulation.
