# Discussion

We have confirmed that promoter-terminator pairings change the effectiveness of each region's contribution to gene expression. This supports previous work by Curran et al. 2013 and  Wei et al. 2017 showing that promoter and terminator regions are not entirely modular when regulating gene expression. We explored whether the changes in regulatory behavior according to overall context is explained by changes in the regulatory effects of individual cis-regulatory elements within these regions. We have shown that 3'UTR sequence motifs, confirmed to to have a statistically significant contributions to transcript decay rates, have effects on gene expression that are dependent on overall terminator context and promoter pairing. 

## Mechanisms for motif termintor context dependence

Encoded within the terminator sequence, the 3’ untranslated region (UTR) has been shown to play an important role in coordinating transcript stability and localisation (Mayr 2016). These functions are largely mediated by the association of RNA binding proteins (RBP) to the corresponding cis-regulatory motif on the 3’UTR and subsequent recruitment effector proteins. Therefore the changes to the mRNA fate is determined by the available binding sites on the 3’UTR for RBPs at a given moment. In addition, the binding affinities of mRNAs and their effector proteins are not solely defined by the presence or absence of sequence motifs.  Transcript secondary structures and protein-protein interactions between concurrently binding RBPs can contribute to overall recruitment.

Transcript secondary structures can alter RNA-protein binding affinity leading to motif dependence on terminator context. Even for same-gene isoforms, transcripts differing only in 3'UTR length can have significantly different upstream secondary structures and up to 2-fold changes in Pab1 binding (Moqtaderi et al., 2018). Sequence motifs may also need to be within structural motifs to facilitate protein binding such as the She2p binding motif spread across stem loops in Ash1 transcripts or the Smg recognition elements found in transcript targets of Vts1p (Olivier et al., 2005; Aviv et al., 2006). Although, the more common requirement for protein binding is for the sequence motif to be within unstructured single-stranded regions of RNA (Ellis et al., 2007). And, the introduction of a stable stem loop within the 3’UTR has been shown to have a stabilising effect and associated reduction in binding of known destabilising RBPs (Giesberg et al. 2014).

Terminators containing multiple adjacent cis-regulatory motifs can enhance motif effects by enabling interactions between co-dependent proteins or reduce motif effects by introducing binding competition. The cooperativity between multiple RBP binding to adjacent cis-regulatory motifs in an mRNA have been shown to improve affinity and specificity of She2p and She3p RBP binding to ASH1 mRNA compared to their individual interactions (Shen et al. 2009; Niedner et al. 2014). Adjacent cis-regulatory motifs on a 3’UTR or motifs brought into close proximity due to secondary structures can also have a competitive interaction, with preferential binding of one RBP through improved affinity mediated under specific conditions (Dassi 2017). The accessibility of RBPs to these cis-regulatory elements are also influenced by structural modifications or localisation to different compartments of the cell. 


## Implications for novel cis-regulatory element searches

The qPCR results for RPS3, TSA1 and PIR1 show motifs vary their contribution to gene expression, but further high-throughput results are required to confirm the extent to which this behavior is reflected across the genome. Nevertheless, these results cast doubt on whether future computational and experimental searches can continue to assume modularity if more complex regulatory elements are to be found.

Determining co-dependent motifs is further complicated by the fact that distances of varying number of nucleotides can separate motifs (Wilinski et al., 2015). Incorporating transcript secondary structures into motif searching algorithms has been conducted with limited success, such as the MEMERIS extension to MEME, but only simple conformaties are considered and the effects of protein binding are ignored (Hiller et al., 2006; Achar and Seatrom, 2015). 

## Implications for experimental biology

Our results suggest that fold changes in overall gene expression due to poor promoter-terminator pairings are an order of magnitude lower than the selection of an optimal promoter alone. Although this implies promoter-terminator pairings will have minimal impact on the construction of genetic pathways. 

