# Discussion

Saccharomyces cerevisiae is a well-established host platform for industrial production due to its traits as a robust organism under fermentation conditions, the wealth of molecular biology tools developed for its genetic manipulation and its status as a generally regarded as safe (GRAS) organism. Over the past two decades, it has been optimised for the production of antibiotics, painkillers, antioxidants and anti-cancer drugs (Siewers et al 2009, Gidijala et al. 2009, Yan et al. 2005, Koopman et al. 2012, Rodriguez et al. 2017). This process for pathway engineering is reliant on precise capabilities in modulating native and heterologous enzyme expression. Over the past few years, the development of an increasing number of modular toolkits containing standardised genetic parts for the rapid and combinatorial assembly of genes have expedited our capability to undergo multiple iterations of the design-build-test-learn cycle for pathway optimisatiion (Lee et al. 2015, Garcia-Ruiz et al. 2018, Andreou et al. 2018). 

These toolkits often utilise native yeast promoter/5'UTR and 3'UTR/terminator regions to provide parts to enable modular assembly of genes for pathway expression. Non-coding regulatory elements, comprising of promoters, untranslated regions (5’UTR and 3’UTR) and terminators, possess a large amount of information for the regulation of mRNA and protein abundance. Perhaps the best studied is the role of the promoter, which plays a central role in coordinating the rate of new transcript production (Ottoz et al. 2018). Transcription factor (TF) binding sites upstream of the core promoter region modulate promoter activity through the recruitment of single TFs or synergistic combinations to facilitate recruitment of transcriptional machinery. Coupled with chromatin context and other bound TFs, these ultimately control transcription rates (Levo 2017, Blazeck et al. 2012). The importance of terminator choice in gene expression has not been as extensively studied, with only a handful of common terminators (tADH1, tCYC1, tPGK1) often used for heterologous expression or pathway asssemblies.

Several groups have sought to elucidate the effect of different native yeast terminators with a goal of maximal protein expression. Yamanishi et al. identified a terminator (tTPS1) which increased fluorescent protein production by 50% compared to the tCYC1 terminator (Yamanishi et al. 2011). Ito et al. identified tDIT1 was associated with a 2.5-fold improved protein production mediated by the recruitment of two RBPs (Nab6 and Pap1) to the 3’UTR (Ito et al. 2016). The same group further identified that the effect of a library of 5302 terminators spanned an 81-fold range difference in fluorescence production, with several resulting in a doubling of protein production relative to tPGK1 (Yamanishi et al. 2013). When coupled with different promoters, the relative difference in protein output between terminators vary, with a maximum difference of 11-fold when coupled with a low expression promoter (Curran et al. 2013, Wei et al. 2017).

The identification of the functional elements within the terminator region which control gene expression are confounded by its combined involvement in transcriptional, post-transcriptional and translational regulation. The functional elements contained in the terminator are known to recoordinate transcriptional termination and polyadenylation (Guo et al. 1996) but after transcription a sub-sequence of variable length remains to form the 3'UTR of the mRNA transcript. The 3'UTR itself has been shown to play important roles in coordinating mRNA stability, translation and localisation. It contributes through the formation of secondary or tertiary structures and by containing binding motifs for RNA binding proteins (RBPs), both leading to dynamic RNA-protein interactions that are responsive to metabolic or environmental stimuli. Meanwhile, inferring a cis-regulatory motifs role from its paired RBPs is itself difficult because they bind many mRNA targets but binding specificity can be modulated through their association with protein interaction partners. For example She2p and She3p RBPs show enhanced blinding to ASH1 mRNA compared to their individual interactions, leading to localisation of ASH1 mRNA to the budding daughter cell (Shen et al.  2009, Niedner et al. 2014). As such changes to the mRNA fate is determined by the available binding sites on the UTR for RBPs at a given moment. 

To control the numerous effects confounding the search for 3'UTR cis-regulatory motifs, experimental methods for the identification and dissection of RNA-protein interactions networks have predominately been in vitro. These techniques centre around the immobilisation of either mRNA (Hartmuth et al. 2004) or proteins (Kretz et al. 2013, Siprashvili et al. 2016) before exposing them to cell extract. Whilst these approaches maximise control, they remove much of the cellular context, with synthesised mRNA being devoid of post transcriptional chemical modifications. In contrast, in vivo strategies centre around cross-linking RNA and RBPs in living cells using UV or formaldehyde to form covalent bonds between RNA and proteins that are in direct contact with each other. Formaldehyde based cross-linking has the advantage of much higher efficiency of cross-linking between RNA and RNA binding proteins, however a key caveat is that protein-protein cross-linking also occurs, meaning that it can be more challenging to discern between RBPs and proteins that interact with RBPs. In contrast UV cross-linking approaches are much more specific, however not only is cross-linking less efficient, but further biases have been observed for cross-linking with uridines (Sugimoto et al. 2012) and varying efficiencies of cross-linking are seen across different amino acids (Meisenheimer et al. 1997).