# Materials and methods

### Strains and media
The BY4741 (*MATa his3$\Delta1$ leu2$\Delta0$ met15$\Delta0$ ura3$\Delta0$*) wild type strain was used in this study and grown in either synthetic complete medium without uracil (SC -URA medium), containing 0.69% Yeast Nitrogen Base without amino acids and with ammonium sulfate (Formedium, Norfolk, UK), 0.193% amino acid drop-out supplement mixture (Formedium, Norfolk, UK) and 2% glucose, or YPDA medium, containing 2% peptone, 1% yeast extract, 2% glucose and 0.004% adenine. 

### Determining 3'UTR decay motifs

Following the methodology presented in Cheng et al (2017), a linear model predicting transcript half life using codon usage and 3'UTR motif frequency was used to determine motif significance. 
$$log_2(\lambda_g^{1/2}) = \sum_c\beta_cp_{cg} + \sum_m\alpha_mn_{mg} + \epsilon_g$$
where $\lambda_g^{1/2}$ is the half life of gene g, $\beta_c$ is the coefficient associate with codon c, $p_{cg}$ is the proportion of gene g's ORF that corresponds to codon c, $\alpha_m$ is the coefficient associated with motif m, $n_{mg}$ is number of counts of motif m in gene g's 3'UTR and $\epsilon_g$ is the noise term. The original list of 69 3'UTR motifs to investigate was collated from three separate studies of cis-regulatory sequences suspected to regulate mRNA decay: Hogan et al (2008), Shalgi et al (2005) and Cheng et al (2017). The Pelechano et al (2013) 3'UTR data set of median length yeast 3'UTR's was used to calculate the frequency of each of the 69 motifs in every gene across the whole yeast genome. Our analysis was performed separately on two independent, genome-wide studies of yeast mRNA decay rates to confirm motif significance, Sun et al (2013) and Chan et al (2017).

### Selection criteria of 3'UTRs devoid of sequence motifs of interest (for introducing sequence motifs)

A pair of suitable 3'UTR's from the Pelechano et al (2013) median length 3'UTR dataset were chosen for introducing the selected motifs of interest. The primary selection criteria for these 3'UTRs where; 1) native 3'UTR does not contain any motif of interest, 2) <300 nt in length including terminator, 3) top 5% most abundant in the Nadal-Ribelles et al (2019) scRNA-seq dataset. The selection process is summarised in supplementary figure 1 with supplementary table 1 containing the shortlist of genes.


- Extract 3'UTR sequences (Median 3'UTR lengths from Pelechano dataset)
- Filter for 3'UTRs of less than 300 bp
- Motif table of interest
- Screen for 3'UTRs without motifs
- Check if it can be synthesised by IDT
- Check if there are no BsaI or BsmBI sites
- Introduce insertion sites (M1, M2 and M3) that minimally disrupts 3'UTR secondary structure compared to WT 3'UTR

### Selection criteria of 3'UTRs with sequence motifs of interest (for removing sequence motifs)
- Extract 3'UTR sequences (Median 3'UTR lengths from Pelechano dataset)
- Filter for 3'UTRs of less than 300 bp
- Motif table of interest
- Screen for 3'UTRs with decay motifs
- Check if it can be synthesised by IDT
- Check if there are no BsaI or BsmBI sites
- Design 3'UTRs without motifs

### Construction of chimera strains
All chimera plasmids were constructed as described in Garcia-Ruiz et al. 2018. Promoters, ORFs and terminators were either amplified from the yeast genome or synthesised by a commercial vendor (IDT) then cloned into a parts accepting vector (HcKan_P for promoters, HcKan_O for ORFs and HcKan_T for terminators) by Golden Gate assembly as described in Garcia-Ruiz et al. 2018. Reporter/chimeric plasmids, consisting of an mCherry gene expressed under the control of either pRPS3, pPGK1 or pTSA1. The tRPS3 and tTSA1 3'UTR-terminators consist of the median length of the 3'UTR for its respective gene (RPS3 and TSA1) as described in Pelechano et al. 2013, along with ~150 bp of the sequence downstream of the 3'UTR **Check if it is 150 bp **. The promoter, ORF and terminators are asssembled together into the  transcription unit acceptor vector (POT1-ccdB) by Golden Gate assembly as described in Garcia-Ruiz et al. 2018. Plasmids were confrmed by PCR and Sanger sequencing (MRC PPU DNA Sequencing and Services, Dundee). These plasmids were transformed into BY4741 and selecting in Sc-URA plates for isolating yeast **(how - LiOAC & SS)** containing the chimeric plasmids. DNA sequences used in this study are summarised in **Supplementary table 1**


### Strain growth, RNA extraction and reverse transcription
BY4741 yeast strains containing the chimeric reporter plasmid were pre-cultured in a 12 column deep well plate containing 5 ml **Check volume** of SC -URA medium containing 2% glucose and grown for ~20 hours at 30&deg;C in a shaking incbator. Following the overnight growth, the OD~600~ was measured using a spectrophotometer and diluted in a 12 column deep well plate to a starting OD~600~ of 0.2. Diluted cultures were grown at 30&deg;C in a shaking incbator to an OD~600~ of 0.5-0.6 then pelleted by centrifugation. All remaining media was removed with an aspirator. 

The pelleted cells were resuspended in 400 $\mu$l RNA binding buffer (Zymo cat #R1013-2) transerred to 2 ml screw cap tubes containing zirconia beads, lysed using the Precellys Evolution homogeniser **Should we include the actual protocol + number of cycles?** then centrifuged. The supernatant was transferred to a Zymo Spin IIICG column (Zymo cat #C1006) then centrifuged. The flow through is mixed with 1 volume of ethanol then transferred to a Zymo Spin IIC column (Zymo cat #C1011) and centrifuged. The flow through was discarded and 1 volume of DNA/RNA Prep buffer (Zymo cat #D7010-2) was added then centrifuged. The column was washed with 700 $\mu$l of Zymo DNA/RNA Wash buffer (Zymo cat #D7010-3) then centrifuged. The column was washed a second time, but with 400 $\mu$l of Zymo DNA/RNA Wash buffer (Zymo cat #D7010-3). The column was centrifuged once more to remove residual wash buffer in the column. 30 $\mu$l of nuclease free water was added to the column then eluted.

The quantity and quality of the RNA was measured using both the nanodrop **Check brand and company** and Fragment Analyser (Agilent). 4 $\mu$g of RNA is treated with Turbo DNAse (Invitrogen cat #AM2238). 2.5 $\mu$l of Random primer mix (NEB cat #S1330S) was added to the mixture then separated into 2 PCR tubes, one for -RT and one for +RT). The RNA is denatured at 70&deg;C then cooled on ice. Reverse transcription (RT) master mix was prepared, containing 2 $\mu$l of First Strand synthesis buffer, 0.75 $\mu$l of 10mM dNTP mix, 1.5 $\mu$l of nuclease free water, 0.25 $\mu$l of RNase inhibitor and 0.5 $\mu$l of SuperScript IV Reverse Transcriptase (Invitrogen cat #18091050) per reactioon. 5 $\mu$l of the RT master mix was added to the denatured RNA then incubated at 25&deg;C for 5 minutes then 55&deg;C for 1 hour. The cDNA was diluted with 180 $\mu$l of nuclease free water.


### Quantitative PCR analysis
Quantification of target cDNAs were measured by quantitative PCR with Brilliant II Fast SYBR Green QPCR Master Mix (Agilent) using a Lightcycler 480 qPCR machine (Roche). RPS3, PGK1 and URA3 were used as the internal standard. **qPCR program**

### RT-qPCR data analysis and linear modelling 
qPCR data was analysed using the tidyqPCR R package. $\Delta$Cq values calculated by normalising the mCherry Cq values by the median Cq values of the internal standards (RPS3, PGK1 and URA3). $\Delta\Delta$Cq values were calculated by normalising mCherry $\Delta$Cq the by the control mod0 strain. 